<!DOCTYPE html>
<!-- saved from url=(0066)http://www.sciencedirect.com/science/article/pii/S0888754310000492 -->
<html><script type="text/javascript" async="" src="./Assembly algorithms for next-generation sequencing data_files/204774041.js"></script><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><script type="text/javascript" async="" src="./Assembly algorithms for next-generation sequencing data_files/geo2.js"></script>
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">

<meta name="robots" content="NOARCHIVE">
<meta name="robots" content="noodp">
<meta name="robots" content="noydir">
 
    <meta name="citation_fulltext_world_readable" content="">  

<link rel="canonical" href="./Assembly algorithms for next-generation sequencing data_files/Assembly algorithms for next-generation sequencing data.html">


<link rel="shortcut icon" href="http://cdn.els-cdn.com/sd/favSD.ico" type="image/x-icon">
<title>
 Assembly algorithms for next-generation sequencing data
 </title> 

<link type="text/css" rel="StyleSheet" href="./Assembly algorithms for next-generation sequencing data_files/gadgets_ext,gadgets,toolbar,SDArticleGadgetOverRide,jquery-ui-1.8.4.custom.css">


<link rel="stylesheet" href="./Assembly algorithms for next-generation sequencing data_files/article_1601R2.css" type="text/css">
<link rel="stylesheet" href="./Assembly algorithms for next-generation sequencing data_files/articleSprite_1601R2.css" type="text/css">
<link rel="stylesheet" href="./Assembly algorithms for next-generation sequencing data_files/jquery-ui_1601R2.css" type="text/css">




<!--[if lt IE 9]>
<link rel="stylesheet" Title="Default Styles" href="http://cdn.els-cdn.com/sd/css/css_gen_ielt9_1601R2.css" type="text/css">
<![endif]-->

<script>
var ab=new Object();
ab.entitled = true;
</script>
<script>(function() {var scriptTag = document.createElement('script'); 
scriptTag.type = 'text/javascript';scriptTag.async = true;
scriptTag.src = '//cdn.optimizely.com/js/204774041.js';
var head = document.getElementsByTagName('head')[0];
head.parentNode.insertBefore(scriptTag, head);
})();function optimizelyTimeout() {
window.optimizely = window.optimizely|| []; 
if(!window.optimizely.data) {
window.optimizely.push("timeout"); } }
setTimeout(optimizelyTimeout, 1000); </script>

<script>
var prs = new Object();prs.rt=function(label,ts){this[label]=(ts||ts==0?ts:new Date().getTime());};
</script>



<!--[if lte IE 7]>
<script src="http://cdn.els-cdn.com/sd/ut/js/json2_1601R2.js.min"></script>
<![endif]-->



<style type="text/css" media="print">.usabilla_live_button_container { display: none; }</style><script src="./Assembly algorithms for next-generation sequencing data_files/shindig-container-pubsub-2.js" class="lazyload" charset="utf-8"></script><script src="./Assembly algorithms for next-generation sequencing data_files/SciverseShindigService,GadgetToolBar,jquery-outside,jquery.ba-bbq.min,SciverseShindigContainer,ScienceDirectShindigServices.js" class="lazyload" charset="utf-8"></script></head>

<body><div id="lightningjs-usabilla_live" style="display: none;"><div><iframe frameborder="0" id="lightningjs-frame-usabilla_live"></iframe></div></div><div id="isIElte8test"><!--[if lte IE 8]><div id="isIElte8"></div><![endif]--></div><div id="isIE9test"><!--[if  IE 9]><div id="isIE9"></div><![endif]--></div><div id="isIE8test"><!--[if  IE 8]><div id="isIE8"></div><![endif]--></div><div id="isIE7test"><!--[if  IE 7]><div id="isIE7"></div><![endif]--></div><div id="isIElt7test"><!--[if lt IE 7]><div id="isIElt7"></div><![endif]--></div>

<div class="ua_btn" role="region" aria-label="screen reader compatability"><a role="button" rel="nofollow" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492?np=y">Screen reader users, click here to load entire article</a>This page uses JavaScript to progressively load the article content as a user scrolls.  Screen reader users, click the load entire article button to bypass dynamically loaded article content.</div>



<script>
var refLinkURL = "http://www.sciencedirect.com/science/relLink/ajaxReq?_eid=1-s2.0-S0888754310000492";
var updatedCitedBy = "http://www.sciencedirect.com/science/referenceResolution/ajaxRefResol?_updateCitedBy=";
var refResolvePath = "http://www.sciencedirect.com/science/referenceResolution/ajaxRefResol?_pii=S0888754310000492&_refCnt=87&_docType=article";

// these variables handle the reference resolution. Please ignore the naming.
// the first variable myMap will hold an associative array of links by bibID
// the second variable myXabsCounts holds the scopus counts.
var myMap = []; // html of links for references.
var myXabsCounts = []; // scopus counts
</script>
<div id="header-area">
  


<!-- ************* For A/B testing FAST and SOLR Proxy ***************** -->

<!-- *************End for A/B testing FAST and SOLR Proxy *****************-->



<div id="ieWarningMsg">
    <span class="infoIcon_ie spErrorIcon">
        Please note that Internet Explorer version 8.x will not be supported as of January 1, 2016. Please refer to <a href="http://blog.sciencedirect.com/posts/have-you-heard-spread-the-word-about-upgrading-your-browser"> this blog post</a> for more information.
    </span>
    <a class="closeBt" title="Close Internet Explorer 7 warning">Close</a>
</div>



<div id="header">
<div class="headerBlk">
    <div class="headerInner" role="navigation" aria-label="ScienceDirect">
        <div class="home">
            <iframe style="display:none;" src="./Assembly algorithms for next-generation sequencing data_files/update.html"></iframe><iframe style="display:none;" src="./Assembly algorithms for next-generation sequencing data_files/update(1).html"></iframe><iframe style="display:none;" src="./Assembly algorithms for next-generation sequencing data_files/update(2).html"></iframe><iframe style="display:none;" src="./Assembly algorithms for next-generation sequencing data_files/update(3).html"></iframe>
            <!--INSERTS THE TEMPLATE VARIABLES POPULATED FROM DB for SCIVERSE and PRODUCT labels-->
            
                <a href="http://www.sciencedirect.com/" title="ScienceDirect -The world&#39;s leading full-text scientific database" class="padLeft"> <span class="logoScienceDirectImg">&nbsp;</span></a>
            
        </div>

        
            <ul class="publication">
                <!-- Journals link -->
                
                    
                        <li>
                            <a href="http://www.sciencedirect.com/science/journals">Journals</a>
                        </li>
                     
                

                <!-- Books link -->
                
                    
                        <li class="sep">
                            <a href="http://www.sciencedirect.com/science/bookbshsrw">Books</a>
                        </li>
                     
                
            </ul>
        

    </div>
    
         <!-- if not logged in -->
         
            
            <div id="login" class="clearfix" aria-label="Help and Sign In" role="navigation" style="margin-right:230px;">
            <ul class="userlinks">
                
                    <li>
                        <a href="http://www.sciencedirect.com/science?_ob=ShoppingCartURL&amp;_method=display&amp;_zone=TopNavBar&amp;_origin=article&amp;md5=98a3e89aa38c5044516e5e4ca8b64e74" target="_top" name="cart" title="Shopping Cart">Shopping cart</a>
                    </li>
                
                
                
            
                <!--top right widget-->
                <li id="loginPlus">
                    <h2><a id="loginPlusScript" href="javascript:LoginBox.flipLogin(&#39;loginBox&#39;,null);" title="Sign in to ScienceDirect" class="novisit plus nojs S_C_SignIn" aria-haspopup="true" aria-expanded="false" aria-controls="loginBox" role="button">Sign in</a></h2>
                    <noscript>&lt;a href="http://www.sciencedirect.com/science?_ob=HomePageURL&amp;_method=userHomePage&amp;_fl=y&amp;md5=6dada1fb726164470bad9e40bbc4d7fa" title="Open the Sign in Box" class="plus S_C_SignIn" id="loginPlusNoScript" &gt;Sign in&lt;/a&gt;</noscript>
					
                    <div id="loginBox" role="dialog" aria-labelledby="headingSignIn" class="bannerAvail" style="width:600px;display:none;"> 
                        
                        <div style="width:285px;float:right;border-left:1px solid #9B9B9B;padding-left:10px;min-height:180px;">
                            
                                <div class="openAthens"><a href="https://auth.elsevier.com/ShibAuth/institutionLogin?entityID=https%3A%2F%2Fidp.eduserv.org.uk%2Fopenathens&amp;appReturnURL=http%3A%2F%2Fwww.sciencedirect.com%2Fscience%2Farticle%2Fpii%2FS0888754310000492" title="Log in using OpenAthens">OpenAthens login</a></div>
                            
                            
                            <div class="sampletext">
                                <div style="font-weight:bold;padding-bottom:7px;"><h3><b>Login via your institution</b></h3></div>
                                
                                
                                    <div class="fedLink"><a href="http://www.sciencedirect.com/science/externallogin" title="Other institution login">Other institution login</a></div>
                                
                            </div>
                            
                            
                            
                                
                            
                            
                            
                        </div>
                        
                        <div class="padding" style="width:285px;">
                            <form name="sdLogin" method="POST" action="https://www.sciencedirect.com/science/login" id="login-box-form" style="margin:0;">
                            <b><h3 id="headingSignIn">Sign in using your ScienceDirect credentials</h3></b>
                            <input type="hidden" name="login" value="login">
                            <div style="margin-top:20px;">
                            <label for="username"><span style="text-align:right;">Username:</span>
                            <input id="username" type="text" name="userid" title="Please enter your username here"></label>
                            </div>
                            
                            <div style="margin-top:7px;">
                            <label for="password"><span style="text-align:right;">Password:</span>
                            <input id="password" type="password" name="password" maxlength="20" title="Please enter your password here" onkeypress="if(13==event.keyCode)this.form.submit();if (13==event.keyCode)return false;"></label>
                            </div>
                                   
                                       
                            
                            <div class="remember">
                            <input type="checkbox" checked="" name="rememberid" id="rememberMe" value="Y" class="allCheck top2fix" title="Checking this option will log you in automatically next time you visit ScienceDirect">
                            <label for="rememberMe" title="Checking this option will log you in automatically next time you visit ScienceDirect">Remember me</label>
                            </div>
                            
                            <input type="hidden" name="_ob" value="MiamiLogonURL">
                            <input type="hidden" name="targetURL" value="http://www.sciencedirect.com/science/article/pii/S0888754310000492">
                            <input type="hidden" name="returnURL" value="http://www.sciencedirect.com/science/article/pii/S0888754310000492">
                            <div class="submit">
                            
                            <div style="float:left;padding-left:0px;"><input class="button" type="Submit" name="arrow" alt="Sign in to ScienceDirect" title="Sign in to ScienceDirect" value="Sign in"></div>
                        
                            
                            
                            <div class="notRegistered"><span class="astPad">| </span><span><a href="https://www.sciencedirect.com/science/register-new-user" title="Register a username and password for using alerts and other personal features" target="_top">Not Registered?</a></span></div>
                            
                            
                            <div style="clear:left;"></div>
                            </div>
                            
                            <div class="forgotpass">
                            <a href="http://www.sciencedirect.com/science?_ob=ReminderURL&amp;_method=display&amp;md5=4a97631c25ac81c09ad2f961106e9da2" target="passwd" onclick="var remindWin;remindWin=window.open(&#39;&#39;,&#39;passwd&#39;,&#39;scrollbars=yes,resizable=yes,directories=no,toolbar=n,menubar=no,status=no,width=650,height=300&#39;);" title="Click here for help in recovering your ScienceDirect access">Forgotten username or password?</a>
                            </div>
                            </form>
                        </div>
                    </div>
                </li>
                
                <li class="last"><a class="S_C_helpLink" href="http://www.sciencedirect.com/science?_ob=HelpURL&amp;_file=browse_open_doc.htm&amp;_btn=Y&amp;_zone=TopNavBar&amp;_origin=article" target="sdhelp" title="Help (Opens new window)" onclick="var helpWin; helpWin=window.open(&#39;/science?_ob=HelpURL&amp;_file=browse_open_doc.htm&amp;_btn=Y&amp;_zone=TopNavBar&amp;_origin=article&#39;,&#39;sdhelp&#39;,&#39;scrollbars=yes,resizable=yes,directories=no,toolbar=no,menubar=no,status=no,width=984,height=738&#39;); helpWin.focus(); return false">Help</a></li>
                
            </ul>
            </div>
            
            
        <!-- logged in -->
             
    
</div>


    
                        
        <div class="clearfix resizedBanner" id="bannerTop" role="banner" aria-label="Institutional Access">                        
            
                <a href="http://www.elsevier.com/solutions/sciencedirect/who-uses/guest-users" onclick="openNS(&#39;http://www.elsevier.com/solutions/sciencedirect/who-uses/guest-users&#39;); return false;">
                    <img src="./Assembly algorithms for next-generation sequencing data_files/dynamic_scidir_home_guest_user.png" border="0">                    
                </a>
            
        </div>
        
    
           


</div><!--header-->









</div><!--header-area-->
<div id="page-area">
    
     
     
    <div id="articleToolbar" class="clearfix quicksearch prd_sci_dir addPadding low fixedPane" data-top="42">
        <div id="articleNav" style="margin-left: 6px;">
  <ul aria-label="Article options" role="navigation" class="clearfix">
    <!-- return to gateway -->
     






<li>
  <div class="icon_pdf toolbar">
    <div class="largeIcon"></div>
    <a id="pdfLink" rel="nofollow" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492/pdfft?md5=b1ae89a9b03441bc00073f886e25daf9&amp;pid=1-s2.0-S0888754310000492-main.pdf" querystr="?_origin=article&amp;_zone=toolbar" target="newPdfWin" pdfurl="http://www.sciencedirect.com/science/article/pii/S0888754310000492/pdfft?md5=b1ae89a9b03441bc00073f886e25daf9&amp;pid=1-s2.0-S0888754310000492-main.pdf" class="S_C_pdfLink big pdf ext_sdlink cLink" style="cursor:pointer" title="Download PDF">Download PDF
    <span id="pdfExplanation" class="offscreen">Opens in a new window.  Article suggestions will be shown in a dialog on return to ScienceDirect.</span>
    </a>
  </div>
</li>
 
 







<li class="exportPopupListitem">
<div class="pulldown">
    <div class="exportDivision" title="Export citations" aria-expanded="false">
    <div class="exportIconText">
    <div class="icon_exportarticlesci_dir initialImage export_icon" tabindex="0"> &nbsp; </div>
    <input id="icon_exportarticle" value="Export" name="export" class="listAction export export_general" tabindex="-1" type="button">
     <span class="downSM_sci_dir exportArrow"></span>
    </div>
    <input class="zone" name="zone" type="hidden" value="exportDropDown">
    </div>
    <div class="menu export_popup hidden" id="export_popup">
    <div class="warningIcon" title="Warning Icon" alt="Warning Icon" style="display: none;"></div>
    <form name="exportCite" method="post" action="http://www.sciencedirect.com/science?_ob=DownloadURL&amp;_method=finish&amp;_eidkey=1-s2.0-S0888754310000492&amp;count=1&amp;_docType=FLA&amp;md5=7525f31dc71a13477d2ec99e566797dd" target="">
    <input class="zone" name="zone" type="hidden" value="exportDropDown">
            <script language="javascript" type="text/javascript">
<!--

function checkExportFormat(limit) {

    var msg = "";

    for (i=0;i<document.exportCite.elements.length;i++) {
        if (document.exportCite.elements[i].value == "REFWX") {
            if (document.exportCite.elements[i].checked) {
                document.exportCite.setAttribute('target',"RefWorksMain");
                if (checkLimit()) {
                    if (limit) {
                        msg = "It is possible that some users may experience problems when downloading large numbers of citations. If this happens please contact RefWorks support at your regionally hosted site or e-mail support@refworks.com to assist you further.\n\n";
                    }
                    msg += "Exporting to RefWorks may take some time, please wait....";
                    alert(msg);
                }
            }
            else {
                document.exportCite.setAttribute('target',"");
            }
            break;
        }
    }
    return true;
}

function determineCount(numStr) {
    var count=1;

    if(numStr.indexOf("-")) {
        var rangeStr = numStr.split("-");

        if(rangeStr.length == 2) {
            var highNum = parseInt(rangeStr[1]);
            var lowNum = parseInt(rangeStr[0]);

            if(highNum >= lowNum) { count = (highNum - lowNum) + 1; }
        }
    }
    return count;
}

function checkLimit(){

    var selectNum = "";
    var useSelected = false;
    var allNum = 0;
    var count = 0;
    var rdoc = 0;

    for (i=0;i<document.exportCite.elements.length;i++) {
        if (document.exportCite.elements[i].name == "count") {
            allNum = document.exportCite.elements[i].value;
        }
        if (document.exportCite.elements[i].name == "NUMBER_LIST") {
            selectNum = document.exportCite.elements[i].value;
        }
        if (document.exportCite.elements[i].name == "_rdoc") {
            rdoc = document.exportCite.elements[i].value;
        }
        if (document.exportCite.elements[i].value == "selected") {
            if (document.exportCite.elements[i].checked) {
                useSelected = true;
            }
        }
    }

    if (useSelected) {
        selectNum = selectNum.replace(/ /g,"");
        if(selectNum.indexOf(",")) {
            var commaList = selectNum.split(",");
            for(i=0; i < commaList.length; i++) {
                count += determineCount(commaList[i]);
            }
        }
        else {
            count += determineCount(selectNum);
        }
    }
    else { 
        if (rdoc > 0) {
            count = 1;
        }
        else {
            count = allNum;
        }
    }

    if(count > 50) { return true; }
  
    return false;
}

//-->
</script>

    <div class="exportInfo">
        <div class="chkboxCount"></div>
        <div class="exportHelp">
            <a target="sdhelp" href="http://www.sciencedirect.com/science?_ob=HelpURL&amp;_file=art_export.htm&amp;_btn=Y&amp;_zone=TopNavBar&amp;_origin=article" title="Help (Opens new window)" onclick="var helpWin; helpWin=window.open(&#39;/science?_ob=HelpURL&amp;_file=art_export.htm&amp;_btn=Y&amp;_zone=TopNavBar&amp;_origin=article&#39;,&#39;sdhelp&#39;,&#39;scrollbars=yes,resizable=yes,directories=no,toolbar=no,menubar=no,status=no,width=984,height=738&#39;); helpWin.focus(); return false">Help</a>
        </div>
    </div>

    <div class="expContainer">
      <ul class="Export_Popup" width="100%">
        <li class="subHeading"><h3> Direct export </h3></li>
        <li class="direct_export">
         
            <div>
              <div class="btn_export directExport">
                <div class="icon_Mendeley" title="Save to Mendeley">&nbsp;</div>
                <input id="directExport" type="button" value="Save to Mendeley" name="Save to Mendeley" alt="Save to Mendeley" class="mendeleyBT button">
              </div>
            </div>
        
        
            <div class="btnLine">
              <div class="btn_export refworks">
                <div class="icon_Refworks" title="Save to RefWorks">&nbsp;</div>
                <input id="REFWX" type="button" value="Save to RefWorks" name="Save to RefWorks" alt="Save to RefWorks" class="button refworks">
              </div> 
            </div>
            <div class="clear"> </div>
        
        </li>
        <li class="subHeading big"> <h3>Export file</h3></li>
        <li class="export_radio RIS"><input type="radio" id="RIS" checked="" name="citation-type" value="RIS" class="artRadio expRadio RIS">&nbsp;<label for="RIS">RIS&nbsp;(for EndNote, Reference Manager, ProCite)</label>
        </li>
        <li class="export_radio"><input type="radio" id="BIBTEX" name="citation-type" value="BIBTEX" class="artRadio expRadio BIBTEX">&nbsp;<label for="BIBTEX">BibTeX</label>
        </li>
        <li class="export_radio"><input type="radio" id="ASCII" name="citation-type" value="ASCII" class="artRadio expRadio ASCII">&nbsp;<label for="ASCII">Text</label>
        </li>
        <li class="export_radio hidden"><input type="radio" id="REFWX" name="citation-type" value="REFWX" class="artRadio expRadio REFWX">&nbsp;<label for="REFWX">RefWorks Direct Export</label>
        </li>
        <li class="subHeading big"><h3>Content</h3></li>
        <li class="expCitab export_radio"><input type="radio" id="cite" class="cite artRadio expRadio" name="format" value="cite">&nbsp;<label class="cit" for="cite"> Citation Only </label>
        </li>
        <li class="expCitab export_radio citab"><input type="radio" id="cite-abs" class="cite-abs artRadio expRadio" name="format" value="cite-abs" checked="">&nbsp;<label class="cab" for="cite-abs"> Citation and Abstract </label>
        </li>
        <li class="export_submit">
            <div class="downBtn export_downBtn">
               <input type="submit" value="Export" name="export" alt="Export" class="button export_button" id="export_button">
            </div> 
        </li>
      </ul>

    </div>


        
        
            
         

    </form>
    </div> 
</div>    
</li>
  



</ul>
</div>

<div class="sdSearch" aria-label="Quick search" role="search">
    <fieldset>
        <div class="left">
            <form id="quickSearch" name="qkSrch" method="get" target="_top" action="http://www.sciencedirect.com/science">
            <input type="hidden" name="_ob" value="QuickSearchURL">
            <input type="hidden" name="_method" value="submitForm">
            <input type="hidden" name="_acct" value="C000228598">
            <input type="hidden" name="searchtype" value="a">
            <input type="hidden" name="_origin" value="article">
            <input type="hidden" name="_zone" value="qSearch">
            <input type="hidden" name="md5" value="89d09e911d3872e16ae140ceae8c0b3b">
            
            <input id="quickSearch" type="text" name="qs_all" placeholder="Search ScienceDirect" value="" size="30" maxlength="450" title="Perform a quick search on ScienceDirect"><input type="submit" class="submit" value="" title="Submit quick search">
            
            <a class="advanced S_C_advanced" href="http://www.sciencedirect.com/science/search">Advanced search</a>
            </form>
        </div>
    </fieldset>
</div>
<div class="clear"></div>
<div id="erToolbar"></div>


<div style="color : #FF0000; font-weight : bold;border:0px solid red;">
  
  
  
  
</div>





        <!-- PDFKING/PDFCITE -->
        <div id="pdfModalWindow" class="pdfModalWindow" style="display:none;" onclick="pdfCite.closePopup();"></div>
        <div id="suggestedPdfList" style="display:none;">
        
        </div>
        <!-- PDFKING/PDFCITE End -->
    </div>
    <div id="articleToolbarDummy" style="display: block; height: 62px;"></div>
  <div style="clear:both;"></div>
  
  <div id="leftPane" class="column small" aria-label="table of contents" role="navigation" style="height: 107px;">
    <div id="leftPaneInner" class="fixedPane" style="height: 149px; width: 28px; top: 101px; left: -22.8571px;">
        <div>
        <div id="leftButton" class="small">
            <ul>
                <li class="tocbutton">
                    <a class="outline-toggle" title="Open to view article outline" alt="Open to view article outline" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#" aria-expanded="false">
                        <span class="icon outline_big"></span>
                    </a>
                </li>
            </ul>
        </div>
        <div id="outlineContainer" class="small" style="top: 0px;">
            <a class="closeBt" title="Close">&nbsp;</a>
            <div id="outline" class="outline" data-st="0" style="overflow: hidden;"><div id="olGraphCbBox" style="width: 231px;"><div class="olHead"><h2 id="toc_title">Article outline</h2><label style="display: block;"><input id="outlineGraphicsCheckBox" type="checkBox">Show full outline</label></div><ul id="itemOutline"><li class="li_h2"><div id="ol_author1" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#author1" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">Abstract<br></a></div></li><li class="li_h2"><div id="ol_kwd_1" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#kwd_1" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">Keywords<br></a></div></li><li class="li_h2"><div id="ol_section_lbl1" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl1" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">Introduction<br></a></div></li><li class="li_h3"><div id="ol_section_lbl1.1" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl1.1" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">Next-generation Sequence Data<br></a></div></li><li class="li_h3"><div id="ol_section_lbl1.2" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl1.2" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">What is an Assembly?<br></a></div></li><li class="li_h3"><div id="ol_section_lbl1.3" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl1.3" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">The Challenge of Assembly<br></a></div></li><li class="li_h3"><div id="ol_section_lbl1.4" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl1.4" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">Metagenomics<br></a></div></li><li class="li_h2"><div id="ol_section_lbl2" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl2" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">Graph Algorithms for Assembly<br></a></div></li><li class="li_h2"><div id="ol_section_lbl3" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl3" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">Greedy Graph-based Assemblers<br></a></div></li><li class="li_h2"><div id="ol_section_lbl4" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl4" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">Overlap/Layout/Consensus Assemblers<br></a></div></li><li class="li_h2"><div id="ol_section_lbl5" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl5" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">The de Bruijn Graph Approach<br></a></div></li><li class="li_h3"><div id="ol_section_lbl5.1" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl5.1" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">The de Bruijn Graph in Euler<br></a></div></li><li class="li_h3"><div id="ol_section_lbl5.2" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl5.2" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">The de Bruijn Graph in Velvet<br></a></div></li><li class="li_h3"><div id="ol_section_lbl5.3" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl5.3" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">The de Bruijn Graph in ABySS<br></a></div></li><li class="li_h3"><div id="ol_section_lbl5.4" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl5.4" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">The de Bruijn Graph in AllPaths<br></a></div></li><li class="li_h3"><div id="ol_section_lbl5.5" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl5.5" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">The de Bruijn Graph in SOAPdenovo<br></a></div></li><li class="li_h2"><div id="ol_section_lbl6" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl6" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">Other Software<br></a></div></li><li class="li_h2"><div id="ol_section_lbl7" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#section_lbl7" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">Summary and Outlook<br></a></div></li><li class="li_h2"><div id="ol_ack001" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#ack001" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">Acknowledgments<br></a></div></li><li class="li_h2"><div id="ol_bibl001" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bibl001" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">References<br></a></div></li><li class="li_h2"><div id="ol_m4vt1" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#m4vt1" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;);">Vitae<br></a></div></li></ul><script>prs.rt('scr_toc')</script></div>
              <script type="text/javascript" style="width: 243px;">
              document.write("<div class='outlineMsg'>Article outline is loading...</div>");
              </script>
              <noscript style="width: 243px;">
              &lt;div class="infoMsg"&gt;
              &lt;div style="float: left; width: 20px; "&gt;&lt;img border="0" src="http://cdn.els-cdn.com/sd/alertIcon.gif" valign="top" alt="Alert message"&gt;&lt;/div&gt;
              &lt;div style="padding-left:25px;"&gt;JavaScript is disabled on your browser. &lt;br&gt;&lt;div style="padding-top:10px;"&gt;
              Please enable JavaScript to use all the features on this page.&lt;/div&gt;&lt;/div&gt;
              &lt;/div&gt;
              </noscript>
            <div id="fig" role="navigation" aria-labelledby="fig_title" style="width: 231px;"><hr><h2 id="fig_title">Figures and tables</h2><ol class="figures " role="tree"><li><div id="ol_fig1" class="io item fig" data-parent="section_lbl2"><a class="tocLink S_C_tocLinkFig" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig1" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;,&#39;article&#39;,&#39;leftPane&#39;, &#39;serial&#39;);"><img src="./Assembly algorithms for next-generation sequencing data_files/1-s2.0-S0888754310000492-gr1.sml" class="displayImg" title="A read represented by K-mer graphs. (a) The read is represented by two types of ..." alt="A read represented by K-mer graphs. (a) The read is represented by two types of ..." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0888754310000492-gr1.sml" data-thumbheight="118" data-loaded="true"></a></div></li><li><div id="ol_fig2" class="io item fig" data-parent="section_lbl2"><a class="tocLink S_C_tocLinkFig" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig2" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;,&#39;article&#39;,&#39;leftPane&#39;, &#39;serial&#39;);"><img src="./Assembly algorithms for next-generation sequencing data_files/1-s2.0-S0888754310000492-gr2.sml" class="displayImg" title="A pair-wise overlap represented by a K-mer graph. (a) Two reads have an ..." alt="A pair-wise overlap represented by a K-mer graph. (a) Two reads have an ..." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0888754310000492-gr2.sml" data-thumbheight="118" data-loaded="true"></a></div></li><li><div id="ol_fig3" class="io item fig" data-parent="section_lbl2"><a class="tocLink S_C_tocLinkFig" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig3" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;,&#39;article&#39;,&#39;leftPane&#39;, &#39;serial&#39;);"><img src="./Assembly algorithms for next-generation sequencing data_files/grey_pxl.gif" class="smlImgLazyJSB greyImg" title="Complexity in K-mer graphs can be diagnosed with read multiplicity information. ..." alt="Complexity in K-mer graphs can be diagnosed with read multiplicity information. ..." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0888754310000492-gr3.sml" data-thumbheight="164"></a></div></li><li><div id="ol_fig4" class="io item fig" data-parent="section_lbl5.1"><a class="tocLink S_C_tocLinkFig" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig4" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;,&#39;article&#39;,&#39;leftPane&#39;, &#39;serial&#39;);"><img src="./Assembly algorithms for next-generation sequencing data_files/grey_pxl.gif" class="smlImgLazyJSB greyImg" title="Three methods to resolve graph complexity. (a) Read threading joins paths across ..." alt="Three methods to resolve graph complexity. (a) Read threading joins paths across ..." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0888754310000492-gr4.sml" data-thumbheight="134"></a></div></li><li><div id="ol_tbl1" class="io item tbl" data-parent="section_lbl7"><a class="tocLink olIcon olIconTbl S_C_tocLinkTbl" title="Feature comparison between de novo assemblers for whole-genome shotgun data from ..." alt="Feature comparison between de novo assemblers for whole-genome shotgun data from ..." href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#tbl1" onclick="SD_UTIL.sendUserKeyEvent(&#39;displayTOCSection&#39;, &#39;article&#39;, &#39;leftPane&#39;, &#39;serial&#39;); "><div class="olIcon olIconTbl"></div>Table&nbsp;1</a></div></li></ol></div><div id="articleLeftAd" class="adBlock" style="width: 243px; height: 669px;"><hr style="margin-left:12px;"><div class="adText">ADVERTISEMENT<div class="closeBt"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#" title="Close Advertisement">&nbsp;</a></div></div><iframe scrolling="no" frameborder="0" border="0" cellspacing="0" src="./Assembly algorithms for next-generation sequencing data_files/Y.html" style="height: 604px;"></iframe></div></div>
        </div>
        </div>
    </div>
  </div>
  <div id="centerPane" class="column" tabindex="-1">
    <div class="topBorder fixedPane" style="top: 101px; left: 5.14286px;"></div>
    <div id="centerContent" class="centerContent">
    <div id="centerInner" class="centerInner svBigBox" aria-label="Article" role="main">
        
      
      <noscript>
      &lt;div class="msgBox" style="margin-top:10px;"&gt;
          &lt;span class="errMsg"&gt;
            &lt;div&gt;JavaScript is disabled on your browser.  Please enable JavaScript to use all the features on this page. This page uses JavaScript to progressively load the article content as a user scrolls. Click the View full text link to bypass dynamically loaded article content. &lt;a rel="nofollow" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492?np=y"&gt;View full text&lt;/a&gt;&lt;/div&gt;
         &lt;/span&gt;
      &lt;/div&gt;&lt;br/&gt;
      </noscript>
      <div class="head headTemplate page_fragment_ind">

  
<div class="publisherLogo">
  <span class="spElsevierPubIcon"></span>
</div>


  
    <!--?xml version="1.0" encoding="UTF-8"?--><div xmlns:bk="http://www.elsevier.com/xml/bk/dtd" class="publicationHead"><div class="title"><a title="Go to Genomics on ScienceDirect" href="http://www.sciencedirect.com/science/journal/08887543" querystr="?zone=centerPane" class="cLink"><span>Genomics</span></a></div><p class="volIssue"><a class="S_C_volIss" title="Go to table of contents for this volume/issue" href="http://www.sciencedirect.com/science/journal/08887543/95/6">Volume 95, Issue 6</a>, June 2010, Pages 315–327</p></div>
  
    <div class="publicationCover cover2">
    
      
    <a querystr="?zone=publicationLogo" class="cLink" href="http://www.sciencedirect.com/science/journal/08887543/95/6">
      
    
    <img alt="Cover image" src="./Assembly algorithms for next-generation sequencing data_files/1-s2.0-S0888754310X00064-cov150h.gif" class="toprightlogo">
    <noscript>&lt;img alt="Cover image" src="http://ars.els-cdn.com/content/image/1-s2.0-S0888754310X00064-cov150h.gif" class="toprightlogo" /&gt;</noscript>
    </a>
    
  </div>


</div>

      <div id="frag_1" class="page_fragment auth_frag" data-first="true" data-fid="1"><div id="" class="publicationType"><span id="">Review</span></div><div class="module_topic_paths"></div><h1 class="svTitle" id="">Assembly algorithms for next-generation sequencing data</h1><ul class="authorGroup noCollab svAuthor" style="display: block;"><li class="smh5"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#" class="authorName svAuthor" data-orcid="" id="authname_N4447e770N67a4c508" data-t="a" data-fn="Jason R." data-ln="Miller" data-pos="1" data-tb="">Jason R. Miller</a><a title="Corresponding author contact information" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#cor1" id="bcor1" class="intra_ref auth_corr"></a><sup>, </sup><a href="mailto:jmiller@jcvi.org" class="auth_mail" title="E-mail the corresponding author"></a><span><a class="authorVitaeLink" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#vt2">Author Vitae</a></span>, </li><li class="smh5"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#" class="authorName svAuthor" data-orcid="" id="authname_N4447e770N67a4c5f8" data-t="a" data-fn="Sergey" data-ln="Koren" data-pos="2" data-tb="">Sergey Koren</a><span><a class="authorVitaeLink" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#vt1">Author Vitae</a></span>, </li><li class="smh5"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#" class="authorName svAuthor" data-orcid="" id="authname_N4447e770N67a4c634" data-t="a" data-fn="Granger" data-ln="Sutton" data-pos="3" data-tb="">Granger Sutton</a><span><a class="authorVitaeLink" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#vt3">Author Vitae</a></span></li></ul><!--VALIDHTML--><ul class="affiliation authAffil smh" style="display: none;"><li class="smh5plus" style="display: none;"> <span id="">J. Craig Venter Institute, 9704 Medical Center Drive, Rockville MD 20850-3343, USA</span></li></ul><!--VALIDHTML--><!--VALIDHTML--><div id="currtAsOf"></div><dl class="articleDates smh" style="display: none;"><dd>Received 4 August 2009, Accepted 2 March 2010, Available online 6 March 2010</dd></dl><!--VALIDHTML--><div id="showMoreButtons"><a class="showLess moreLink"><span class="CollapseText">&nbsp;</span><span class="showInfo expand" title="Show more author and article information">Show more</span></a><a class="showLess lessLink"><span class="ExpandText" style="display: none;">&nbsp;</span><span class="showInfo collapse" title="Hide more author and article information">Show less</span></a></div><!--VALIDHTML--><dl class="extLinks nonEmpty"><dd class="doiLink"><dd class="doi"><a id="ddDoi" class="S_C_ddDoi" href="http://dx.doi.org/10.1016/j.ygeno.2010.03.001" target="doilink" onclick="var doiWin; doiWin=window.open(&#39;http://dx.doi.org/10.1016/j.ygeno.2010.03.001&#39;,&#39;doilink&#39;,&#39;scrollbars=yes,resizable=yes,directories=yes,toolbar=yes,menubar=yes,status=yes&#39;); doiWin.focus()">doi:10.1016/j.ygeno.2010.03.001</a></dd></dd><dd class="rightsLink"><a class=" cExLink S_C_outLinks" href="https://s100.copyright.com/AppDispatchServlet?publisherName=ELS&amp;contentID=S0888754310000492&amp;orderBeanReset=true" data-url="/science/RedirectURL?_method=outwardLink&amp;_partnerName=936&amp;_eid=1-s2.0-S0888754310000492&amp;_pii=S0888754310000492&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=https://s100.copyright.com/AppDispatchServlet?publisherName%3DELS%26contentID%3DS0888754310000492%26orderBeanReset%3Dtrue" target="_blank" title="Get rights and content"><span class="outText">Get rights and content</span></a></dd></dl><div class="moreInformation"><dl class="citation"><dd class="oaUserLicense wrap2">Under an Elsevier <span><a id="ddOAuserLicense" href="http://www.elsevier.com/open-access/userlicense/1.0/" target="_blank">user license</a></span></dd></dl></div><div class="articleOAlabelForced"></div><div id="refersToAndreferredToBy"><dl id="referredToBy" class="documentThread"><!--Referred To By--></dl></div><!--FRAGMENTEND--><div class="page_fragment_ind auth_frag" data-id="frag_2"></div></div><div id="frag_2" class="page_fragment" data-fid="2"><script>prs.rt("abs_1st_beg");</script><!--VALIDHTML--><div class="OAlabel"><span class="articleOAstatus">&nbsp;&nbsp;Open Archive</span><hr id="abs_author1" class="artHeader"></div><div class="abstract svAbstract " data-etype="ab"><h2 class="secHeading" id="author1">Abstract</h2><p id="">The emergence of next-generation sequencing platforms led to resurgence of research in whole-genome shotgun assembly algorithms and software. DNA sequencing data from the Roche 454, Illumina/Solexa, and ABI SOLiD platforms typically present shorter read lengths, higher coverage, and different error profiles compared with Sanger sequencing data. Since 2005, several assembly software packages have been created or revised specifically for de novo assembly of next-generation sequencing data. This review summarizes and compares the published descriptions of packages named SSAKE, SHARCGS, VCAKE, Newbler, Celera Assembler, Euler, Velvet, ABySS, AllPaths, and SOAPdenovo. More generally, it compares the two standard methods known as the de Bruijn graph approach and the overlap/layout/consensus approach to assembly.</p></div><!--VALIDHTML--><h2 id="kwd_1" class="svKeywords">Keywords</h2><ul class="keyword" id=""><li id="" class="svKeywords"><span id="">Genome assembly algorithms</span>; </li><li id="" class="svKeywords"><span id="">Next-generation sequencing</span></li></ul><!--VALIDHTML--><div id="embeddedPDF" style="display:none"></div><script>prs.rt("abs_end");</script><div id="SD_BA1P" class="sgfNoTitleBar sgfNoGadgetBorder svDoNotLink ui-sortable"></div><!--VALIDHTML--><hr class="artHeader"><h2 class="svArticle" id="section_lbl1">Introduction</h2><p class="svArticle section clear" id="">The advent of short-read sequencing machines gave rise to a new generation of assembly algorithms and software. This survey reviews algorithms for <em>de novo</em> whole-genome shotgun assembly from next-generation sequencing data. It describes and compares algorithms that have been presented in the scientific literature and implemented in software. We use a narrow definition of <em>de novo</em> whole-genome shotgun assembly. The shotgun process takes reads from random positions along a target molecule  <span id="bbib1"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib1" id="ancbbib1" class="intra_ref">[1]</a></span>. Whole-genome shotgun (WGS) sequencing samples the chromosomes that make up one genome. WGS assembly is the reconstruction of sequence up to chromosome length. The assembly task is relegated to computer software <span id="bbib2"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib2" id="ancbbib2" class="intra_ref">[2]</a></span>. Assembly is possible when the target is over-sampled by the shotgun reads, such that reads overlap. <em>De novo</em> WGS assembly refers to reconstruction in its pure form, without consultation to previously resolved sequence including from genomes, transcripts, and proteins. <em>De novo</em> WGS assembly of next-generation sequencing (NGS) data is a specialized problem due to the short read lengths and large volumes of NGS data. Benchmarking the implementations is beyond the scope of this review. Broader introductions can be found elsewhere, <em>e.g.</em> <span id="bbib3"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib3" id="ancbbib3" class="intra_ref">[3]</a></span>.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl1.1">Next-generation Sequence Data</h3><p class="svArticle section clear" id="">Today's commercial DNA sequencing platforms include the Genome Sequencer from Roche 454 Life Sciences (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.454.com"><a href="http://www.454.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.454.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.454.com</a></span>), the Solexa Genome Analyzer from Illumina (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.illumina.com"><a href="http://www.illumina.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.illumina.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.illumina.com</a></span>), the SOLiD System from Applied Biosystems (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.appliedbiosystems.com"><a href="http://www.appliedbiosystems.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.appliedbiosystems.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.appliedbiosystems.com</a></span>), the Heliscope from Helicos (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.helicos.com"><a href="http://www.helicos.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.helicos.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.helicos.com</a></span>), and the commercialized Polonator (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.polonator.org"><a href="http://www.polonator.org/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.polonator.org" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.polonator.org</a></span>). These platforms have been well reviewed, e.g. <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib4" id="bbib4" class="intra_ref">[4]</a>, <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib5" id="bbib5" class="intra_ref">[5]</a>, <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib6" id="bbib6" class="intra_ref">[6]</a>&nbsp;and&nbsp;<a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib7" id="bbib7" class="intra_ref">[7]</a>. A distinguishing characteristic of these platforms is that they do not rely on Sanger chemistry <span id="bbib8"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib8" id="ancbbib8" class="intra_ref">[8]</a></span> as did first-generation machines including the Applied Biosystems Prism 3730 and the Molecular Dynamics MegaBACE. The second-generation machines are characterized by highly parallel operation, higher yield, simpler operation, much lower cost per read, and (unfortunately) shorter reads. Today's machines are commonly referred to as short-read sequencers or next-generation sequencers (NGS) though their successors may be on the horizon. Pacific Biosciences machines <span id="bbib9"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib9" id="ancbbib9" class="intra_ref">[9]</a></span> might produce reads longer than first-generation machines. First-generation reads were commonly 500&nbsp;bp to 1000&nbsp;bp long. Today's NGS reads are in the 400&nbsp;bp range (from 454 machines), the 100&nbsp;bp range (from the Solexa and SOLID machines), or less. Shorter reads deliver less information per read, confounding the computational problem of assembling chromosome-size sequences. Assembly of shorter reads requires higher coverage, in part to satisfy minimum detectable overlap criteria. High coverage increases complexity and intensifies computational issues related to large data sets.</p><!--VALIDHTML--><p class="svArticle section clear" id="">All sequencers produce observations of the target DNA molecule in the form of reads: sequences of single-letter base calls plus a numeric quality value (QV) for each base call <span id="bbib10"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib10" id="ancbbib10" class="intra_ref">[10]</a></span>. Although QVs offer extra information, their use generally increases a program's CPU and RAM requirements. Only some of the NGS assembly software exploits QVs.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The NGS platforms have characteristic error profiles that change as the technologies improve. Error profiles can include enrichment of base call error toward the 3′ (terminal) ends of reads, compositional bias for or against high-GC sequence, and inaccurate determination of simple sequence repeats. There are published error profiles for the 454 GS 20 <span id="bbib11"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib11" id="ancbbib11" class="intra_ref">[11]</a></span>, the Illumina 1G Analyzer <span id="bbib12"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib12" id="ancbbib12" class="intra_ref">[12]</a></span>, and comparisons of three platforms <span id="bbib13"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib13" id="ancbbib13" class="intra_ref">[13]</a></span>. Some NGS software is tuned for platform-specific error profiles. Others may have unintentional bias where development targeted one data type.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Sanger platforms could deliver paired-end reads, that is, pairs of reads with a constraint on their relative orientation and separation in the target. Paired ends were essential to assembly of cellular genomes small <span id="bbib14"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib14" id="ancbbib14" class="intra_ref">[14]</a></span> and large <span id="bbib15"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib15" id="ancbbib15" class="intra_ref">[15]</a></span> due to their ability to span repeats longer than individual reads. Paired ends, also called mate pairs, have a separation estimate that is usually provided to software as the fragment size distribution measured on a so-called library of reads. A sufficient variety of paired end separations should help resolve large chromosomes to single scaffolds <span id="bbib16"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib16" id="ancbbib16" class="intra_ref">[16]</a></span>. Early NGS sequencers offered unpaired reads but late models support paired-end protocols. Early NGS assembly software targeted unpaired reads but later programs exploit paired ends as read placement constraints.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl1.2">What is an Assembly?</h3><p class="svArticle section clear" id="">An assembly is a hierarchical data structure that maps the sequence data to a putative reconstruction of the target. It groups reads into contigs and contigs into scaffolds. Contigs provide a multiple sequence alignment of reads plus the consensus sequence. The scaffolds, sometimes called supercontigs or metacontigs, define the contig order and orientation and the sizes of the gaps between contigs. Scaffold topology may be a simple path or a network. Most assemblers output, in addition, a set of unassembled or partially assembled reads. The most widely accepted data file format for an assembly is FASTA, wherein contig consensus sequence can be represented by strings of the characters A, C, G, T, plus possibly other characters with special meaning. Dashes, for instance, can represent extra bases omitted from the consensus but present in a minority of the underlying reads. Scaffold consensus sequence may have N's in the gaps between contigs. The number of consecutive N's may indicate the gap length estimate based on spanning paired ends.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Assemblies are measured by the size and accuracy of their contigs and scaffolds. Assembly size is usually given by statistics including maximum length, average length, combined total length, and N50. The contig N50 is the length of the smallest contig in the set that contains the fewest (largest) contigs whose combined length represents at least 50% of the assembly. The N50 statistics for different assemblies are not comparable unless each is calculated using the same combined length value. Assembly accuracy is difficult to measure. Some inherent measure of accuracy is provided by the degrees of mate-constraint satisfaction and violation <span id="bbib17"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib17" id="ancbbib17" class="intra_ref">[17]</a></span>. Alignment to reference sequences is useful whenever trusted references exist.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl1.3">The Challenge of Assembly</h3><p class="svArticle section clear" id="">DNA sequencing technologies share the fundamental limitation that read lengths are much shorter than even the smallest genomes. WGS overcomes this limitation by over-sampling the target genome with short reads from random positions. Assembly software reconstructs the target sequence.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Assembly software is challenged by repeat sequences in the target. Genomic regions that share perfect repeats can be indistinguishable, especially if the repeats are longer than the reads. For repeats that are inexact, high-stringency alignment can separate the repeat copies. Careful repeat separation involves correlating reads by patterns in the different base calls they may have <span id="bbib18"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib18" id="ancbbib18" class="intra_ref">[18]</a></span>. Repeat separation is assisted by high coverage but confounded by high sequencing error. For repeats whose fidelity exceeds that of the reads, repeat resolution depends on “spanners,” that is, single reads that span a repeat instance with sufficient unique sequence on either side of the repeat. Repeats longer than the reads can be resolved by spanning paired ends, but the analysis is more complicated. Complete resolution usually requires two resources: pairs that straddle the repeat with each end in unique sequence, and pairs with exactly one end in the repeat. The limit of repeat resolution can be explored for finished genomes under some strict assumptions. For instance, it was shown that the theoretical best assembly of the <em>E. coli</em> genome from 20&nbsp;bp unpaired reads would put 10% of bases in contigs of 10&nbsp;Kbp or longer given infinite coverage and error-free reads <span id="bbib19"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib19" id="ancbbib19" class="intra_ref">[19]</a></span>. The limit calculation is not straightforward for reads with sequencing error, paired-end reads, or unfinished genomes. Careful estimates of repeat resolution involve the ratio of read length (or paired-end separation) to repeat length, repeat fidelity, read accuracy, and read coverage. In regard to NGS data, shorter reads have less power to resolve genomic repeats but higher coverage increases the chance of spanning short repeats.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Repeat resolution is made more difficult by sequencing error. Software must tolerate imperfect sequence alignments to avoid missing true joins. Error tolerance leads to false positive joins. This is a problem especially with reads from inexact (polymorphic) repeats. False-positive joins can induce chimeric assemblies. In practice, tolerance for sequencing error makes it difficult to resolve a wide range of genomic phenomena: polymorphic repeats, polymorphic differences between non-clonal asexual individuals, polymorphic differences between non-inbred sexual individuals, and polymorphic haplotypes from one non-inbred individual. If the sequencing platforms ever generate error-free reads at high coverage, assembly software might be able to operate at 100% stringency.</p><!--VALIDHTML--><p class="svArticle section clear" id="">WGS assembly is confounded by non-uniform coverage of the target. Coverage variation is introduced by chance, by variation in cellular copy number between source DNA molecules, and by compositional bias of sequencing technologies. Very low coverage induces gaps in assemblies. Coverage variability invalidates coverage-based statistical tests, and undermines coverage-based diagnostics designed to detect over-collapsed repeats.</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_3"></div><script>prs.rt("abs_1st_end");</script></div><div id="frag_3" class="page_fragment" data-fid="3"><p class="svArticle section clear" id="">WGS assembly is complicated by the computational complexity of processing larger volumes of data. For efficiency, all assembly software relies to some extent on the notion of a K-mer. This is a sequence of K base calls, where K is any positive integer. In most implementations, only consecutive bases are used. Intuitively, reads with high sequence similarity must share K-mers in their overlapping regions, and shared K-mers are generally easier to find than overlaps. Fast detection of shared K-mer content vastly reduces the computational cost of assembly, especially compared to all-against-all pair-wise sequence alignment. A tradeoff of K-mer based algorithms is lower sensitivity, thus missing some true overlaps. The probability that a true overlap spans shared K-mers depends on the value of K, the length of the overlap, and the rate of error in the reads. An appropriate value of K should be large enough that most false overlaps don't share K-mers by chance, and small enough that most true overlaps do share K-mers. The choice should be robust to variation in read coverage and accuracy.</p><!--VALIDHTML--><p class="svArticle section clear" id="">WGS assembly algorithms, and their implementations, are typically complex. Assembly operation can require high-performance computing platforms for large genomes. Algorithmic success can depend on pragmatic engineering and heuristics, that is, empirically derived rules of thumb. Heuristics help overcome convoluted repeat patterns in real genomes, random and systematic error in real data, and the physical limitations of real computers.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl1.4">Metagenomics</h3><p class="svArticle section clear" id="">Metagenomics is the sequencing of DNA in an environmental sample. Whereas WGS targets one genome, metagenomics usually targets several. The metagenomics assembly problem is confounded by genomic diversity and variable abundance within populations. Assembly reconstructs the most abundant sequences <span id="bbib20"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib20" id="ancbbib20" class="intra_ref">[20]</a></span>. Simulations indicate high rates of chimera, especially in short contigs assembled from complex mixtures <span id="bbib21"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib21" id="ancbbib21" class="intra_ref">[21]</a></span>. Studies that rely on characterization of individual reads prefer long reads <span id="bbib22"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib22" id="ancbbib22" class="intra_ref">[22]</a></span>. The role for <em>de novo</em> genomic assembly from NGS metagenomics data should grow as NGS read lengths and NGS paired-end options increase.</p><!--VALIDHTML--><h2 class="svArticle" id="section_lbl2">Graph Algorithms for Assembly</h2><p class="svArticle section clear" id="">We organize the NGS assemblers into three categories, all based on graphs. The Overlap/Layout/Consensus (OLC) methods rely on an overlap graph. The de Bruijn Graph (DBG) methods use some form of K-mer graph. The greedy graph algorithms may use OLC or DBG.</p><!--VALIDHTML--><p class="svArticle section clear" id="">A graph is an abstraction used widely in computer science. It is a set of nodes plus a set of edges between the nodes. Nodes and edges may also be called vertices and arcs, respectively. If the edges may only be traversed in one direction, the graph is known as a directed graph. The graph can be conceptualized as balls in space with arrows connecting them. Importantly, each directed edge represents a connection from one source node to one sink node. Collections of edges form paths that visit nodes in some order, such that the sink node of one edge forms the source node for any subsequent nodes. A special kind of path, called a simple path, is one that contains only distinct nodes (each node is visited at most once). A simple path may not intersect itself, by definition, and one may additionally require that no other paths intersect it. The nodes and edges may be assigned a variety of attributes and semantics.</p><!--VALIDHTML--><p class="svArticle section clear" id="">An overlap graph represents the sequencing reads and their overlaps <span id="bbib23"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib23" id="ancbbib23" class="intra_ref">[23]</a></span>. The overlaps must be pre-computed by a series of (computationally expensive) pair-wise sequence alignments. Conceptually, the graph has nodes to represent the reads and edges to represent overlaps. In practice, the graph might have distinct elements or attributes to distinguish the 5′ and 3′ ends of reads, the forward and reverse complement sequences of reads, the lengths of reads, the lengths of overlaps, and the type of overlap (suffix-to-prefix or containment). Paths through the graph are the potential contigs, and paths can be converted to sequence. Paths may have mirror images representing the reverse complement sequence. There are two ways to force paths to obey the semantics of double-stranded DNA. If the graph has separate nodes for read ends, then paths must exit the opposite end of the read they enter. If the graph has separate edges for the forward and reverse strands, then paths must exit a node on the same strand they enter.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The de Bruijn graph was developed outside the realm of DNA sequencing to represent strings from a finite alphabet. The nodes represent all possible fixed-length strings. The edges represent suffix-to-prefix perfect overlaps.</p><!--VALIDHTML--><p class="svArticle section clear" id="">A K-mer graph is a form of de Bruijn graph. Its nodes represent all the fixed-length subsequences drawn from a larger sequence. Its edges represent all the fixed-length overlaps between subsequences that were consecutive in the larger sequence. In one formulation <span id="bbib24"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib24" id="ancbbib24" class="intra_ref">[24]</a></span>, there is one edge for the K-mer that starts at each base (excluding the last K-1 bases). The nodes represent overlaps of K-1 bases. Alternately <span id="bbib25"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib25" id="ancbbib25" class="intra_ref">[25]</a></span>, there is one node representing the K-mer that starts at each base. The edges represent overlaps of K-1 bases. By construction, the graph contains a path corresponding to the original sequence (<span id="bfig1"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig1" id="ancbfig1" class="intra_ref">Fig.&nbsp;1</a></span>). The path converges on itself at graph elements representing K-mers in the sequence whose multiplicity is greater than one.</p><!--VALIDHTML--><div class="figTblUpiOuter svArticle" id="figure_fig1"><div><dl class="figure" id="fig1" data-t="f"><dt class="autoScroll" data-style="height:168px;width:311px;"><a class="figureLink" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#gr1" title="A read represented by K-mer graphs. (a) The read is represented by two types of ..."><img class="imgLazyJSB figure large nrmImg" border="0" alt="A read represented by K-mer graphs. (a) The read is represented by two types of ..." src="./Assembly algorithms for next-generation sequencing data_files/1-s2.0-S0888754310000492-gr1.jpg" data-thumbeid="1-s2.0-S0888754310000492-gr1.sml" data-fulleid="1-s2.0-S0888754310000492-gr1.jpg" data-imgeids="1-s2.0-S0888754310000492-gr1.jpg" height="168" width="311" data-thumbheight="118" data-thumbwidth="219" data-fullheight="168" data-fullwidth="311" data-loaded="true" style="display: inline; height: 168px; width: 311px;"><noscript class="figure large" border="0" alt="Full-size image" data-thumbeid="1-s2.0-S0888754310000492-gr1.sml" data-fulleid="1-s2.0-S0888754310000492-gr1.jpg">&lt;img class="figure large" border="0" alt="A read represented by K-mer graphs. (a) The read is represented by two types of ..." src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0888754310000492-gr1.jpg" data-thumbEID="1-s2.0-S0888754310000492-gr1.sml" data-imgEIDs="1-s2.0-S0888754310000492-gr1.jpg" data-fullEID="1-s2.0-S0888754310000492-gr1.jpg"&gt;</noscript></a></dt><dd id="labelCaptionfig1"><div class="caption">Fig.&nbsp;1.&nbsp;<p id="">A read represented by K-mer graphs. (a) The read is represented by two types of K-mer graph with K&nbsp;=&nbsp;4. Larger values of K are used for real data. (b) The graph has a node for every K-mer in the read plus a directed edge for every pair of K-mers that overlap by K-1 bases in the read. (c) An equivalent graph has an edge for every K-mer in the read and the nodes implicitly represent overlaps of K-1 bases. In these examples, the paths are simple because the value K&nbsp;=&nbsp;4 is larger than the 2&nbsp;bp repeats in the read. The read sequence is easily reconstructed from the path in either graph.</p><!--caption--></div></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#">Figure options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0888754310000492&amp;_image=1-s2.0-S0888754310000492-gr1.jpg&amp;_cid=272501&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=08887543&amp;_pii=S0888754310000492&amp;md5=39d161ef65f87d842adc4f1dc693b2ef" data-label="labelCaptionfig1" data-ref="" data-format="S0888754310000492" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#gr1" title="View full-size image - Opens in new window">Download full-size image</a></li><li><a class="S_C_high_quality" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0888754310000492&amp;_image=1-s2.0-S0888754310000492-gr1_lrg.jpg&amp;_cid=272501&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_isHiQual=Y&amp;_issn=08887543&amp;_pii=S0888754310000492&amp;md5=b8671e1c7c791676ae4dd5857cae050d" data-label="labelCaptionfig1" data-ref="" data-format="S0888754310000492" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#gr1" title="View high quality image - Opens in new window">Download high-quality image (50 K)</a></li><li><a class="ppt S_C_ppt" title="Download as PowerPoint slide" href="http://services.elsevier.com/SDWebExport/export/figure/S0888754310000492/1-s2.0-S0888754310000492-gr1.jpg/ppt">Download as PowerPoint slide</a></li></ul></dd></dl></div></div><!--VALIDHTML--><p class="svArticle section clear" id="">A repeat graph is an application of the K-mer graph&nbsp;<span id="bbib26"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib26" id="ancbbib26" class="intra_ref">[26]</a></span>. It provides a succinct graph representation of the repetitiveness of a genome. Nodes and edges are drawn from an assembled reference sequence. Whereas non-repetitive genomic sequence would induce a single path through the graph, repeats induce convergence and divergence of paths, as well as cycles. Repeat graphs can be used to identify and catalog repeats <span id="bbib27"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib27" id="ancbbib27" class="intra_ref">[27]</a></span>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">A K-mer graph may represent many sequences instead of one. In its application to WGS assembly, the K-mer graph represents the input reads. Each read induces a path. Reads with perfect overlaps induce a common path. Thus, perfect overlaps are detected implicitly without any pair-wise sequence alignment calculation (<span id="bfig2"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig2" id="ancbfig2" class="intra_ref">Fig.&nbsp;2</a></span>). Compared to overlap graphs, K-mer graphs are more sensitive to repeats and sequencing errors. Paths in overlap graphs converge at repeats longer than a read, but paths in K-mer graphs converge at perfect repeats of length K or more, and K must be less than the read length. Each single-base sequencing error induces up to K false nodes in the K-mer graph. Each false node has a chance of matching some other node and thereby inducing a false convergence of paths.</p><!--VALIDHTML--><div class="figTblUpiOuter svArticle" id="figure_fig2"><div><dl class="figure" id="fig2" data-t="f"><dt class="autoScroll" data-style="height:180px;width:335px;"><a class="figureLink" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#gr2" title="A pair-wise overlap represented by a K-mer graph. (a) Two reads have an ..."><img class="imgLazyJSB figure large nrmImg" border="0" alt="A pair-wise overlap represented by a K-mer graph. (a) Two reads have an ..." src="./Assembly algorithms for next-generation sequencing data_files/1-s2.0-S0888754310000492-gr2.jpg" data-thumbeid="1-s2.0-S0888754310000492-gr2.sml" data-fulleid="1-s2.0-S0888754310000492-gr2.jpg" data-imgeids="1-s2.0-S0888754310000492-gr2.jpg" height="180" width="335" data-thumbheight="118" data-thumbwidth="219" data-fullheight="180" data-fullwidth="335" data-loaded="true" style="display: inline; height: 180px; width: 335px;"><noscript class="figure large" border="0" alt="Full-size image" data-thumbeid="1-s2.0-S0888754310000492-gr2.sml" data-fulleid="1-s2.0-S0888754310000492-gr2.jpg">&lt;img class="figure large" border="0" alt="A pair-wise overlap represented by a K-mer graph. (a) Two reads have an ..." src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0888754310000492-gr2.jpg" data-thumbEID="1-s2.0-S0888754310000492-gr2.sml" data-imgEIDs="1-s2.0-S0888754310000492-gr2.jpg" data-fullEID="1-s2.0-S0888754310000492-gr2.jpg"&gt;</noscript></a></dt><dd id="labelCaptionfig2"><div class="caption">Fig.&nbsp;2.&nbsp;<p id="">A pair-wise overlap represented by a K-mer graph. (a) Two reads have an error-free overlap of 4 bases. (b) One K-mer graph, with K&nbsp;=&nbsp;4, represents both reads. The pair-wise alignment is a by-product of the graph construction. (c) The simple path through the graph implies a contig whose consensus sequence is easily reconstructed from the path.</p><!--caption--></div></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#">Figure options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0888754310000492&amp;_image=1-s2.0-S0888754310000492-gr2.jpg&amp;_cid=272501&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=08887543&amp;_pii=S0888754310000492&amp;md5=fb25a34ce1b36bab3725e684671a902a" data-label="labelCaptionfig2" data-ref="" data-format="S0888754310000492" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#gr2" title="View full-size image - Opens in new window">Download full-size image</a></li><li><a class="S_C_high_quality" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0888754310000492&amp;_image=1-s2.0-S0888754310000492-gr2_lrg.jpg&amp;_cid=272501&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_isHiQual=Y&amp;_issn=08887543&amp;_pii=S0888754310000492&amp;md5=fe46054c0e16ad73be44460ef50248cb" data-label="labelCaptionfig2" data-ref="" data-format="S0888754310000492" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#gr2" title="View high quality image - Opens in new window">Download high-quality image (64 K)</a></li><li><a class="ppt S_C_ppt" title="Download as PowerPoint slide" href="http://services.elsevier.com/SDWebExport/export/figure/S0888754310000492/1-s2.0-S0888754310000492-gr2.jpg/ppt">Download as PowerPoint slide</a></li></ul></dd></dl></div></div><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_4"></div></div><div id="frag_4" class="page_fragment" data-fid="4" style="display: block;"><p class="svArticle section clear" id="">Real-world WGS data induces problems in overlap graphs and K-mer graphs.</p><dl class="listitem" id="list_"><dt class="label">•</dt><dd><p id="">Spurs are short, dead-end divergences from the main path (<span id="bfig3"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig3" id="ancbfig3" class="intra_ref">Fig.&nbsp;3</a></span>a). They are induced by sequencing error toward one end of a read. They can be induced by coverage dropping to zero.</p><div class="figTblUpiOuter svArticle" id="figure_fig3"><div><dl class="figure" id="fig3" data-t="f"><dt class="autoScroll" data-style="height:233px;width:266px;"><a class="figureLink" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#gr3" title="Complexity in K-mer graphs can be diagnosed with read multiplicity information. ..."><img class="imgLazyJSB figure large nrmImg" border="0" alt="Complexity in K-mer graphs can be diagnosed with read multiplicity information. ..." src="./Assembly algorithms for next-generation sequencing data_files/1-s2.0-S0888754310000492-gr3.jpg" data-thumbeid="1-s2.0-S0888754310000492-gr3.sml" data-fulleid="1-s2.0-S0888754310000492-gr3.jpg" data-imgeids="1-s2.0-S0888754310000492-gr3.jpg" height="233" width="266" data-thumbheight="164" data-thumbwidth="187" data-fullheight="233" data-fullwidth="266" data-loaded="true" style="display: inline; height: 233px; width: 266px;"><noscript class="figure large" border="0" alt="Full-size image" data-thumbeid="1-s2.0-S0888754310000492-gr3.sml" data-fulleid="1-s2.0-S0888754310000492-gr3.jpg">&lt;img class="figure large" border="0" alt="Complexity in K-mer graphs can be diagnosed with read multiplicity information. ..." src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0888754310000492-gr3.jpg" data-thumbEID="1-s2.0-S0888754310000492-gr3.sml" data-imgEIDs="1-s2.0-S0888754310000492-gr3.jpg" data-fullEID="1-s2.0-S0888754310000492-gr3.jpg"&gt;</noscript></a></dt><dd id="labelCaptionfig3"><div class="caption">Fig.&nbsp;3.&nbsp;<p id="">Complexity in K-mer graphs can be diagnosed with read multiplicity information. In these graphs, edges represented in more reads are drawn with thicker arrows. (a) An errant base call toward the end of a read causes a “spur” or short dead-end branch. The same pattern could be induced by coincidence of zero coverage after polymorphism near a repeat. (b) An errant base call near a read middle causes a “bubble” or alternate path. Polymorphisms between donor chromosomes would be expected to induce a bubble with parity of read multiplicity on the divergent paths. (c) Repeat sequences lead to the “frayed rope” pattern of convergent and divergent paths.</p><!--caption--></div></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#">Figure options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0888754310000492&amp;_image=1-s2.0-S0888754310000492-gr3.jpg&amp;_cid=272501&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=08887543&amp;_pii=S0888754310000492&amp;md5=334599c580859baf4ccd3e08af21e2f7" data-label="labelCaptionfig3" data-ref="" data-format="S0888754310000492" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#gr3" title="View full-size image - Opens in new window">Download full-size image</a></li><li><a class="S_C_high_quality" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0888754310000492&amp;_image=1-s2.0-S0888754310000492-gr3_lrg.jpg&amp;_cid=272501&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_isHiQual=Y&amp;_issn=08887543&amp;_pii=S0888754310000492&amp;md5=a3000833310e39450020d04921f7bc02" data-label="labelCaptionfig3" data-ref="" data-format="S0888754310000492" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#gr3" title="View high quality image - Opens in new window">Download high-quality image (50 K)</a></li><li><a class="ppt S_C_ppt" title="Download as PowerPoint slide" href="http://services.elsevier.com/SDWebExport/export/figure/S0888754310000492/1-s2.0-S0888754310000492-gr3.jpg/ppt">Download as PowerPoint slide</a></li></ul></dd></dl></div></div><p></p></dd><dt class="label">•</dt><dd><p id="">Bubbles are paths that diverge then converge (<span id="bfig3"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig3" id="ancbfig3" class="intra_ref">Fig.&nbsp;3</a></span>b). They are induced by sequencing error toward the middle of a read, and by polymorphism in the target. Efficient bubble detection is non-trivial <span id="bbib28"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib28" id="ancbbib28" class="intra_ref">[28]</a></span>.</p></dd><dt class="label">•</dt><dd><p id="">Paths that converge then diverge form the frayed rope pattern (<span id="bfig3"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig3" id="ancbfig3" class="intra_ref">Fig.&nbsp;3</a></span>c). They are induced by repeats in the target genome.</p></dd><dt class="label">•</dt><dd><p id="">Cycles are paths that converge on themselves. They are induced by repeats in the target. For instance, short tandem repeats induce small cycles.</p></dd></dl><p></p><!--VALIDHTML--><p class="svArticle section clear" id="">In general, branching and convergence increases graph complexity, leading to tangles that are difficult to resolve. Much of the complexity is due to repeats in the target and sequencing error in the reads.</p><!--VALIDHTML--><p class="svArticle section clear" id="">In the graph context, assembly is a graph reduction problem. Most optimal graph reductions belong to a class of problems, called NP-hard, for which no efficient solution is known <span id="bbib29"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib29" id="ancbbib29" class="intra_ref">[29]</a></span>. Therefore, assemblers rely on heuristic algorithms and approximation algorithms to remove redundancy, repair errors, reduce complexity, enlarge simple paths and otherwise simplify the graph.</p><!--VALIDHTML--><h2 class="svArticle" id="section_lbl3">Greedy Graph-based Assemblers</h2><p class="svArticle section clear" id="">The first NGS assembly packages used greedy algorithms. These have been reviewed well elsewhere <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib3" id="bbib3" class="intra_ref">[3]</a>&nbsp;and&nbsp;<a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib30" id="bbib30" class="intra_ref">[30]</a>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The greedy algorithms apply one basic operation: given any read or contig, add one more read or contig. The basic operation is repeated until no more operations are possible. Each operation uses the next highest-scoring overlap to make the next join. The scoring function measures, for instance, the number of matching bases in the overlap. Thus the contigs grow by greedy extension, always taking on the read that is found by following the highest-scoring overlap. The greedy algorithms can get stuck at local maxima if the contig at hand takes on reads that would have helped other contigs grow even larger.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The greedy algorithms are implicit graph algorithms. They drastically simplify the graph by considering only the high-scoring edges. As an optimization, they may actually instantiate just one overlap for each read end they examine. They may also discard each overlap immediately after contig extension.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Like all assemblers, the greedy algorithms need mechanisms to avoid incorporating false-positive overlaps into contigs. Overlaps induced by repetitive sequence may score higher than overlaps induced by common position of origin. An assembler that builds on false-positive overlaps will join unrelated sequences to either side of a repeat to produce chimera.</p><!--VALIDHTML--><p class="svArticle section clear" id="">SSAKE <span id="bbib31"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib31" id="ancbbib31" class="intra_ref">[31]</a></span> was the first short-read assembler. It was designed for unpaired short reads of uniform length. It was based on the notion that high coverage would provide a tiling in error-free reads if the erroneous reads could be avoided. SSAKE does not use a graph explicitly. It does use a lookup table of reads indexed by their prefixes. SSAKE iteratively searches for reads that overlap one contig end. Its candidate reads must have a prefix-to-suffix identical overlap whose length is above a threshold. SSAKE chooses carefully among multiple reads with equally long overlaps. First, it prefers reads with end-to-end confirmation in other reads. This favors error-free reads. Second, the software detects when the set of candidates presents multiple extensions. In particular, it detects when the candidate read suffixes exhibit differences that are each confirmed in other reads. This is equivalent to finding a branch in a graph. At this point, the software terminates the contig extension. Users can elect to override the “stringent” behavior, in which case SSAKE takes the higher-scoring extension. When no reads satisfy the initial minimum threshold, the program decrements the threshold until a second minimum is reached. Thus, user settings determine how aggressively SSAKE extends through possible repeat boundaries and low-coverage regions. SSAKE has been extended to exploit paired-end reads and imperfectly matching reads <span id="bbib32"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib32" id="ancbbib32" class="intra_ref">[32]</a></span>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">SHARCGS <span id="bbib33"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib33" id="ancbbib33" class="intra_ref">[33]</a></span> also operates on uniform-length, high-coverage, unpaired short reads. It adds pre- and post-processor functionality to the basic SSAKE algorithm. The pre-processor filters erroneous reads by requiring a minimum number of full-length exact matches in other reads. An even higher-stringency filter is optional, requiring that the combined QVs of matching reads exceed a minimum threshold. SHARCGS filters the raw read set three times, each at a different stringency setting, to generate three filtered sets. It assembles each set separately by iterative contig extension. Then, in a post-process, it merges the three contig sets using sequence alignment. The merger aims to extend contigs from highly confirmed reads by integrating longer contigs from lower-stringency filters.</p><!--VALIDHTML--><p class="svArticle section clear" id="">VCAKE <span id="bbib34"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib34" id="ancbbib34" class="intra_ref">[34]</a></span> is another iterative extension algorithm. Unlike its predecessors, it could incorporate imperfect matches during contig extension. VCAKE was later combined with Newbler in a pipeline for Solexa+454 hybrid data <span id="bbib35"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib35" id="ancbbib35" class="intra_ref">[35]</a></span>. Another pipeline had combined Newbler and Celera Assembler for 454+Sanger hybrid data <span id="bbib36"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib36" id="ancbbib36" class="intra_ref">[36]</a></span>. Both pipelines “shred” contigs from the first assembler to produce pseudo-reads suitable for the second assembler. The latter pipeline adjusts the read coverage and base call quality values in the pseudo-reads it generates. This helps the secondary assembler give appropriate weight to high-coverage contigs from the primary assembly, for instance during consensus base calling.</p><!--VALIDHTML--><h2 class="svArticle" id="section_lbl4">Overlap/Layout/Consensus Assemblers</h2><p class="svArticle section clear" id="">The OLC approach was typical of the Sanger-data assemblers. It was optimized for large genomes in software including Celera Assembler <span id="bbib37"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib37" id="ancbbib37" class="intra_ref">[37]</a></span>, Arachne <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib38" id="bbib38" class="intra_ref">[38]</a>&nbsp;and&nbsp;<a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib39" id="bbib39" class="intra_ref">[39]</a>, and CAP and PCAP <span id="bbib40"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib40" id="ancbbib40" class="intra_ref">[40]</a></span>. The OLC approach has been reviewed elsewhere <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib41" id="bbib41" class="intra_ref">[41]</a>, <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib42" id="bbib42" class="intra_ref">[42]</a>&nbsp;and&nbsp;<a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib43" id="bbib43" class="intra_ref">[43]</a>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">OLC assemblers use an overlap graph. Their operation has three phases:</p><dl class="listitem" id="list_"><dt class="label">1.</dt><dd><p id="">Overlap discovery involves all-against-all, pair-wise read comparison. The seed &amp; extend heuristic algorithm is used for efficiency. The software pre-computes K-mer content across all reads, selects overlap candidates that share K-mers, and computes alignments using the K-mers as alignment seeds. Overlap discovery is sensitive to settings of K-mer size, minimum overlap length, and minimum percent identity required for an overlap. These three parameters affect robustness in the face of base calling error and low-coverage sequencing. Larger parameter values lead to more accurate but shorter contigs. Overlap discovery can run in parallel with a matrix partition.</p></dd><dt class="label">2.</dt><dd><p id="">Construction and manipulation of an overlap graph leads to an approximate read layout. The overlap graph need not include the sequence base calls, so large-genome graphs may fit into practical amounts of computer memory.</p></dd><dt class="label">3.</dt><dd><p id="">Multiple sequence alignment (MSA) determines the precise layout and then the consensus sequence. There is no known efficient method to compute the optimal MSA <span id="bbib44"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib44" id="ancbbib44" class="intra_ref">[44]</a></span>. Therefore, the consensus phase uses progressive pair-wise alignments guided by, for instance, the approximate read layout. The consensus phase must load the sequence base calls into memory. It can run in parallel, partitioned by contig.</p></dd></dl><p></p><!--VALIDHTML--><p class="svArticle section clear" id="">Newbler <span id="bbib45"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib45" id="ancbbib45" class="intra_ref">[45]</a></span> is widely used software distributed by 454 Life Sciences. The first release, described in the published supplement, targeted unpaired reads of approximately 100&nbsp;bp as generated by the GS 20 machine. Newbler has since been revised, in particular to build scaffolds from paired-end constraints. As described in 2005, Newbler implements OLC twice. The first-phase OLC generates unitigs from reads. Unitigs are mini-assemblies that are, ideally, uncontested by overlaps to reads in other unitigs <span id="bbib37"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib37" id="ancbbib37" class="intra_ref">[37]</a></span>. The unitigs serve as preliminary, high-confidence, conservative contigs that seed the rest of the assembly pipeline. The second-phase OLC generates larger contigs from the unitigs. This phase joins unitigs into a contig layout based on pair-wise overlaps between unitigs. It may split unitigs whose prefix and suffix align to different contigs. Unitig splitting may split individual reads, leading to reads placed in multiple contigs. Such reads may have been chimera or derived from a repeat boundary.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Newbler exploits coverage, if possible, to overcome base calling error. In particular, it uses instrument metrics to overcome inaccurate calls of the number of bases in homopolymer repeats. Newbler calculates unitig and contig consensus in “flow space” using the platform-supplied signal strength associated with each flow of a particular nucleotide. The normalized signal is proportionally correlated to the number of direct repeats of that nucleotide at that position in the read. Consensus calculation in “base space,” equivalent to rounding the signals before averaging, would sacrifice precision. For each column in the MSA, Newbler calculates the average signal and rounds to an integer to form the consensus.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The Newbler package offers functionality beyond <em>de novo</em> assembly, and it includes a comprehensive user guide. The Newbler software is distributed with the 454 sequencing machines. Customers receive frequent updates. Release descriptions indicate that recent versions differ from the published algorithm. The source code is not generally available.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The Celera Assembler <span id="bbib37"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib37" id="ancbbib37" class="intra_ref">[37]</a></span> is a Sanger-era OLC assembler revised for 454 data <span id="bbib46"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib46" id="ancbbib46" class="intra_ref">[46]</a></span>. The revised pipeline, called CABOG, discovers overlaps using compressed seeds. CABOG reduces homopolymer runs, that is, repeats of single letters, to single bases to overcome homopolymer run length uncertainty in data. CABOG builds initial unitigs excluding reads that are substrings of other reads. Substrings account for a large portion of the data in high-coverage 454 data due to highly variable read lengths. CABOG avoids the substring-reads initially because they are more susceptible to repeat-induced false overlaps.</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_5"></div></div><div id="frag_5" class="page_fragment" data-fid="5" style="display: block;"><p class="svArticle section clear" id="">CABOG applies a base call correction scheme first described for Arachne <span id="bbib38"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib38" id="ancbbib38" class="intra_ref">[38]</a></span>. It compares each read to its set of overlapping reads. It infers sequencing error at any base contradicted by a preponderance of overlaps. It does not fix the read. Rather, it adjusts the tabulated error rates in overlaps spanning the inferred error. Next, it applies a user-supplied threshold for error rates. From the overlaps that survive the error filter and a filter for minimum alignment length, CABOG selects one “best” overlap per read end. Best is defined as aligning the most bases. The best-overlap filter is presumed to eliminate many of the same overlaps removed by the more time-costly transitive edge removal algorithm <span id="bbib23"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib23" id="ancbbib23" class="intra_ref">[23]</a></span>, which was employed by the original Celera Assembler. CABOG constructs an overlap graph from the reads and “best” overlaps. Within the graph, it builds unitigs from maximal simple paths that are free of branches and intersections. CABOG next constructs a graph of unitigs plus paired-end constraints. Within that graph, it joins unitigs into contigs and connects contigs into scaffolds. It applies a series of graph reductions including removal of transitively inferable edges. Finally, CABOG derives consensus sequences by computing multiple sequence alignments from the scaffold layouts plus the read sequences.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Two assemblers apply the OLC approach to the short reads from the Solexa and SOLiD platforms. The Edena software <span id="bbib47"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib47" id="ancbbib47" class="intra_ref">[47]</a></span> discards duplicate reads and finds all perfect, error-free, overlaps. It removes individual overlaps that are redundant with pairs of other overlaps, an application of the transitive overlap reduction algorithm <span id="bbib23"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib23" id="ancbbib23" class="intra_ref">[23]</a></span>. Edena prunes spurs and bubbles. Edena was designed for unpaired reads of uniform length. The Shorty software <span id="bbib48"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib48" id="ancbbib48" class="intra_ref">[48]</a></span> attacks the special case where a few long reads are available to act as seeds to recruit short reads and their mate pairs. Proceeding in iterations, Shorty uses contigs to seed larger contigs.</p><!--VALIDHTML--><h2 class="svArticle" id="section_lbl5">The de Bruijn Graph Approach</h2><p class="svArticle section clear" id="">The third approach to assembly is most widely applied to the short reads from the Solexa and SOLiD platforms. It relies on K-mer graphs, whose attributes make it attractive for vast quantities of short reads. The K-mer graph does not require all-against-all overlap discovery, it does not (necessarily) store individual reads or their overlaps, and it compresses redundant sequence. Conversely, the K-mer graph does contain actual sequence and the graph can exhaust available memory on large genomes. Distributed memory approaches may alleviate this constraint.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The K-mer graph approach dates to an algorithm for Sanger read assembly <span id="bbib24"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib24" id="ancbbib24" class="intra_ref">[24]</a></span> based on a proposal <span id="bbib49"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib49" id="ancbbib49" class="intra_ref">[49]</a></span> for an even older sequencing technology; see <span id="bbib3"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib3" id="ancbbib3" class="intra_ref">[3]</a></span> for review. The approach is commonly called a de Bruijn graph (DBG) approach or an Eulerian approach <span id="bbib50"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib50" id="ancbbib50" class="intra_ref">[50]</a></span> based on an ideal scenario. Given perfect data – error-free K-mers providing full coverage and spanning every repeat – the K-mer graph would be a de Bruijn graph and it would contain an Eulerian path, that is, a path that traverses each edge exactly once. The path would be trivial to find making the assembly problem trivial by extension. Of course, K-mer graphs built from real sequencing data are more complicated.</p><!--VALIDHTML--><p class="svArticle section clear" id="">To the extent that the data is ideal, assembly is a by-product of the graph construction. The graph construction phase proceeds quickly using a constant-time hash table lookup for the existence of each K-mer in the data stream. Although the hash table consumes extra memory, the K-mer graph itself stores each K-mer at most once, no matter how many times the K-mer occurs in the reads. In terms of computer memory, the graph is smaller than the input reads to the extent that the reads share K-mers.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Pevzner <span id="bbib49"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib49" id="ancbbib49" class="intra_ref">[49]</a></span> explored problems that genomic repeats introduce. Repeats induce cycles in the K-mer graph. These would allow more than one possible reconstruction of the target. Idury and Waterman <span id="bbib24"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib24" id="ancbbib24" class="intra_ref">[24]</a></span> also explored problems of real data. They added two extra types of information to the K-mer graph and named the result a sequence graph. Each edge was labeled with the reads, and positions within each read, of the sequences that induced it. Where nodes had one inbound and one outbound edge, the three elements could be compressed into one edge. This was called the elimination of singletons. Further research led to the Euler software implementation <span id="bbib50"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib50" id="ancbbib50" class="intra_ref">[50]</a></span> for Sanger data. Impractical for large-scale Sanger sequencing projects, Euler and the DBG approach were well positioned when the Illumina platform started to produce data composed of very short unpaired reads of uniform size.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Three factors complicate the application of K-mer graphs to DNA sequence assembly.</p><dl class="listitem" id="list_"><dt class="label">1.</dt><dd><p id="">DNA is double stranded. The forward sequence of any given read may overlap the forward or reverse complement sequence of other reads. One K-mer graph implementation contains nodes and edges for both strands, taking care to avoid output of the entire assembly twice <span id="bbib24"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib24" id="ancbbib24" class="intra_ref">[24]</a></span>. Another implementation stores forward and reverse sequence together as cognate half-nodes with the constraint that paths enter and exit the same half <span id="bbib25"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib25" id="ancbbib25" class="intra_ref">[25]</a></span>. Yet another implementation represents alternate strands in a single node with two sides, constraining paths to enter and exit opposite sides <span id="bbib51"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib51" id="ancbbib51" class="intra_ref">[51]</a></span>.</p></dd><dt class="label">2.</dt><dd><p id="">Real genomes present complex repeat structures including tandem repeats, inverted repeats, imperfect repeats, and repeats inserted within repeats. Repeats longer than K lead to tangled K-mer graphs that complicate the assembly problem. Perfect repeats of length K or greater collapse inside the graph, leaving a local graph structure that resembles a rope with frayed ends (<span id="bfig3"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig3" id="ancbfig3" class="intra_ref">Fig.&nbsp;3</a></span>c); paths converge for the length of the repeat and then they diverge. Successful assembly requires separation of the converged path, which represents a collapsed repeat. The graph contains insufficient information to disambiguate the repeat. Assemblers typically consult the reads, and possibly the mate pairs, in attempts to resolve these regions.</p></dd><dt class="label">3.</dt><dd><p id="">A palindrome is a DNA sequence that is its own reverse complement. Palindromes induce paths that fold back on themselves. At least one assembler avoids these elegantly; Velvet <span id="bbib25"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib25" id="ancbbib25" class="intra_ref">[25]</a></span> requires K, the length of a K-mer, to be odd. An odd-size K-mer cannot match its reverse complement.</p></dd><dt class="label">4.</dt><dd><p id="">Real data includes sequencing error. DBG assemblers use several techniques to reduce sensitivity to this problem. First, they pre-process the reads to remove error. Second, they weight the graph edges by the number of reads that support them, and then “erode” the lightly supported paths. Third, they convert paths to sequences and use sequence alignment algorithms to collapse nearly identical paths. Many of these techniques derive from the Euler family of assemblers.</p></dd></dl><p></p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl5.1">The de Bruijn Graph in Euler</h3><p class="svArticle section clear" id="">The Euler software was developed for Sanger reads <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib26" id="bbib26" class="intra_ref">[26]</a>, <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib50" id="bbib50" class="intra_ref">[50]</a>&nbsp;and&nbsp;<a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib52" id="bbib52" class="intra_ref">[52]</a>. It was subsequently modified for short 454 GS20 reads <span id="bbib53"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib53" id="ancbbib53" class="intra_ref">[53]</a></span>, even shorter unpaired Illumina/Solexa reads <span id="bbib54"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib54" id="ancbbib54" class="intra_ref">[54]</a></span>, and paired-end Solexa reads <span id="bbib55"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib55" id="ancbbib55" class="intra_ref">[55]</a></span>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Euler applies a filter to the reads before it builds its graph. The filter detects erroneous base calls by noting low-frequency K-mers. The filter relies on redundancy of reads: most true K-mers should be repeated in several reads. The filter also relies on randomness of sequencing error: for any K where 4<sup>K</sup> exceeds twice the genome size, most erroneous K-mers should be unique. The Euler filter is implemented with a list of K-mers and their observed frequencies in the reads. The filter excludes or corrects low-frequency K-mers. Correction is especially important for short reads with high error and high coverage. Correction reduces the total number of K-mers and thus the node count in the graph. Correction risks masking true polymorphism <span id="bbib50"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib50" id="ancbbib50" class="intra_ref">[50]</a></span>. Correction can corrupt valid K-mers that had low coverage by chance. Correction could leave a read incorrect by settling on K-mers that each occur in several reads but never occur together in any read. OLC assemblers have an analogous base call correction step that uses overlaps rather than K-mers.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The Euler filter process is called spectral alignment <span id="bbib50"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib50" id="ancbbib50" class="intra_ref">[50]</a></span>. The software identifies sequencing error by comparing K-mer content between individual reads and all reads. It distrusts individual-read K-mers whose frequency in all reads is below a threshold. The threshold is chosen after calculating the distribution of K-mer frequencies present in the reads. The distribution is usually bi-modal. The first peak represents the many K-mers that occur once or twice, due to sequencing error (or low coverage). The second peak represents the redundant K-mers induced by the read coverage (or genomic repeats). Euler selects the threshold between the peaks and effectively labels all K-mers as bad or good. Euler then examines each read. For each read with bad K-mers, it executes a greedy exploration for base call substitutions that reduce the bad K-mer count <span id="bbib55"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib55" id="ancbbib55" class="intra_ref">[55]</a></span>. Finally, it either accepts a fully corrected read or rejects the read. (Rejected reads can be re-introduced to patch low-coverage regions after assembly.) Note Euler corrects substitution errors but not insertions or deletions, i.e. indels. Substitutions are the most common base call error type in Solexa data <span id="bbib12"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib12" id="ancbbib12" class="intra_ref">[12]</a></span>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Euler builds a K-mer graph from the filtered and corrected reads. Then it applies a series of graph manipulations designed to overcome the effects of sequencing errors and genomic repeats.</p><!--VALIDHTML--><p class="svArticle section clear" id="">By processing K-mers not reads, the K-mer graph construction discards long-range continuity information in the reads. Euler repairs this defect by threading the reads through the graph. Mapping a read onto the graph is easy. At least initially, the K-mers in reads map to unique nodes and reads are consistent with some path. Exploitation of the mapping is more complex. Reads ending inside a repeat are consistent with any path exiting the repeat, but reads spanning a repeat are consistent with fewer paths. For the latter, read threading pulls out one piece of string from a frayed rope pattern, thus resolving one copy of the collapsed repeat (<span id="bfig4"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig4" id="ancbfig4" class="intra_ref">Fig.&nbsp;4</a></span>a). Thus read threading constrains the set of valid paths through the graph. This allows resolution of repeats whose length is between K and the read length.</p><!--VALIDHTML--><div class="figTblUpiOuter svArticle" id="figure_fig4"><div><dl class="figure" id="fig4" data-t="f"><dt class="autoScroll" data-style="height:299px;width:489px;"><a class="figureLink" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#gr4" title="Three methods to resolve graph complexity. (a) Read threading joins paths across ..."><img class="imgLazyJSB figure large nrmImg" border="0" alt="Three methods to resolve graph complexity. (a) Read threading joins paths across ..." src="./Assembly algorithms for next-generation sequencing data_files/1-s2.0-S0888754310000492-gr4.jpg" data-thumbeid="1-s2.0-S0888754310000492-gr4.sml" data-fulleid="1-s2.0-S0888754310000492-gr4.jpg" data-imgeids="1-s2.0-S0888754310000492-gr4.jpg" height="299" width="489" data-thumbheight="134" data-thumbwidth="219" data-fullheight="299" data-fullwidth="489" data-loaded="true" style="display: inline; height: 299px; width: 489px;"><noscript class="figure large" border="0" alt="Full-size image" data-thumbeid="1-s2.0-S0888754310000492-gr4.sml" data-fulleid="1-s2.0-S0888754310000492-gr4.jpg">&lt;img class="figure large" border="0" alt="Three methods to resolve graph complexity. (a) Read threading joins paths across ..." src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0888754310000492-gr4.jpg" data-thumbEID="1-s2.0-S0888754310000492-gr4.sml" data-imgEIDs="1-s2.0-S0888754310000492-gr4.jpg" data-fullEID="1-s2.0-S0888754310000492-gr4.jpg"&gt;</noscript></a></dt><dd id="labelCaptionfig4"><div class="caption">Fig.&nbsp;4.&nbsp;<p id="">Three methods to resolve graph complexity. (a) Read threading joins paths across collapsed repeats that are shorter than the read lengths. (b) Mate threading joins paths across collapsed repeats that are shorter than the paired-end distances. (c) Path following chooses one path if its length fits the paired-end constraint. Reads and mates are shown as patterned lines. Not all tangles can be resolved by reads and mates. The non-branching paths are illustrative; they could be simplified to single edges or nodes.</p><!--caption--></div></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#">Figure options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0888754310000492&amp;_image=1-s2.0-S0888754310000492-gr4.jpg&amp;_cid=272501&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=08887543&amp;_pii=S0888754310000492&amp;md5=9e2701b854f4e8e56d4afb62eda6e56c" data-label="labelCaptionfig4" data-ref="" data-format="S0888754310000492" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#gr4" title="View full-size image - Opens in new window">Download full-size image</a></li><li><a class="S_C_high_quality" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0888754310000492&amp;_image=1-s2.0-S0888754310000492-gr4_lrg.jpg&amp;_cid=272501&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_isHiQual=Y&amp;_issn=08887543&amp;_pii=S0888754310000492&amp;md5=94ca28501f2cb37123245fe974acddcd" data-label="labelCaptionfig4" data-ref="" data-format="S0888754310000492" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#gr4" title="View high quality image - Opens in new window">Download high-quality image (126 K)</a></li><li><a class="ppt S_C_ppt" title="Download as PowerPoint slide" href="http://services.elsevier.com/SDWebExport/export/figure/S0888754310000492/1-s2.0-S0888754310000492-gr4.jpg/ppt">Download as PowerPoint slide</a></li></ul></dd></dl></div></div><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_6"></div></div><div id="frag_6" class="page_fragment" data-fid="6" style="display: block;"><p class="svArticle section clear" id="">A paired-end read is effectively a long read that is missing base calls in the middle. Euler treats paired ends this way to resolve repeats longer than individual reads. The technique could be called mate threading. Paired ends that span a repeat provide the evidence to join one path that enters a repeat to one path that exits the repeat (<span id="bfig4"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig4" id="ancbfig4" class="intra_ref">Fig.&nbsp;4</a></span>b). Paired ends can also resolve some complex tangles induced by repeats. A complex graph may have multiple paths between two nodes corresponding to opposite ends of a mate pair. Each path implies a putative DNA sequence. In many cases, only one of the paths implies a sequence whose length satisfies the paired-end constraint (<span id="bfig4"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig4" id="ancbfig4" class="intra_ref">Fig.&nbsp;4</a></span>c). Between any mate pair, there could be too many paths for exhaustive search to be feasible. Euler seems to restrict the search space using the mate constraint as a bound on path length. (The number of paths in a general graph scales with N<sup>E</sup> for N nodes and E out-going edges per node. K-mer graphs of DNA sequence can be constrained to E&nbsp;≤&nbsp;4 to represent the 4 possible one-letter extensions to a sequence. The constraint is violated by some graph simplifications, and N<sup>4</sup> is still not tractable.)</p><!--VALIDHTML--><p class="svArticle section clear" id="">After threading, Euler implements graph simplifications at regions with low and high depth of coverage in reads. Euler's spur erosion reduces branching in graph paths and thereby lengthens simple paths. The spurs are presumed due to sequencing error that survived the spectral alignment filter. Euler identifies remaining edges that appear repetitive and removes them from the set of paths. This is equivalent to breaking contigs at repeat boundaries in OLC assembly.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Reads from many platforms contain lower quality base calls at their 3′ ends, and the problem can be exacerbated by long-read protocols on short-read platforms. Euler addresses this problem by trusting read prefixes more than their suffixes. It chooses trustable prefixes during the error correction step. Prefix length varies per read. During read threading, prefixes and suffixes can map to multiple paths. By a heuristic, Euler trusts mappings that are significantly better than their second-best choice. Just as the suffixes would add coverage to a multiple sequence alignment, they add connectivity to the graph. The extra sequence leads to greater contig size. Euler chooses not to alter the assembly consensus sequence based on the suffixes, so the mapped suffixes contribute connectivity only.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Overlap graphs are sensitive to the minimum overlap length threshold, and K-mer graphs are sensitive to the parameter K. Larger values of K resolve longer repeats but they also fracture assemblies in regions of low read coverage. Euler addresses this with a heuristic. Euler constructs and simplifies two K-mer graphs with different values of K. It identifies edges present in the smaller-K graph that are missing in the larger-K graph. It adds corresponding pseudo-edges to the second graph. The borrowed edges extend paths in the second graph and thus enlarge contigs in the assembly. This technique effectively uses large K-mers to build reliable initial contigs, and then fills gaps with more prolific small K-mers. This is analogous to gap filling approaches in OLC assemblers <span id="bbib37"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib37" id="ancbbib37" class="intra_ref">[37]</a></span>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Some of the Euler software incorporates another structure called the A-Bruijn graph. It gets its name from being a combination of a de Bruijn graph and an adjacency matrix or A-matrix. Nodes of the graph represent consecutive columns in multiple sequence alignments. Compared to nodes representing K-mers in individual reads, the adjacency nodes can be less sensitive to sequencing error. The A-Bruijn graph was deployed for converting a genome sequence to a repeat graph and classifying repeats. It was proposed as a basis for assembly <span id="bbib26"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib26" id="ancbbib26" class="intra_ref">[26]</a></span>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">In summary, Euler compares de Bruijn graphs built from different K-mer sizes. Euler applies heuristics to mitigate graph complexity induced by sequencing error. It exploits low-quality read ends and paired-end constraints to tease apart graph tangles induced by genomic repeats. The software targets de novo assembly from short reads, including paired-ends, from the Solexa platform.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl5.2">The de Bruijn Graph in Velvet</h3><p class="svArticle section clear" id="">Velvet <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib25" id="bbib25" class="intra_ref">[25]</a>&nbsp;and&nbsp;<a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib56" id="bbib56" class="intra_ref">[56]</a> is a reliable and easy to use DBG assembler. Velvet makes extensive use of graph simplification to reduce simple non-intersecting paths to single nodes. Simplification compresses the graph without loss of information. Velvet invokes the simplification step during graph construction and again several times during the assembly process. The technique, introduced as elimination of singletons for K-mer graphs&nbsp;<span id="bbib24"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib24" id="ancbbib24" class="intra_ref">[24]</a></span>, is analogous to unitig formation in overlap graphs&nbsp;<span id="bbib23"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib23" id="ancbbib23" class="intra_ref">[23]</a></span> and OLC assemblers <span id="bbib37"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib37" id="ancbbib37" class="intra_ref">[37]</a></span>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Velvet prunes the K-mer graph by removing spurs iteratively. Its tip removal algorithm is similar to Euler's erosion procedure. The spur removal drastically reduced the graph size on real data <span id="bbib25"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib25" id="ancbbib25" class="intra_ref">[25]</a></span>, possibly because it was the pipeline's first attempt at filtering out base call errors. Velvet does not implement Euler's spectral alignment filter. Velvet has a parameter for the minimum number of occurrences in the reads for a K-mer to qualify as a graph node. The Velvet publication seems to discourage use of this naïve filter.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Velvet reduces graph complexity with a bounded search for bubbles in the graph. Velvet's tour bus algorithm uses breadth-first-search, fanning out as much as possible, starting at nodes with multiple out-going edges. Since graphs of real data can have bubbles within bubbles, an exhaustive search for all bubbles would be impractical. The search is bounded to make it tractable; the candidate paths are traversed in step, moving ahead one node on all paths per iteration, until the path lengths exceed a threshold. Velvet narrows the bubble candidates to those with a sequence similarity requirement on the alternate paths. Having found a bubble, Velvet removes the path representing fewer reads and, working outside the graph, re-aligns reads from the removed path to the remaining path. Because higher read multiplicity determines the target path, the re-aligner effectively calls the consensus bases by a column-wise voting algorithm. The operation risks “papering over” genuine sequence differences due to polymorphism in the donor DNA or over-collapse of near-identical repeats. Velvet's algorithm is similar to bulge removal in Euler <span id="bbib26"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib26" id="ancbbib26" class="intra_ref">[26]</a></span> and analogous to bubble detection and bubble smoothing in OLC assemblers <span id="bbib28"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib28" id="ancbbib28" class="intra_ref">[28]</a></span>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Velvet further reduces graph complexity by read threading. It removes paths that represent fewer reads than a threshold. This operation risks removing low-coverage sequence but it is thought to remove mostly spurious connections induced by convergent sequencing errors. Velvet exploits long reads, if any were provided, by an algorithm it calls Rock Band. This forms nodes out of paths that are confirmed by two or more long reads provided no other two long reads provide a consistent contradiction.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Velvet's final graph reduction involves mate pairs. Early versions used an algorithm called breadcrumb <span id="bbib25"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib25" id="ancbbib25" class="intra_ref">[25]</a></span> that is similar to mate pair threading in DBG algorithms <span id="bbib52"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib52" id="ancbbib52" class="intra_ref">[52]</a></span> and gap filling in OLC algorithms <span id="bbib37"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib37" id="ancbbib37" class="intra_ref">[37]</a></span>. It operated on pairs of long contigs (simple paths) connected by paired-ends. Using the long contigs as anchors, it tried to fill the gap between them with short contigs. It gathered short contigs linked to the long contigs, and applied a breadth-first search through the DBG for a single path linking the long contigs and by traversing the short contigs. Later versions of Velvet use an algorithm called pebble <span id="bbib56"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib56" id="ancbbib56" class="intra_ref">[56]</a></span>. In this algorithm, unique and repeat contigs substitute for breadcrumb's long and short contigs, respectively. The unique/repeat classifier is based on read coverage per contig. Velvet uses a statistical test similar to Celera Assembler's A-stat <span id="bbib37"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib37" id="ancbbib37" class="intra_ref">[37]</a></span> and a given coverage expectation. It exploits the given (per-library) insert length distribution to build an approximate contig layout. It searches the DBG for a path that is consistent with the layout.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Velvet may be run several times per data set to optimize selection of three critical parameters. The length of the K-mers is constrained to be odd to preclude nodes representing palindromic repeats. The minimum frequency expected of K-mers in the reads determines which K-mers are pruned <em>a-priori</em>. The expected coverage of the genome in reads controls spurious connection breaking.</p><!--VALIDHTML--><p class="svArticle section clear" id="">In summary, Velvet offers a full implementation of DBG assembly. It does not use an error-correction pre-processor, though it does have an error-avoidance read filter. It applies a series of heuristics that reduce graph complexity. The heuristics exploit local graph topology, read coverage, sequence identity, and paired-end constraints. The software targets de novo assembly from short reads with paired ends from the Solexa platform. An extension allows it to assemble data sets composed solely of SOLiD reads (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.solidsoftwaretools.com"><a href="http://www.solidsoftwaretools.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.solidsoftwaretools.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.solidsoftwaretools.com</a></span>). Memory requirements currently preclude Velvet from assembling very large genomes.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl5.3">The de Bruijn Graph in ABySS</h3><p class="svArticle section clear" id="">ABySS is a distributed implementation <span id="bbib51"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib51" id="ancbbib51" class="intra_ref">[51]</a></span> designed to addresses memory limitations of mammalian-size genome assembly by DBG. ABySS distributes the K-mer graph, and the graph computations, across a compute grid whose combined memory is quite large. This scheme allowed it to assemble 3.5 billion Solexa reads from a human donor.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Several problems confront any single-CPU algorithm that is ported to a grid. It must be possible to partition the problem for parallel computation, to distribute the partitions evenly across the grid, and to marshal the results. ABySS partitions the assembly at the granularity of the individual graph node. (Each graph node is processed separately. For efficiency, many graph nodes are assigned to each CPU.) The assignment of a graph node to CPU is accomplished by converting the K-mer to an integer. The formula is strand-neutral such that a K-mer and its reverse complement map to the same integer. The formula would be helpful if it somehow mapped neighbor K-mers to the same CPU. It is not clear to what extent this was accomplished.</p><!--VALIDHTML--><p class="svArticle section clear" id="">ABySS introduces parallel implementations of graph simplifications present in Euler and Velvet. ABySS iteratively removes spurs shorter than a threshold. ABySS performs bubble smoothing by bounded search and prefers the path supported by more reads. It also transforms simple non-intersecting paths into contigs and performs mate threading. ABySS uses a compact representation of the K-mer graph. Each graph node represents a K-mer and its reverse complement. Each graph node keeps 8 bits of extra information: the existence or non-existence of each of the four possible one-letter extensions at each end. The graph edges are implicit in this extra information. ABySS follows paths in parallel starting at arbitrary graph nodes per CPU. From any node, ABySS finds its successor elegantly: the node's last K-1 bases, plus a one-base extension indicated by an edge, is converted numerically to the address (including CPU assignment) of the successor node. When a path traverses a node on a different CPU, the process emits a request for information. Since inter-CPU communication is typically slow, the process works on other graph nodes while waiting for the response.</p><!--VALIDHTML--><p class="svArticle section clear" id="">In a post-process, ABySS exploits paired-end reads to merge contigs. ABySS does not build scaffolds. In summary, ABYSS is scalable assembly software for Solexa short reads and paired end reads.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl5.4">The de Bruijn Graph in AllPaths</h3><p class="svArticle section clear" id="">AllPaths is a DBG assembler intended for application to large genomes. It was published with results on simulated data <span id="bbib57"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib57" id="ancbbib57" class="intra_ref">[57]</a></span> and revised for real data <span id="bbib58"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib58" id="ancbbib58" class="intra_ref">[58]</a></span>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">AllPaths uses a read-correcting pre-processor related to spectral alignment in Euler. It trusts K-mers that occur at high frequency in reads and at high quality (where each base must be confirmed by a minimum number of base calls with QV above a threshold). It retains reads whose K-mers are trusted. The filter operates on K-mers for three values of K. The filter is relaxed in two ways. Reads are restored if up to two substitutions to low-QV base calls make its K-mers trusted. K-mers are restored later if they are essential for building a path between paired-end reads.</p><!--VALIDHTML--><p class="svArticle section clear" id="">AllPaths invokes a second pre-processor that creates “unipaths.” The process begins with the calculation of perfect read overlaps seeded by K-mers. It assigns numerical identifiers to the K-mers such that many K-mers seen consecutively in reads and overlaps receive consecutive identifiers. It populates a database of identifier intervals and the read links between them. It merges intervals to an extent that is consistent with all the reads. The operation is equivalent to DBG construction followed by elimination of singletons. The database implementation probably reduces the RAM requirement for the graph construction, which comes next.</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_7"></div></div><div id="frag_7" class="page_fragment" data-fid="7" style="display: block;"><p class="svArticle section clear" id="">AllPaths builds a DBG from the database. Its first graph operation is spur erosion, which it calls unitig graph shaving. AllPaths partitions the graph. Its goal is to resolve genomic repeats by assembling regions that are locally non-repetitive. It applies heuristics to choose partitions that form a tiling path across the genome. It seeds partitions with nodes corresponding to long, moderately covered, widely separated contigs. It populates partitions with nodes and reads linked by alignments and mate pairs. It seeks to fill gaps between paired-end reads by searching the K-mer graph for instances where exactly one path satisfies the distance constraint. For tighter variance on constraints, it uses short-range paired-ends first. AllPaths assembles each partition separately and in parallel. Then, AllPaths “glues” the local graphs where they have overlapping structure. This is analogous to joining contigs based on sequence overlaps. In the global graph, AllPaths heuristically removes spurs, small disconnected components, and paths not spanned by paired-ends. It unrolls cycles to match paired-end distance constraints; this decides copy number on tandem repeats. It also uses paired-ends to tease apart collapsed repeats that display the frayed rope pattern.</p><!--VALIDHTML--><p class="svArticle section clear" id="">In summary, AllPaths targets large-genome assembly using paired-end short reads from the Solexa platform. Its read filter uses quality values to fix some substitution errors. It simplifies its K-mer graph initially based on reads and overlaps. Going beyond read and mate threading, it applies the read and paired-end data outside the graph in a divide-and-conquer approach.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl5.5">The de Bruijn Graph in SOAPdenovo</h3><p class="svArticle section clear" id="">The SOAPdenovo program <span id="bbib59"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib59" id="ancbbib59" class="intra_ref">[59]</a></span> generated high-quality mammalian genome sequences from billions of Solexa reads of length 75&nbsp;bp and less <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib60" id="bbib60" class="intra_ref">[60]</a>&nbsp;and&nbsp;<a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib61" id="bbib61" class="intra_ref">[61]</a>. The program is freely available but without source code.</p><!--VALIDHTML--><p class="svArticle section clear" id="">SOAP filters and corrects reads using pre-set thresholds for K-mer frequencies. It builds a de Bruijn graph and erodes the tips. SOAP threads the reads and splits paths that display a symmetrical frayed rope pattern. SOAP removes bubbles with an algorithm like Velvet's tour bus, with higher read coverage determining the surviving path. Though SOAP's DBG implementation borrows from Euler and Velvet, its graph is more space-efficient. Devoid of read-tracking information, the graph required only 120&nbsp;GB RAM to store 5 billion nodes from 3.3 billion reads (after filtering). As described <span id="bbib59"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib59" id="ancbbib59" class="intra_ref">[59]</a></span>, the graph was constructed without data from large-insert libraries because those were suspected of generating many chimeric reads.</p><!--VALIDHTML--><p class="svArticle section clear" id="">SOAP builds contigs from reads by the DBG method. Then it discards the de Bruijn graph to build scaffolds. It maps all paired reads to the contig consensus sequences, including reads not used in the DBG. Then it builds a contig graph whose edges represent the inter-contig mate pair constraints. SOAP reduces complexity in the contig graph by removing edges that are transitively inferable from others. It also isolates contigs traversed by multiple, incompatible paths, since these appear to be collapsed repeats. Similar techniques are implemented in CABOG and its predecessor, Celera Assembler. As does AllPaths, SOAP processes the edges in order of insert size, from small to large. SOAP does this to preclude construction of scaffolds that interleave others. SOAP uses mate pairs to assign reads to gaps between neighbor contigs within a scaffold. This is similar to CABOG's “rocks and stones” technique <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib37" id="bbib37" class="intra_ref">[37]</a>&nbsp;and&nbsp;<a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib62" id="bbib62" class="intra_ref">[62]</a> and to Velvet's breadcrumbs and pebble techniques. SOAP uses de Bruijn graphs to assemble the reads assigned to each gap. In summary, SOAP is a large-genome implementation that amalgamates OLC and DBG techniques from its NGS assembly predecessors.</p><!--VALIDHTML--><h2 class="svArticle" id="section_lbl6">Other Software</h2><p class="svArticle section clear" id="">The PCAP long-read assembler also assembles 454 reads <span id="bbib40"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib40" id="ancbbib40" class="intra_ref">[40]</a></span>. LOCAS targets low-coverage short-read data (www-ab.informatik.uni-tuebingen.de/software/locas). The MIRA long-read assembler also assembles short reads (chevreux.org/projects_mira.html), as does Forge <span id="bbib63"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib63" id="ancbbib63" class="intra_ref">[63]</a></span>. Similar to Edena, Taipan <span id="bbib64"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib64" id="ancbbib64" class="intra_ref">[64]</a></span> implements greedy contig extension and transitive overlap reduction for unpaired short reads. Proprietary, commercial short-read assemblers include CLC Workbench (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.clcbio.com"><a href="http://www.clcbio.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.clcbio.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.clcbio.com</a></span>) and SeqMan (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.dnastart.com"><a href="http://www.dnastart.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.dnastart.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.dnastart.com</a></span>). Complete Genomics offers human genome sequencing services without distributing its sequencing or assembly technology (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.completegenomics.com"><a href="http://www.completegenomics.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.completegenomics.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.completegenomics.com</a></span>). SHRAP is a protocol for sequencing human-scale genomes <span id="bbib65"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib65" id="ancbbib65" class="intra_ref">[65]</a></span> with short reads by clone-based sequencing and a hierarchical assembly strategy. The protocol was demonstrated with Euler on simulated data. Network flow analysis has been described for resolution of repeat copy number during genome assembly <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib52" id="bbib52" class="intra_ref">[52]</a>&nbsp;and&nbsp;<a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib66" id="bbib66" class="intra_ref">[66]</a>. It was implemented with generic flow analysis software <span id="bbib67"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib67" id="ancbbib67" class="intra_ref">[67]</a></span>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The SOLiD platform from ABI has an unusual characteristic that requires special attention in software. The base calls are represented using the four digits 0, 1, 2, 3. These are referred to as colors. Each color is a di-base encoding of one base in relation to its preceding base. The reads' first base is a constant per run. Software gains robustness to error by forming alignments in color space and testing their validity by conversion to base space. Software support for <em>de novo</em> assembly of SOLiD reads is limited and has not been formally described. ABI provided an update to Velvet (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.solidsoftwaretools.com"><a href="http://www.solidsoftwaretools.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.solidsoftwaretools.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.solidsoftwaretools.com</a></span>) that assembles homogeneous SOLiD data sets. There is <em>de novo</em> assembly support in commercial software such as CLC Workbench (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.clcbio.com"><a href="http://www.clcbio.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.clcbio.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.clcbio.com</a></span>). Some packages that assemble SOLiD reads in the presence of Sanger reads, 454 reads, or contigs include Shorty <span id="bbib48"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib48" id="ancbbib48" class="intra_ref">[48]</a></span>, SeqWrite (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.seqwright.com"><a href="http://www.seqwright.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.seqwright.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.seqwright.com</a></span>) and NextGENe (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.softgenetics.com"><a href="http://www.softgenetics.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.softgenetics.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.softgenetics.com</a></span>).</p><!--VALIDHTML--><p class="svArticle section clear" id="">An alternative to <em>de novo</em> assembly is mapping. For some applications, sufficient information can be extracted from the mapping of reads to a reference sequence, such as a finished genome from a related individual. Mapped reads can reveal small-scale population differences such as substitutions and indels. (These are routinely referred to as single-nucleotide substitutions, or SNPs, and deletion-insertion polymorphisms, or DIPs, respectively.) Mapped mate pairs can reveal larger indels and structural re-arrangements. Using a mapping, it is possible to construct contigs and scaffolds, each with their consensus sequences, as in <em>de novo</em> assembly.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Short-read mapping software is widely available. Published software includes SOAP <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib68" id="bbib68" class="intra_ref">[68]</a>&nbsp;and&nbsp;<a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib69" id="bbib69" class="intra_ref">[69]</a>, MAQ <span id="bbib70"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib70" id="ancbbib70" class="intra_ref">[70]</a></span>, Bowtie <span id="bbib71"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib71" id="ancbbib71" class="intra_ref">[71]</a></span>, RMAP <span id="bbib72"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib72" id="ancbbib72" class="intra_ref">[72]</a></span>, CloudBurst <span id="bbib73"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib73" id="ancbbib73" class="intra_ref">[73]</a></span>, SHRiMP <span id="bbib74"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib74" id="ancbbib74" class="intra_ref">[74]</a></span>, RazerS <span id="bbib75"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib75" id="ancbbib75" class="intra_ref">[75]</a></span>, PerM <span id="bbib76"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib76" id="ancbbib76" class="intra_ref">[76]</a></span>, segemehl <span id="bbib77"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib77" id="ancbbib77" class="intra_ref">[77]</a></span>, GenomeMapper <span id="bbib78"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib78" id="ancbbib78" class="intra_ref">[78]</a></span>, and BOAT <span id="bbib79"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib79" id="ancbbib79" class="intra_ref">[79]</a></span>. Platform-specific mappers are available from platform vendors: Eland from Illumina (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.illumina.com"><a href="http://www.illumina.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.illumina.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.illumina.com</a></span>), Corona from ABI <span id="bbib80"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib80" id="ancbbib80" class="intra_ref">[80]</a></span>, and Reference Mapper from 454 (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.454.com"><a href="http://www.454.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.454.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.454.com</a></span>). Other commercial or unpublished solutions include ZOOM <span id="bbib81"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib81" id="ancbbib81" class="intra_ref">[81]</a></span>, CLC Workbench (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.clcbio.com"><a href="http://www.clcbio.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.clcbio.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.clcbio.com</a></span>), Novoalign (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.novocraft.com"><a href="http://www.novocraft.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.novocraft.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.novocraft.com</a></span>), Myrialign (savannah.nongnu.org/projects/myrialign) and Mr. Fast (mrfast.sf.net). Mapping-based variant discovery algorithms include MoDIL <span id="bbib82"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib82" id="ancbbib82" class="intra_ref">[82]</a></span>, VariationHunter <span id="bbib83"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib83" id="ancbbib83" class="intra_ref">[83]</a></span> and BreakDancer <span id="bbib84"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib84" id="ancbbib84" class="intra_ref">[84]</a></span>. Somewhere between <em>de novo</em> and mapping assembly there is reference-guided assembly. Software implementations include AMOScmp  <span id="bbib85"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib85" id="ancbbib85" class="intra_ref">[85]</a></span> as revised for short reads, MOSAIK as featured in a short-read, cross-species comparison <span id="bbib86"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib86" id="ancbbib86" class="intra_ref">[86]</a></span>, and SeqMan which is commercial software (<span id="" class="interref" data-locatortype="url" data-locatorkey="http://www.dnastar.com"><a href="http://www.dnastar.com/" data-url="/science/RedirectURL?_method=externObjLink&amp;_locator=url&amp;_cdi=272501&amp;_issn=08887543&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=http%253A%252F%252Fwww.dnastar.com" class="cExLink" target="externObjLink" onclick="var newWidth=((window.screen.availWidth*90)/100);var newHeight=((window.screen.availHeight*90)/100);var parms = &#39;status=yes,location=yes,&#39; + &#39;scrollbars=yes,resizable=yes,directories=yes,&#39; + &#39;toolbar=yes,menubar=yes,&#39; + &#39;width=&#39; + newWidth + &#39;,height=&#39; + newHeight + &#39;,screenX=10,screenY=10&#39;;var externalWin; externalWin=window.open(&#39;&#39;,&#39;externObjLink&#39;,parms); externalWin.focus()" data-itrprs="Y">www.dnastar.com</a></span>). A related approach, called gene-boosted assembly, finishes an assembly by comparison of related genomes and protein sequences <span id="bbib87"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib87" id="ancbbib87" class="intra_ref">[87]</a></span>. Our survey is no doubt incomplete and we apologize for omissions.</p><!--VALIDHTML--><h2 class="svArticle" id="section_lbl7">Summary and Outlook</h2><p class="svArticle section clear" id="">We have compared a dozen algorithms for the <em>de novo</em> assembly of whole-genome shotgun (WGS) data from next-generation sequencing (NGS) platforms.  <span id="btbl1"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#tbl1" id="ancbtbl1" class="intra_ref">Table&nbsp;1</a></span> offers a summary feature comparison. Our comparison is an interpretation of the primary literature. Few of the published descriptions include measures of the relative contributions by their various algorithmic features, and none include the unit tests and controls that might verify that the implementations follow the algorithms. No algorithm or implementation solves the WGS assembly problem. Each of the various software packages was published with claims about its own superiority. Each package seems to improve in subsequent software releases. In our experience, the success of an assembler depends largely on the sophistication of its heuristics for real reads including error, real genomes including repeats, and the limitations of modern computers.</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_8"></div></div><div id="frag_8" class="page_fragment" data-fid="8" style="display: block;"><div class="figTblUpiOuter svArticle" id="table_tbl1"><div><dl id="tbl1" data-t="t" class="table " data-label="Table 1"><dd class="lblCap"><div class="caption"><span class="label">Table&nbsp;1.
        </span><p>Feature comparison between de novo assemblers for whole-genome shotgun data from next-generation sequencing platforms. OLC refers to the overlap/layout/consensus architecture. DBG refers to the de Bruijn graph architecture. The table is based on the literature cited in the text. It may not reflect the current state of each software package.</p></div></dd><dd class="table" title="Feature comparison between de novo assemblers for whole-genome shotgun data from ..."><table><colgroup><col><col><col><col></colgroup><thead><tr><th rowspan="1" colspan="1" scope="col">Algorithm Feature</th><th rowspan="1" colspan="1" scope="col">Greedy Assemblers</th><th rowspan="1" colspan="1" scope="col">OLC Assemblers</th><th rowspan="1" colspan="1" scope="col">DBG Assemblers</th></tr></thead><tbody><tr><td rowspan="1" colspan="4" class="alignTDleft"><em>Modeled features of reads</em></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Base substitutions</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Euler, AllPaths, SOAP</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Homopolymer miscount</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG</td><td rowspan="1" colspan="1" class="alignTD"></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Concentrated error in 3′ end</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Euler</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Flow space</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Newbler</td><td rowspan="1" colspan="1" class="alignTD"></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Color space</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Shorty</td><td rowspan="1" colspan="1" class="alignTDleft">Velvet</td></tr><tr><td rowspan="1" colspan="4" class="alignTD"><br></td></tr><tr><td rowspan="1" colspan="4" class="alignTDleft"><em>Removal of erroneous reads</em></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Based on K-mer frequencies</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Euler, Velvet, AllPaths</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Based on K-mer freq and QV</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">AllPaths</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">For multiple values of K</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">AllPaths</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">By alignment to other reads</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG</td><td rowspan="1" colspan="1" class="alignTD"></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">By alignment and QV</td><td rowspan="1" colspan="1" class="alignTDleft">SHARCGS</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td></tr><tr><td rowspan="1" colspan="4" class="alignTD"><br></td></tr><tr><td rowspan="1" colspan="4" class="alignTDleft"><em>Correction of erroneous base calls</em></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Based on K-mer frequencies</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Euler, SOAP</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Based on Kmer freq and QV</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">AllPaths</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Based on alignments</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG</td><td rowspan="1" colspan="1" class="alignTD"></td></tr><tr><td rowspan="1" colspan="4" class="alignTD"><br></td></tr><tr><td rowspan="1" colspan="4" class="alignTDleft"><em>Approaches to graph construction</em></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Implicit</td><td rowspan="1" colspan="1" class="alignTDleft">SSAKE, SHARCGS, VCAKE</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Reads as graph nodes</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG, Newbler, Edena</td><td rowspan="1" colspan="1" class="alignTD"></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">K-mers as graph nodes</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Euler, Velvet, ABySS, SOAP</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Simple paths as graph nodes</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">AllPaths</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Multiple values of K</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Euler</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Multiple overlap stringencies</td><td rowspan="1" colspan="1" class="alignTDleft">SHARCGS</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td></tr><tr><td rowspan="1" colspan="4" class="alignTD"><br></td></tr><tr><td rowspan="1" colspan="4" class="alignTDleft"><em>Approaches to graph reduction</em></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Filter overlaps</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG</td><td rowspan="1" colspan="1" class="alignTD"></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Greedy contig extension</td><td rowspan="1" colspan="1" class="alignTDleft">SSAKE, SHARCGS, VCAKE</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Collapse simple paths</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG, Newbler</td><td rowspan="1" colspan="1" class="alignTDleft">Euler, Velvet, SOAP</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Erosion of spurs</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG, Edena</td><td rowspan="1" colspan="1" class="alignTDleft">Euler, Velvet, AllPaths, SOAP</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Transitive overlap reduction</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Edena</td><td rowspan="1" colspan="1" class="alignTD"></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Bubble smoothing</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Edena</td><td rowspan="1" colspan="1" class="alignTDleft">Euler, Velvet, SOAP</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Bubble detection</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">AllPaths</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Reads separate tangled paths</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Euler, SOAP</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Break at low coverage</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Velvet, SOAP</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Break at high coverage</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG</td><td rowspan="1" colspan="1" class="alignTDleft">Euler</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">High coverage indicates repeat</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG</td><td rowspan="1" colspan="1" class="alignTDleft">Velvet</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Special use of long reads</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Shorty</td><td rowspan="1" colspan="1" class="alignTDleft">Velvet</td></tr><tr><td rowspan="1" colspan="4" class="alignTD"><br></td></tr><tr><td rowspan="1" colspan="4" class="alignTDleft"><em>Graph partitions</em></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Partition by K-mers</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">ABySS</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Partition by scaffolds</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">AllPaths</td></tr><tr><td rowspan="1" colspan="4" class="alignTD"><br></td></tr><tr><td rowspan="1" colspan="4" class="alignTDleft"><em>Uses for mate pairs</em></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Constrain path searches</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Euler, Velvet, AllPaths</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Guide path selection</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">Euler, Allpaths</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Detect misassembled contigs</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG, Shorty</td><td rowspan="1" colspan="1" class="alignTD"></td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Merge contigs or fill gaps</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG, Shorty</td><td rowspan="1" colspan="1" class="alignTDleft">Velvet, ABySS, SOAP</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Transitive link reduction</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG</td><td rowspan="1" colspan="1" class="alignTDleft">SOAP</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Detect, avoid repeat contigs</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG</td><td rowspan="1" colspan="1" class="alignTDleft">Velvet, SOAP</td></tr><tr><td rowspan="1" colspan="1" class="alignTDleft">Create scaffolds</td><td rowspan="1" colspan="1" class="alignTD"></td><td rowspan="1" colspan="1" class="alignTDleft">CABOG, Shorty</td><td rowspan="1" colspan="1" class="alignTDleft">Euler, Velvet, AllPaths, SOAP</td></tr></tbody></table></dd><dd class="fullsizeTable"><a data-url="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0888754310000492&amp;_image=tbl1&amp;_cid=272501&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=08887543&amp;_pii=S0888754310000492&amp;_isTablePopup=Y&amp;md5=9dfbab08eefe2dbda68bee02bd5955bc" class="tableAnchor" title="Full-size table - Opens new window" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#">Full-size table</a></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#">Table options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#tbl1" class="viewWS S_C_table_view_WS" title="View in workspace">View in workspace</a></li><li><div class="downloadCsv"><a class="S_C_csv_download" title="Download as CSV" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#">Download as CSV</a></div></li></ul></dd></dl></div></div><!--VALIDHTML--><p class="svArticle section clear" id="">Large genomes tend to present more complex repeat structures. Most of the NGS assemblers discussed here were initially published with assemblies of bacterial-length genomes (or larger genomes by simulated reads). Four <em>de novo</em> assemblies of mammalian-scale genomes from Solexa short reads have been described in the scientific literature. The first assembly of the human genome from short reads  <span id="bbib51"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib51" id="ancbbib51" class="intra_ref">[51]</a></span> represented a software engineering feat. Later assemblies of panda <span id="bbib60"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib60" id="ancbbib60" class="intra_ref">[60]</a></span> and two humans <span id="bbib61"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib61" id="ancbbib61" class="intra_ref">[61]</a></span> used slightly larger reads and more variety in pair insert sizes and generated larger contigs. Improved mammalian assemblies are certain to become reality thanks to decreases in sequencing cost, increasing read lengths by every platform, more variety in paired-end protocols, continual improvement to assembly software, and the application of more powerful compute nodes and grids.</p><!--VALIDHTML--><p class="svArticle section clear" id="">In the graph context, optimal path discovery is impractical because there can be an exponential number of paths between any source and sink node. Each assembler searches the graph but each constrains its searches for reasons of scale. A search in CABOG's overlap graph follows only one outward edge per node. Searches in AllPaths and Velvet examine local regions of the K-mer graph that are anchored by large contigs and populated by direct and transitive paired-end linkage. Euler and ABySS seem to rely on branch-and-bound search algorithms.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Common to most assemblers, a core set of features is apparent:</p><dl class="listitem" id="list_"><dt class="label">•</dt><dd><p id="">Error detection and correction based on sequence composition of the reads.</p></dd><dt class="label">•</dt><dd><p id="">Graph construction to represent reads and their shared sequence.</p></dd><dt class="label">•</dt><dd><p id="">Reduction of simple non-intersecting paths to single nodes in the graph.</p></dd><dt class="label">•</dt><dd><p id="">Removal of error-induced paths. These are recognized as spurs or bubbles.</p></dd><dt class="label">•</dt><dd><p id="">Collapse of polymorphism-induced complexity. This is recognized as bubbles.</p></dd><dt class="label">•</dt><dd><p id="">Simplification of tangles using information outside the graph. Individual reads or paired-end reads act as constraints on path distance and outcome.</p></dd><dt class="label">•</dt><dd><p id="">Conversion of reduced paths to contigs and scaffolds.</p></dd><dt class="label">•</dt><dd><p id="">Reduction of alignments to a consensus sequence.</p></dd></dl><p></p><!--VALIDHTML--><p class="svArticle section clear" id="">OLC and DBG are two robust approaches to assembly. Both rely to some extent on a set of overlaps between the input reads. Both represent the overlaps in a directed graph. Their different graph representations are similar if not equivalent <span id="bbib66"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib66" id="ancbbib66" class="intra_ref">[66]</a></span>. The OLC approach directly incorporates connections (overlaps) of varying length, as expected in long-read, low-coverage data. K-mer graphs are limited initially to short connections (shared K-mers) of uniform size, though read threading mitigates this. The DBG approach is more appropriate for the large volumes of reads associated with short-read sequencing. DBG avoids the computationally expensive all-against-all pair-wise read comparisons. DBG avoids loading all the replicate sequences associated with high-coverage sequencing.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Both techniques must contend with noisy data. Sequencing error induces false positive and false negative overlaps. Noise filtration is inherent in overlap graphs constructed from non-identical alignments. K-mer graphs are more sensitive to sequencing error, as every miscalled base introduces up to K erroneous nodes. The OLC and DBG approaches both employ pre-processing steps to filter or correct unconfirmed portions of reads, as well as post-processes to repair graphs by erosion, smoothing, and threading.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The OLC assemblers grew up on long reads. The DBG assemblers proliferated with the introduction of short reads. The OLC assemblers target variable-length reads in the 100-800&nbsp;bp range. The DBG assemblers target uniformly sized reads in the 25-100&nbsp;bp range. Reads of the near future may be intermediate-sized, matching more closely the expectations of OLC assemblers. Further ahead, very long reads may become feasible while short reads may become even more affordable.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Reads of the future will challenge assembly software in many ways. Sequencing platforms may reveal inter-read associations richer than the paired-end model allows. Future platforms may target error rates that are higher or lower than today's standards. Almost certainly, data volume will continue to increase while manufacturing cost declines. The next-generation technology will surely be applied to larger genomes, more repetitive sequences, and less homogeneous samples. Developers of assembly algorithms will continue to be challenged by novel applications and larger data sets. Simultaneously, assembly developers will be challenged to extract more useful information from each sequence to enable lower coverage per genome and thus higher yield. The quest for more powerful and efficient assembly software remains an area of critical research.</p><!--VALIDHTML--><div class="articleText svArticle"><div class="articleText_indent"><h2 class="h2 secHeading" id="ack001">Acknowledgments</h2><p class="svArticle section clear" id="">The authors receive funding for assembly research from the National Institutes of Health via grant 2R01GM077117-04A1 from the National Institute of General Medical Sciences.</p></div></div><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_9"></div></div><div id="frag_9" class="page_fragment" data-fid="9" style="display: block;"><h2 id="bibl001">References</h2><div class="refText svRefs"><ol class="references"><li id="bib1" data-t="r"><ul class="reference" id="bibsbref1"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib1" class="label intra_ref">[1]</a></li><li class="author">F. Sanger, A.R. Coulson, B.G. Barrell, A.J. Smith, B.A. Roe</li><li class="title"><p>Cloning in single-stranded bacteriophage as an aid to rapid DNA sequencing</p></li><li class="source"><p>J. Mol. Biol., 143 (1980), pp. 161–178</p></li><li class="external refPlaceHolder" id="bibsbref1" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref1'] = new Array();
        </script></li><li id="bib2" data-t="r"><ul class="reference" id="bibsbref2"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib2" class="label intra_ref">[2]</a></li><li class="author">R. Staden</li><li class="title"><p>A strategy of DNA sequencing employing computer programs</p></li><li class="source"><p>Nucleic Acids Res., 6 (1979), pp. 2601–2610</p></li><li class="external refPlaceHolder" id="bibsbref2" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref2'] = new Array();
        </script></li><li id="bib3" data-t="r"><ul class="reference" id="bibsbref3"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib3" class="label intra_ref">[3]</a></li><li class="author">M. Pop</li><li class="title"><p>Genome assembly reborn: recent computational challenges</p></li><li class="source"><p>Brief. Bioinform., 10 (2009), pp. 354–366</p></li><li class="external refPlaceHolder" id="bibsbref3" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref3'] = new Array();
        </script></li><li id="bib4" data-t="r"><ul class="reference" id="bibsbref4"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib4" class="label intra_ref">[4]</a></li><li class="author">E.R. Mardis</li><li class="title"><p>The impact of next-generation sequencing technology on genetics</p></li><li class="source"><p>Trends Genet., 24 (2008), pp. 133–141</p></li><li class="external refPlaceHolder" id="bibsbref4" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref4'] = new Array();
        </script></li><li id="bib5" data-t="r"><ul class="reference" id="bibsbref5"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib5" class="label intra_ref">[5]</a></li><li class="author">O. Morozova, M.A. Marra</li><li class="title"><p>Applications of next-generation sequencing technologies in functional genomics</p></li><li class="source"><p>Genomics, 92 (2008), pp. 255–264</p></li><li class="external refPlaceHolder" id="bibsbref5" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref5'] = new Array();
        </script></li><li id="bib6" data-t="r"><ul class="reference" id="bibsbref6"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib6" class="label intra_ref">[6]</a></li><li class="author">R.L. Strausberg, S. Levy, Y.H. Rogers</li><li class="title"><p>Emerging DNA sequencing technologies for human genomic medicine</p></li><li class="source"><p>Drug Discov. Today, 13 (2008), pp. 569–577</p></li><li class="external refPlaceHolder" id="bibsbref6" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref6'] = new Array();
        </script></li><li id="bib7" data-t="r"><ul class="reference" id="bibsbref7"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib7" class="label intra_ref">[7]</a></li><li class="author">E. Pettersson, J. Lundeberg, A. Ahmadian</li><li class="title"><p>Generations of sequencing technologies</p></li><li class="source"><p>Genomics, 93 (2009), pp. 105–111</p></li><li class="external refPlaceHolder" id="bibsbref7" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref7'] = new Array();
        </script></li><li id="bib8" data-t="r"><ul class="reference" id="bibsbref8"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib8" class="label intra_ref">[8]</a></li><li class="author">F. Sanger, S. Nicklen, A.R. Coulson</li><li class="title"><p>DNA sequencing with chain-terminating inhibitors</p></li><li class="source"><p>Proc. Natl. Acad. Sci. U. S. A., 74 (1977), pp. 5463–5467</p></li><li class="external refPlaceHolder" id="bibsbref8" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref8'] = new Array();
        </script></li><li id="bib9" data-t="r"><ul class="reference" id="bibsbref9"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib9" class="label intra_ref">[9]</a></li><li class="author">J. Eid, A. Fehr, J. Gray, K. Luong, J. Lyle, G. Otto, P. Peluso, D. Rank, P. Baybayan, B. Bettman, A. Bibillo, K. Bjornson, B. Chaudhuri, F. Christians, R. Cicero, S. Clark, R. Dalal, A. Dewinter, J. Dixon, M. Foquet, A. Gaertner, P. Hardenbol, C. Heiner, K. Hester, D. Holden, G. Kearns, X. Kong, R. Kuse, Y. Lacroix, S. Lin, P. Lundquist, C. Ma, P. Marks, M. Maxham, D. Murphy, I. Park, T. Pham, M. Phillips, J. Roy, R. Sebra, G. Shen, J. Sorenson, A. Tomaney, K. Travers, M. Trulson, J. Vieceli, J. Wegener, D. Wu, A. Yang, D. Zaccarin, P. Zhao, F. Zhong, J. Korlach, S. Turner</li><li class="title"><p>Real-time DNA sequencing from single polymerase molecules</p></li><li class="source"><p>Science, 323 (2009), pp. 133–138</p></li><li class="external refPlaceHolder" id="bibsbref9" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref9'] = new Array();
        </script></li><li id="bib10" data-t="r"><ul class="reference" id="bibsbref10"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib10" class="label intra_ref">[10]</a></li><li class="author">B. Ewing, P. Green</li><li class="title"><p>Base-calling of automated sequencer traces using phred. II. Error probabilities</p></li><li class="source"><p>Genome Res., 8 (1998), pp. 186–194</p></li><li class="external refPlaceHolder" id="bibsbref10" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref10'] = new Array();
        </script></li><li id="bib11" data-t="r"><ul class="reference" id="bibsbref11"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib11" class="label intra_ref">[11]</a></li><li class="author">S.M. Huse, J.A. Huber, H.G. Morrison, M.L. Sogin, D.M. Welch</li><li class="title"><p>Accuracy and quality of massively parallel DNA pyrosequencing</p></li><li class="source"><p>Genome Biol., 8 (2007), p. R143</p></li><li class="external refPlaceHolder" id="bibsbref11" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref11'] = new Array();
        </script></li><li id="bib12" data-t="r"><ul class="reference" id="bibsbref12"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib12" class="label intra_ref">[12]</a></li><li class="author">J.C. Dohm, C. Lottaz, T. Borodina, H. Himmelbauer</li><li class="title"><p>Substantial biases in ultra-short read data sets from high-throughput DNA sequencing</p></li><li class="source"><p>Nucleic Acids Res., 36 (2008), p. e105</p></li><li class="external refPlaceHolder" id="bibsbref12" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref12'] = new Array();
        </script></li><li id="bib13" data-t="r"><ul class="reference" id="bibsbref13"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib13" class="label intra_ref">[13]</a></li><li class="author">O. Harismendy, P.C. Ng, R.L. Strausberg, X. Wang, T.B. Stockwell, K.Y. Beeson, N.J. Schork, S.S. Murray, E.J. Topol, S. Levy, K.A. Frazer</li><li class="title"><p>Evaluation of next generation sequencing platforms for population targeted sequencing studies</p></li><li class="source"><p>Genome Biol., 10 (2009), p. R32</p></li><li class="external refPlaceHolder" id="bibsbref13" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref13'] = new Array();
        </script></li><li id="bib14" data-t="r"><ul class="reference" id="bibsbref14"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib14" class="label intra_ref">[14]</a></li><li class="author">R.D. Fleischmann, M.D. Adams, O. White, R.A. Clayton, E.F. Kirkness, A.R. Kerlavage, C.J. Bult, J.F. Tomb, B.A. Dougherty, J.M. Merrick,  <em>et al.</em></li><li class="title"><p>Whole-genome random sequencing and assembly of Haemophilus influenzae Rd</p></li><li class="source"><p>Science, 269 (1995), pp. 496–512</p></li><li class="external refPlaceHolder" id="bibsbref14" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref14'] = new Array();
        </script></li><li id="bib15" data-t="r"><ul class="reference" id="bibsbref15"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib15" class="label intra_ref">[15]</a></li><li class="author">M.D. Adams, S.E. Celniker, R.A. Holt, C.A. Evans, J.D. Gocayne, P.G. Amanatides, S.E. Scherer, P.W. Li, R.A. Hoskins, R.F. Galle, R.A. George, S.E. Lewis, S. Richards, M. Ashburner, S.N. Henderson, G.G. Sutton, J.R. Wortman, M.D. Yandell, Q. Zhang, L.X. Chen, R.C. Brandon, Y.H. Rogers, R.G. Blazej, M. Champe, B.D. Pfeiffer, K.H. Wan, C. Doyle, E.G. Baxter, G. Helt, C.R. Nelson, G.L. Gabor, J.F. Abril, A. Agbayani, H.J. An, C. Andrews-Pfannkoch, D. Baldwin, R.M. Ballew, A. Basu, J. Baxendale, L. Bayraktaroglu, E.M. Beasley, K.Y. Beeson, P.V. Benos, B.P. Berman, D. Bhandari, S. Bolshakov, D. Borkova, M.R. Botchan, J. Bouck, P. Brokstein, P. Brottier, K.C. Burtis, D.A. Busam, H. Butler, E. Cadieu, A. Center, I. Chandra, J.M. Cherry, S. Cawley, C. Dahlke, L.B. Davenport, P. Davies, B. de Pablos, A. Delcher, Z. Deng, A.D. Mays, I. Dew, S.M. Dietz, K. Dodson, L.E. Doup, M. Downes, S. Dugan-Rocha, B.C. Dunkov, P. Dunn, K.J. Durbin, C.C. Evangelista, C. Ferraz, S. Ferriera, W. Fleischmann, C. Fosler, A.E. Gabrielian, N.S. Garg, W.M. Gelbart, K. Glasser, A. Glodek, F. Gong, J.H. Gorrell, Z. Gu, P. Guan, M. Harris, N.L. Harris, D. Harvey, T.J. Heiman, J.R. Hernandez, J. Houck, D. Hostin, K.A. Houston, T.J. Howland, M.H. Wei, C. Ibegwam,  <em>et al.</em></li><li class="title"><p>The genome sequence of Drosophila melanogaster</p></li><li class="source"><p>Science, 287 (2000), pp. 2185–2195</p></li><li class="external refPlaceHolder" id="bibsbref15" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref15'] = new Array();
        </script></li><li id="bib16" data-t="r"><ul class="reference" id="bibsbref16"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib16" class="label intra_ref">[16]</a></li><li class="author">A.F. Siegel, G. van den Engh, L. Hood, B. Trask, J.C. Roach</li><li class="title"><p>Modeling the feasibility of whole genome shotgun sequencing using a pairwise end strategy</p></li><li class="source"><p>Genomics, 68 (2000), pp. 237–246</p></li><li class="external refPlaceHolder" id="bibsbref16" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref16'] = new Array();
        </script></li><li id="bib17" data-t="r"><ul class="reference" id="bibsbref17"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib17" class="label intra_ref">[17]</a></li><li class="author">A.M. Phillippy, M.C. Schatz, M. Pop</li><li class="title"><p>Genome assembly forensics: finding the elusive mis-assembly</p></li><li class="source"><p>Genome Biol., 9 (2008), p. R55</p></li><li class="external refPlaceHolder" id="bibsbref17" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref17'] = new Array();
        </script></li><li id="bib18" data-t="r"><ul class="reference" id="bibsbref18"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib18" class="label intra_ref">[18]</a></li><li class="author">J. Kececioglu, J. Ju</li><li class="title"><p>Separating repeats in DNA sequence assembly</p></li><li class="source"><p>Annual Conference on Research in Computational Molecular Biology (2001), pp. 176–183</p></li><li class="external refPlaceHolder" id="bibsbref18" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref18'] = new Array();
        </script></li><li id="bib19" data-t="r"><ul class="reference" id="bibsbref19"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib19" class="label intra_ref">[19]</a></li><li class="author">N. Whiteford, N. Haslam, G. Weber, A. Prugel-Bennett, J.W. Essex, P.L. Roach, M. Bradley, C. Neylon</li><li class="title"><p>An analysis of the feasibility of short read sequencing</p></li><li class="source"><p>Nucleic Acids Res., 33 (2005), p. e171</p></li><li class="external refPlaceHolder" id="bibsbref19" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref19'] = new Array();
        </script></li><li id="bib20" data-t="r"><ul class="reference" id="bibsbref20"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib20" class="label intra_ref">[20]</a></li><li class="author">D.B. Rusch, A.L. Halpern, G. Sutton, K.B. Heidelberg, S. Williamson, S. Yooseph, D. Wu, J.A. Eisen, J.M. Hoffman, K. Remington, K. Beeson, B. Tran, H. Smith, H. Baden-Tillson, C. Stewart, J. Thorpe, J. Freeman, C. Andrews-Pfannkoch, J.E. Venter, K. Li, S. Kravitz, J.F. Heidelberg, T. Utterback, Y.H. Rogers, L.I. Falcon, V. Souza, G. Bonilla-Rosso, L.E. Eguiarte, D.M. Karl, S. Sathyendranath, T. Platt, E. Bermingham, V. Gallardo, G. Tamayo-Castillo, M.R. Ferrari, R.L. Strausberg, K. Nealson, R. Friedman, M. Frazier, J.C. Venter</li><li class="title"><p>The Sorcerer II Global Ocean Sampling expedition: northwest Atlantic through eastern tropical Pacific</p></li><li class="source"><p>PLoS Biol., 5 (2007), p. e77</p></li><li class="external refPlaceHolder" id="bibsbref20" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref20'] = new Array();
        </script></li></ol></div><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_10"></div></div><div id="frag_10" class="page_fragment" data-fid="10" style="display: block;"><div class="refText svRefs"><ol class="references"><li id="bib21" data-t="r"><ul class="reference" id="bibsbref21"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib21" class="label intra_ref">[21]</a></li><li class="author">K. Mavromatis, N. Ivanova, K. Barry, H. Shapiro, E. Goltsman, A.C. McHardy, I. Rigoutsos, A. Salamov, F. Korzeniewski, M. Land, A. Lapidus, I. Grigoriev, P. Richardson, P. Hugenholtz, N.C. Kyrpides</li><li class="title"><p>Use of simulated data sets to evaluate the fidelity of metagenomic processing methods</p></li><li class="source"><p>Nat. Methods, 4 (2007), pp. 495–500</p></li><li class="external refPlaceHolder" id="bibsbref21" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref21'] = new Array();
        </script></li><li id="bib22" data-t="r"><ul class="reference" id="bibsbref22"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib22" class="label intra_ref">[22]</a></li><li class="author">K.E. Wommack, J. Bhavsar, J. Ravel</li><li class="title"><p>Metagenomics: read length matters</p></li><li class="source"><p>Appl. Environ. Microbiol., 74 (2008), pp. 1453–1463</p></li><li class="external refPlaceHolder" id="bibsbref22" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref22'] = new Array();
        </script></li><li id="bib23" data-t="r"><ul class="reference" id="bibsbref23"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib23" class="label intra_ref">[23]</a></li><li class="author">E.W. Myers</li><li class="title"><p>Toward simplifying and accurately formulating fragment assembly</p></li><li class="source"><p>J. Comput. Biol., 2 (1995), pp. 275–290</p></li><li class="external refPlaceHolder" id="bibsbref23" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref23'] = new Array();
        </script></li><li id="bib24" data-t="r"><ul class="reference" id="bibsbref24"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib24" class="label intra_ref">[24]</a></li><li class="author">R.M. Idury, M.S. Waterman</li><li class="title"><p>A new algorithm for DNA sequence assembly</p></li><li class="source"><p>J. Comput. Biol., 2 (1995), pp. 291–306</p></li><li class="external refPlaceHolder" id="bibsbref24" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref24'] = new Array();
        </script></li><li id="bib25" data-t="r"><ul class="reference" id="bibsbref25"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib25" class="label intra_ref">[25]</a></li><li class="author">D.R. Zerbino, E. Birney</li><li class="title"><p>Velvet: algorithms for de novo short read assembly using de Bruijn graphs</p></li><li class="source"><p>Genome Res., 18 (2008), pp. 821–829</p></li><li class="external refPlaceHolder" id="bibsbref25" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref25'] = new Array();
        </script></li><li id="bib26" data-t="r"><ul class="reference" id="bibsbref26"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib26" class="label intra_ref">[26]</a></li><li class="author">P.A. Pevzner, H. Tang, G. Tesler</li><li class="title"><p>De novo repeat classification and fragment assembly</p></li><li class="source"><p>Genome Res., 14 (2004), pp. 1786–1796</p></li><li class="external refPlaceHolder" id="bibsbref26" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref26'] = new Array();
        </script></li><li id="bib27" data-t="r"><ul class="reference" id="bibsbref27"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib27" class="label intra_ref">[27]</a></li><li class="author">D. Zhi, B.J. Raphael, A.L. Price, H. Tang, P.A. Pevzner</li><li class="title"><p>Identifying repeat domains in large genomes</p></li><li class="source"><p>Genome Biol., 7 (2006), p. R7</p></li><li class="external refPlaceHolder" id="bibsbref27" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref27'] = new Array();
        </script></li><li id="bib28" data-t="r"><ul class="reference" id="bibsbref28"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib28" class="label intra_ref">[28]</a></li><li class="author">D. Fasulo, A. Halpern, I. Dew, C. Mobarry</li><li class="title"><p>Efficiently detecting polymorphisms during the fragment assembly process</p></li><li class="source"><p>Bioinformatics, 18 (Suppl 1) (2002), pp. S294–S302</p></li><li class="external refPlaceHolder" id="bibsbref28" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref28'] = new Array();
        </script></li><li id="bib29" data-t="r"><ul class="reference" id="bibsbref29"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib29" class="label intra_ref">[29]</a></li><li class="author">N. Nagarajan, M. Pop</li><li class="title"><p>Parametric complexity of sequence assembly: theory and applications to next generation sequencing</p></li><li class="source"><p>J. Comput. Biol., 16 (2009), pp. 897–908</p></li><li class="external refPlaceHolder" id="bibsbref29" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref29'] = new Array();
        </script></li><li id="bib30" data-t="r"><ul class="reference" id="bibsbref30"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib30" class="label intra_ref">[30]</a></li><li class="author">M. Pop, S.L. Salzberg</li><li class="title"><p>Bioinformatics challenges of new sequencing technology</p></li><li class="source"><p>Trends Genet., 24 (2008), pp. 142–149</p></li><li class="external refPlaceHolder" id="bibsbref30" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref30'] = new Array();
        </script></li><li id="bib31" data-t="r"><ul class="reference" id="bibsbref31"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib31" class="label intra_ref">[31]</a></li><li class="author">R.L. Warren, G.G. Sutton, S.J. Jones, R.A. Holt</li><li class="title"><p>Assembling millions of short DNA sequences using SSAKE</p></li><li class="source"><p>Bioinformatics, 23 (2007), pp. 500–501</p></li><li class="external refPlaceHolder" id="bibsbref31" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref31'] = new Array();
        </script></li><li id="bib32" data-t="r"><ul class="reference" id="bibsbref32"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib32" class="label intra_ref">[32]</a></li><li class="author">R.L. Warren, R.A. Holt</li><li class="title"><p>SSAKE 3.0: Improved speed, accuracy and contiguity</p></li><li class="source"><p>Pacific Symposium on Biocomputing (2008)</p></li><li class="external refPlaceHolder" id="bibsbref32" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref32'] = new Array();
        </script></li><li id="bib33" data-t="r"><ul class="reference" id="bibsbref33"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib33" class="label intra_ref">[33]</a></li><li class="author">J.C. Dohm, C. Lottaz, T. Borodina, H. Himmelbauer</li><li class="title"><p>SHARCGS, a fast and highly accurate short-read assembly algorithm for de novo genomic sequencing</p></li><li class="source"><p>Genome Res., 17 (2007), pp. 1697–1706</p></li><li class="external refPlaceHolder" id="bibsbref33" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref33'] = new Array();
        </script></li><li id="bib34" data-t="r"><ul class="reference" id="bibsbref34"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib34" class="label intra_ref">[34]</a></li><li class="author">W.R. Jeck, J.A. Reinhardt, D.A. Baltrus, M.T. Hickenbotham, V. Magrini, E.R. Mardis, J.L. Dangl, C.D. Jones</li><li class="title"><p>Extending assembly of short DNA sequences to handle error</p></li><li class="source"><p>Bioinformatics, 23 (2007), pp. 2942–2944</p></li><li class="external refPlaceHolder" id="bibsbref34" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref34'] = new Array();
        </script></li><li id="bib35" data-t="r"><ul class="reference" id="bibsbref35"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib35" class="label intra_ref">[35]</a></li><li class="author">J.A. Reinhardt, D.A. Baltrus, M.T. Nishimura, W.R. Jeck, C.D. Jones, J.L. Dangl</li><li class="title"><p>De novo assembly using low-coverage short read sequence data from the rice pathogen Pseudomonas syringae pv. oryzae</p></li><li class="source"><p>Genome Res., 19 (2009), pp. 294–305</p></li><li class="external refPlaceHolder" id="bibsbref35" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref35'] = new Array();
        </script></li><li id="bib36" data-t="r"><ul class="reference" id="bibsbref36"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib36" class="label intra_ref">[36]</a></li><li class="author">S.M. Goldberg, J. Johnson, D. Busam, T. Feldblyum, S. Ferriera, R. Friedman, A. Halpern, H. Khouri, S.A. Kravitz, F.M. Lauro, K. Li, Y.H. Rogers, R. Strausberg, G. Sutton, L. Tallon, T. Thomas, E. Venter, M. Frazier, J.C. Venter</li><li class="title"><p>A Sanger/pyrosequencing hybrid approach for the generation of high-quality draft assemblies of marine microbial genomes</p></li><li class="source"><p>Proc. Natl. Acad. Sci. U. S. A., 103 (2006), pp. 11240–11245</p></li><li class="external refPlaceHolder" id="bibsbref36" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref36'] = new Array();
        </script></li><li id="bib37" data-t="r"><ul class="reference" id="bibsbref37"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib37" class="label intra_ref">[37]</a></li><li class="author">E.W. Myers, G.G. Sutton, A.L. Delcher, I.M. Dew, D.P. Fasulo, M.J. Flanigan, S.A. Kravitz, C.M. Mobarry, K.H. Reinert, K.A. Remington, E.L. Anson, R.A. Bolanos, H.H. Chou, C.M. Jordan, A.L. Halpern, S. Lonardi, E.M. Beasley, R.C. Brandon, L. Chen, P.J. Dunn, Z. Lai, Y. Liang, D.R. Nusskern, M. Zhan, Q. Zhang, X. Zheng, G.M. Rubin, M.D. Adams, J.C. Venter</li><li class="title"><p>A whole-genome assembly of Drosophila</p></li><li class="source"><p>Science, 287 (2000), pp. 2196–2204</p></li><li class="external refPlaceHolder" id="bibsbref37" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref37'] = new Array();
        </script></li><li id="bib38" data-t="r"><ul class="reference" id="bibsbref38"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib38" class="label intra_ref">[38]</a></li><li class="author">S. Batzoglou, D.B. Jaffe, K. Stanley, J. Butler, S. Gnerre, E. Mauceli, B. Berger, J.P. Mesirov, E.S. Lander</li><li class="title"><p>ARACHNE: a whole-genome shotgun assembler</p></li><li class="source"><p>Genome Res., 12 (2002), pp. 177–189</p></li><li class="external refPlaceHolder" id="bibsbref38" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref38'] = new Array();
        </script></li><li id="bib39" data-t="r"><ul class="reference" id="bibsbref39"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib39" class="label intra_ref">[39]</a></li><li class="author">D.B. Jaffe, J. Butler, S. Gnerre, E. Mauceli, K. Lindblad-Toh, J.P. Mesirov, M.C. Zody, E.S. Lander</li><li class="title"><p>Whole-genome sequence assembly for mammalian genomes: Arachne 2</p></li><li class="source"><p>Genome Res., 13 (2003), pp. 91–96</p></li><li class="external refPlaceHolder" id="bibsbref39" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref39'] = new Array();
        </script></li><li id="bib40" data-t="r"><ul class="reference" id="bibsbref40"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib40" class="label intra_ref">[40]</a></li><li class="author">X. Huang, S.P. Yang, Generating a genome assembly with PCAP. Curr Protoc Bioinformatics Chapter 11 (2005) Unit11 3.</li><li class="external refPlaceHolder" id="bibsbref40" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref40'] = new Array();
        </script></li></ol></div><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_11"></div></div><div id="frag_11" class="page_fragment" data-fid="11" style="display: block;"><div class="refText svRefs"><ol class="references"><li id="bib41" data-t="r"><ul class="reference" id="bibsbref41"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib41" class="label intra_ref">[41]</a></li><li class="author">S. Batzoglou</li><li class="title"><p>Algorithmic Challenges in Mammalian Genome Sequence Assembly</p></li><li class="source"><p>M. Dunn, L. Jorde, P. Little, S. Subramaniam (Eds.), Encyclopedia of genomics, proteomics and bioinformatics, John Wiley and Sons, Hoboken (New Jersey) (2005)</p></li><li class="external refPlaceHolder" id="bibsbref41" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref41'] = new Array();
        </script></li><li id="bib42" data-t="r"><ul class="reference" id="bibsbref42"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib42" class="label intra_ref">[42]</a></li><li class="author">M. Pop</li><li class="title"><p>DNA sequence assembly algorithms</p></li><li class="source"><p> McGraw-Hill (Ed.), McGraw-Hill 2006 Yearbook of Science and Technology, McGraw-Hill, New York (2005)</p></li><li class="external refPlaceHolder" id="bibsbref42" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref42'] = new Array();
        </script></li><li id="bib43" data-t="r"><ul class="reference" id="bibsbref43"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib43" class="label intra_ref">[43]</a></li><li class="author">G. Sutton, I. Dew</li><li class="title"><p>Shotgun Fragment Assembly</p></li><li class="source"><p>I. Rigoutsos, G. Stephanopoulos (Eds.), Systems Biology: Genomics, Oxford University Press, New York (2007), pp. 79–117</p></li><li class="external refPlaceHolder" id="bibsbref43" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref43'] = new Array();
        </script></li><li id="bib44" data-t="r"><ul class="reference" id="bibsbref44"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib44" class="label intra_ref">[44]</a></li><li class="author">L. Wang, T. Jiang</li><li class="title"><p>On the complexity of multiple sequence alignment</p></li><li class="source"><p>J. Comput. Biol., 1 (1994), pp. 337–348</p></li><li class="external refPlaceHolder" id="bibsbref44" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref44'] = new Array();
        </script></li><li id="bib45" data-t="r"><ul class="reference" id="bibsbref45"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib45" class="label intra_ref">[45]</a></li><li class="author">M. Margulies, M. Egholm, W.E. Altman, S. Attiya, J.S. Bader, L.A. Bemben, J. Berka, M.S. Braverman, Y.J. Chen, Z. Chen, S.B. Dewell, L. Du, J.M. Fierro, X.V. Gomes, B.C. Godwin, W. He, S. Helgesen, C.H. Ho, G.P. Irzyk, S.C. Jando, M.L. Alenquer, T.P. Jarvie, K.B. Jirage, J.B. Kim, J.R. Knight, J.R. Lanza, J.H. Leamon, S.M. Lefkowitz, M. Lei, J. Li, K.L. Lohman, H. Lu, V.B. Makhijani, K.E. McDade, M.P. McKenna, E.W. Myers, E. Nickerson, J.R. Nobile, R. Plant, B.P. Puc, M.T. Ronan, G.T. Roth, G.J. Sarkis, J.F. Simons, J.W. Simpson, M. Srinivasan, K.R. Tartaro, A. Tomasz, K.A. Vogt, G.A. Volkmer, S.H. Wang, Y. Wang, M.P. Weiner, P. Yu, R.F. Begley, J.M. Rothberg</li><li class="title"><p>Genome sequencing in microfabricated high-density picolitre reactors</p></li><li class="source"><p>Nature, 437 (2005), pp. 376–380</p></li><li class="external refPlaceHolder" id="bibsbref45" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref45'] = new Array();
        </script></li><li id="bib46" data-t="r"><ul class="reference" id="bibsbref46"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib46" class="label intra_ref">[46]</a></li><li class="author">J.R. Miller, A.L. Delcher, S. Koren, E. Venter, B.P. Walenz, A. Brownley, J. Johnson, K. Li, C. Mobarry, G. Sutton</li><li class="title"><p>Aggressive assembly of pyrosequencing reads with mates</p></li><li class="source"><p>Bioinformatics, 24 (2008), pp. 2818–2824</p></li><li class="external refPlaceHolder" id="bibsbref46" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref46'] = new Array();
        </script></li><li id="bib47" data-t="r"><ul class="reference" id="bibsbref47"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib47" class="label intra_ref">[47]</a></li><li class="author">D. Hernandez, P. Francois, L. Farinelli, M. Osteras, J. Schrenzel</li><li class="title"><p>De novo bacterial genome sequencing: millions of very short reads assembled on a desktop computer</p></li><li class="source"><p>Genome Res., 18 (2008), pp. 802–809</p></li><li class="external refPlaceHolder" id="bibsbref47" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref47'] = new Array();
        </script></li><li id="bib48" data-t="r"><ul class="reference" id="bibsbref48"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib48" class="label intra_ref">[48]</a></li><li class="author">M.S. Hossain, N. Azimi, S. Skiena</li><li class="title"><p>Crystallizing short-read assemblies around seeds</p></li><li class="source"><p>BMC Bioinformatics, 10 (Suppl 1) (2009), p. S16</p></li><li class="external refPlaceHolder" id="bibsbref48" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref48'] = new Array();
        </script></li><li id="bib49" data-t="r"><ul class="reference" id="bibsbref49"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib49" class="label intra_ref">[49]</a></li><li class="author">P.A. Pevzner</li><li class="title"><p>1-Tuple DNA sequencing: computer analysis</p></li><li class="source"><p>J. Biomol. Struct. Dyn., 7 (1989), pp. 63–73</p></li><li class="external refPlaceHolder" id="bibsbref49" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref49'] = new Array();
        </script></li><li id="bib50" data-t="r"><ul class="reference" id="bibsbref50"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib50" class="label intra_ref">[50]</a></li><li class="author">P.A. Pevzner, H. Tang, M.S. Waterman</li><li class="title"><p>An Eulerian path approach to DNA fragment assembly</p></li><li class="source"><p>Proc. Natl. Acad. Sci. U. S. A., 98 (2001), pp. 9748–9753</p></li><li class="external refPlaceHolder" id="bibsbref50" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref50'] = new Array();
        </script></li><li id="bib51" data-t="r"><ul class="reference" id="bibsbref51"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib51" class="label intra_ref">[51]</a></li><li class="author">J.T. Simpson, K. Wong, S.D. Jackman, J.E. Schein, S.J. Jones, I. Birol</li><li class="title"><p>ABySS: A parallel assembler for short read sequence data</p></li><li class="source"><p>Genome Res., 19 (2009), pp. 1117–1123</p></li><li class="external refPlaceHolder" id="bibsbref51" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref51'] = new Array();
        </script></li><li id="bib52" data-t="r"><ul class="reference" id="bibsbref52"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib52" class="label intra_ref">[52]</a></li><li class="author">P.A. Pevzner, H. Tang</li><li class="title"><p>Fragment assembly with double-barreled data</p></li><li class="source"><p>Bioinformatics, 17 (Suppl 1) (2001), pp. S225–S233</p></li><li class="external refPlaceHolder" id="bibsbref52" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref52'] = new Array();
        </script></li><li id="bib53" data-t="r"><ul class="reference" id="bibsbref53"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib53" class="label intra_ref">[53]</a></li><li class="author">M. Chaisson, P. Pevzner, H. Tang</li><li class="title"><p>Fragment assembly with short reads</p></li><li class="source"><p>Bioinformatics, 20 (2004), pp. 2067–2074</p></li><li class="external refPlaceHolder" id="bibsbref53" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref53'] = new Array();
        </script></li><li id="bib54" data-t="r"><ul class="reference" id="bibsbref54"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib54" class="label intra_ref">[54]</a></li><li class="author">M.J. Chaisson, P.A. Pevzner</li><li class="title"><p>Short read fragment assembly of bacterial genomes</p></li><li class="source"><p>Genome Res., 18 (2008), pp. 324–330</p></li><li class="external refPlaceHolder" id="bibsbref54" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref54'] = new Array();
        </script></li><li id="bib55" data-t="r"><ul class="reference" id="bibsbref55"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib55" class="label intra_ref">[55]</a></li><li class="author">M.J. Chaisson, D. Brinza, P.A. Pevzner</li><li class="title"><p>De novo fragment assembly with short mate-paired reads: Does the read length matter?</p></li><li class="source"><p>Genome Res., 19 (2009), pp. 336–346</p></li><li class="external refPlaceHolder" id="bibsbref55" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref55'] = new Array();
        </script></li><li id="bib56" data-t="r"><ul class="reference" id="bibsbref56"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib56" class="label intra_ref">[56]</a></li><li class="author">D.R. Zerbino, G.K. McEwen, E.H. Margulies, E. Birney</li><li class="title"><p>Pebble and rock band: heuristic resolution of repeats and scaffolding in the velvet short-read de novo assembler</p></li><li class="source"><p>PLoS One, 4 (2009), p. e8407</p></li><li class="external refPlaceHolder" id="bibsbref56" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref56'] = new Array();
        </script></li><li id="bib57" data-t="r"><ul class="reference" id="bibsbref57"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib57" class="label intra_ref">[57]</a></li><li class="author">J. Butler, I. MacCallum, M. Kleber, I.A. Shlyakhter, M.K. Belmonte, E.S. Lander, C. Nusbaum, D.B. Jaffe</li><li class="title"><p>ALLPATHS: de novo assembly of whole-genome shotgun microreads</p></li><li class="source"><p>Genome Res., 18 (2008), pp. 810–820</p></li><li class="external refPlaceHolder" id="bibsbref57" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref57'] = new Array();
        </script></li><li id="bib58" data-t="r"><ul class="reference" id="bibsbref58"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib58" class="label intra_ref">[58]</a></li><li class="author">I. Maccallum, D. Przybylski, S. Gnerre, J. Burton, I. Shlyakhter, A. Gnirke, J. Malek, K. McKernan, S. Ranade, T.P. Shea, L. Williams, S. Young, C. Nusbaum, D.B. Jaffe</li><li class="title"><p>ALLPATHS 2: small genomes assembled accurately and with high continuity from short paired reads</p></li><li class="source"><p>Genome Biol., 10 (2009), p. R103</p></li><li class="external refPlaceHolder" id="bibsbref58" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref58'] = new Array();
        </script></li><li id="bib59" data-t="r"><ul class="reference" id="bibsbref59"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib59" class="label intra_ref">[59]</a></li><li class="author">R. Li, H. Zhu, J. Ruan, W. Qian, X. Fang, Z. Shi, Y. Li, S. Li, G. Shan, K. Kristiansen, H. Yang, J. Wang</li><li class="title"><p>De novo assembly of human genomes with massively parallel short read sequencing</p></li><li class="source"><p>Genome Res., 20 (2009), pp. 265–272</p></li><li class="external refPlaceHolder" id="bibsbref59" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref59'] = new Array();
        </script></li><li id="bib60" data-t="r"><ul class="reference" id="bibsbref60"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib60" class="label intra_ref">[60]</a></li><li class="author">R. Li, W. Fan, G. Tian, H. Zhu, L. He, J. Cai, Q. Huang, Q. Cai, B. Li, Y. Bai, Z. Zhang, Y. Zhang, W. Wang, J. Li, F. Wei, H. Li, M. Jian, R. Nielsen, D. Li, W. Gu, Z. Yang, Z. Xuan, O.A. Ryder, F.C. Leung, Y. Zhou, J. Cao, X. Sun, Y. Fu, X. Fang, X. Guo, B. Wang, R. Hou, F. Shen, B. Mu, P. Ni, R. Lin, W. Qian, G. Wang, C. Yu, W. Nie, J. Wang, Z. Wu, H. Liang, J. Min, Q. Wu, S. Cheng, J. Ruan, M. Wang, Z. Shi, M. Wen, B. Liu, X. Ren, H. Zheng, D. Dong, K. Cook, G. Shan, H. Zhang, C. Kosiol, X. Xie, Z. Lu, Y. Li, C.C. Steiner, T.T. Lam, S. Lin, Q. Zhang, G. Li, J. Tian, T. Gong, H. Liu, D. Zhang, L. Fang, C. Ye, J. Zhang, W. Hu, A. Xu, Y. Ren, G. Zhang, M.W. Bruford, Q. Li, L. Ma, Y. Guo, N. An, Y. Hu, Y. Zheng, Y. Shi, Z. Li, Q. Liu, Y. Chen, J. Zhao, N. Qu, S. Zhao, F. Tian, X. Wang, H. Wang, L. Xu, X. Liu, T. Vinar, Y. Wang, T.W. Lam, S.M. Yiu,  <em>et al.</em></li><li class="title"><p>The sequence and de novo assembly of the giant panda genome</p></li><li class="source"><p>Nature, 463 (2009), pp. 311–317</p></li><li class="external refPlaceHolder" id="bibsbref60" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref60'] = new Array();
        </script></li></ol></div><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_12"></div></div><div id="frag_12" class="page_fragment" data-fid="12" style="display: block;"><div class="refText svRefs"><ol class="references"><li id="bib61" data-t="r"><ul class="reference" id="bibsbref61"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib61" class="label intra_ref">[61]</a></li><li class="author">R. Li, Y. Li, H. Zheng, R. Luo, H. Zhu, Q. Li, W. Qian, Y. Ren, G. Tian, J. Li, G. Zhou, X. Zhu, H. Wu, J. Qin, X. Jin, D. Li, H. Cao, X. Hu, H. Blanche, H. Cann, X. Zhang, S. Li, L. Bolund, K. Kristiansen, H. Yang, J. Wang</li><li class="title"><p>Building the sequence map of the human pan-genome</p></li><li class="source"><p>Nat. Biotechnol., 28 (2009), pp. 57–63</p></li><li class="external refPlaceHolder" id="bibsbref61" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref61'] = new Array();
        </script></li><li id="bib62" data-t="r"><ul class="reference" id="bibsbref62"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib62" class="label intra_ref">[62]</a></li><li class="author">J.C. Venter, M.D. Adams, E.W. Myers, P.W. Li, R.J. Mural, G.G. Sutton, H.O. Smith, M. Yandell, C.A. Evans, R.A. Holt, J.D. Gocayne, P. Amanatides, R.M. Ballew, D.H. Huson, J.R. Wortman, Q. Zhang, C.D. Kodira, X.H. Zheng, L. Chen, M. Skupski, G. Subramanian, P.D. Thomas, J. Zhang, G.L. Gabor Miklos, C. Nelson, S. Broder, A.G. Clark, J. Nadeau, V.A. McKusick, N. Zinder, A.J. Levine, R.J. Roberts, M. Simon, C. Slayman, M. Hunkapiller, R. Bolanos, A. Delcher, I. Dew, D. Fasulo, M. Flanigan, L. Florea, A. Halpern, S. Hannenhalli, S. Kravitz, S. Levy, C. Mobarry, K. Reinert, K. Remington, J. Abu-Threideh, E. Beasley, K. Biddick, V. Bonazzi, R. Brandon, M. Cargill, I. Chandramouliswaran, R. Charlab, K. Chaturvedi, Z. Deng, V. Di Francesco, P. Dunn, K. Eilbeck, C. Evangelista, A.E. Gabrielian, W. Gan, W. Ge, F. Gong, Z. Gu, P. Guan, T.J. Heiman, M.E. Higgins, R.R. Ji, Z. Ke, K.A. Ketchum, Z. Lai, Y. Lei, Z. Li, J. Li, Y. Liang, X. Lin, F. Lu, G.V. Merkulov, N. Milshina, H.M. Moore, A.K. Naik, V.A. Narayan, B. Neelam, D. Nusskern, D.B. Rusch, S. Salzberg, W. Shao, B. Shue, J. Sun, Z. Wang, A. Wang, X. Wang, J. Wang, M. Wei, R. Wides, C. Xiao, C. Yan,  <em>et al.</em></li><li class="title"><p>The sequence of the human genome</p></li><li class="source"><p>Science, 291 (2001), pp. 1304–1351</p></li><li class="external refPlaceHolder" id="bibsbref62" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref62'] = new Array();
        </script></li><li id="bib63" data-t="r"><ul class="reference" id="bibsbref63"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib63" class="label intra_ref">[63]</a></li><li class="author">S. Diguistini, N.Y. Liao, D. Platt, G. Robertson, M. Seidel, S.K. Chan, T.R. Docking, I. Birol, R.A. Holt, M. Hirst, E. Mardis, M.A. Marra, R.C. Hamelin, J. Bohlmann, C. Breuil, S.J. Jones</li><li class="title"><p>De novo genome sequence assembly of a filamentous fungus using Sanger, 454 and Illumina sequence data</p></li><li class="source"><p>Genome Biol., 10 (2009), p. R94</p></li><li class="external refPlaceHolder" id="bibsbref63" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref63'] = new Array();
        </script></li><li id="bib64" data-t="r"><ul class="reference" id="bibsbref64"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib64" class="label intra_ref">[64]</a></li><li class="author">B. Schmidt, R. Sinha, B. Beresford-Smith, S.J. Puglisi</li><li class="title"><p>A fast hybrid short read fragment assembly algorithm</p></li><li class="source"><p>Bioinformatics, 25 (2009), pp. 2279–2280</p></li><li class="external refPlaceHolder" id="bibsbref64" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref64'] = new Array();
        </script></li><li id="bib65" data-t="r"><ul class="reference" id="bibsbref65"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib65" class="label intra_ref">[65]</a></li><li class="author">A. Sundquist, M. Ronaghi, H. Tang, P. Pevzner, S. Batzoglou</li><li class="title"><p>Whole-genome sequencing and assembly with high-throughput, short-read technologies</p></li><li class="source"><p>PLoS ONE, 2 (2007), p. e484</p></li><li class="external refPlaceHolder" id="bibsbref65" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref65'] = new Array();
        </script></li><li id="bib66" data-t="r"><ul class="reference" id="bibsbref66"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib66" class="label intra_ref">[66]</a></li><li class="author">E.W. Myers</li><li class="title"><p>The fragment assembly string graph</p></li><li class="source"><p>Bioinformatics, 21 (Suppl 2) (2005), pp. ii79–ii85</p></li><li class="external refPlaceHolder" id="bibsbref66" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref66'] = new Array();
        </script></li><li id="bib67" data-t="r"><ul class="reference" id="bibsbref67"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib67" class="label intra_ref">[67]</a></li><li class="author">P. Medvedev, M. Brudno, Ab initio Whole Genome Shotgun Assembly with Mated Short Reads Proceedings of the 12th Annual Research in Computational Biology Conference (RECOMB), 2008.</li><li class="external refPlaceHolder" id="bibsbref67" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref67'] = new Array();
        </script></li><li id="bib68" data-t="r"><ul class="reference" id="bibsbref68"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib68" class="label intra_ref">[68]</a></li><li class="author">R. Li, Y. Li, K. Kristiansen, J. Wang</li><li class="title"><p>SOAP: short oligonucleotide alignment program</p></li><li class="source"><p>Bioinformatics, 24 (2008), pp. 713–714</p></li><li class="external refPlaceHolder" id="bibsbref68" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref68'] = new Array();
        </script></li><li id="bib69" data-t="r"><ul class="reference" id="bibsbref69"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib69" class="label intra_ref">[69]</a></li><li class="author">R. Li, C. Yu, Y. Li, T.W. Lam, S.M. Yiu, K. Kristiansen, J. Wang</li><li class="title"><p>SOAP2: an improved ultrafast tool for short read alignment</p></li><li class="source"><p>Bioinformatics, 25 (2009), pp. 1966–1967</p></li><li class="external refPlaceHolder" id="bibsbref69" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref69'] = new Array();
        </script></li><li id="bib70" data-t="r"><ul class="reference" id="bibsbref70"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib70" class="label intra_ref">[70]</a></li><li class="author">H. Li, J. Ruan, R. Durbin</li><li class="title"><p>Mapping short DNA sequencing reads and calling variants using mapping quality scores</p></li><li class="source"><p>Genome Res., 18 (2008), pp. 1851–1858</p></li><li class="external refPlaceHolder" id="bibsbref70" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref70'] = new Array();
        </script></li><li id="bib71" data-t="r"><ul class="reference" id="bibsbref71"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib71" class="label intra_ref">[71]</a></li><li class="author">B. Langmead, C. Trapnell, M. Pop, S.L. Salzberg</li><li class="title"><p>Ultrafast and memory-efficient alignment of short DNA sequences to the human genome</p></li><li class="source"><p>Genome Biol., 10 (2009), p. R25</p></li><li class="external refPlaceHolder" id="bibsbref71" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref71'] = new Array();
        </script></li><li id="bib72" data-t="r"><ul class="reference" id="bibsbref72"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib72" class="label intra_ref">[72]</a></li><li class="author">A.D. Smith, Z. Xuan, M.Q. Zhang</li><li class="title"><p>Using quality scores and longer reads improves accuracy of Solexa read mapping</p></li><li class="source"><p>BMC Bioinformatics, 9 (2008), p. 128</p></li><li class="external refPlaceHolder" id="bibsbref72" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref72'] = new Array();
        </script></li><li id="bib73" data-t="r"><ul class="reference" id="bibsbref73"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib73" class="label intra_ref">[73]</a></li><li class="author">M.C. Schatz</li><li class="title"><p>CloudBurst: Highly Sensitive Read Mapping with MapReduce</p></li><li class="source"><p>Bioinformatics, 25 (2009), pp. 1363–1369</p></li><li class="external refPlaceHolder" id="bibsbref73" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref73'] = new Array();
        </script></li><li id="bib74" data-t="r"><ul class="reference" id="bibsbref74"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib74" class="label intra_ref">[74]</a></li><li class="author">S.M. Rumble, P. Lacroute, A.V. Dalca, M. Fiume, A. Sidow, M. Brudno</li><li class="title"><p>SHRiMP: accurate mapping of short color-space reads</p></li><li class="source"><p>PLoS Comput. Biol., 5 (2009), p. e1000386</p></li><li class="external refPlaceHolder" id="bibsbref74" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref74'] = new Array();
        </script></li><li id="bib75" data-t="r"><ul class="reference" id="bibsbref75"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib75" class="label intra_ref">[75]</a></li><li class="author">D. Weese, A.K. Emde, T. Rausch, A. Doring, K. Reinert</li><li class="title"><p>RazerS–fast read mapping with sensitivity control</p></li><li class="source"><p>Genome Res., 19 (2009), pp. 1646–1654</p></li><li class="external refPlaceHolder" id="bibsbref75" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref75'] = new Array();
        </script></li><li id="bib76" data-t="r"><ul class="reference" id="bibsbref76"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib76" class="label intra_ref">[76]</a></li><li class="author">Y. Chen, T. Souaiaia, T. Chen</li><li class="title"><p>PerM: efficient mapping of short sequencing reads with periodic full sensitive spaced seeds</p></li><li class="source"><p>Bioinformatics, 25 (2009), pp. 2514–2521</p></li><li class="external refPlaceHolder" id="bibsbref76" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref76'] = new Array();
        </script></li><li id="bib77" data-t="r"><ul class="reference" id="bibsbref77"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib77" class="label intra_ref">[77]</a></li><li class="author">S. Hoffmann, C. Otto, S. Kurtz, C.M. Sharma, P. Khaitovich, J. Vogel, P.F. Stadler, J. Hackermuller</li><li class="title"><p>Fast mapping of short sequences with mismatches, insertions and deletions using index structures</p></li><li class="source"><p>PLoS Comput. Biol., 5 (2009), p. e1000502</p></li><li class="external refPlaceHolder" id="bibsbref77" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref77'] = new Array();
        </script></li><li id="bib78" data-t="r"><ul class="reference" id="bibsbref78"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib78" class="label intra_ref">[78]</a></li><li class="author">K. Schneeberger, J. Hagmann, S. Ossowski, N. Warthmann, S. Gesing, O. Kohlbacher, D. Weigel</li><li class="title"><p>Simultaneous alignment of short reads against multiple genomes</p></li><li class="source"><p>Genome Biol., 10 (2009), p. R98</p></li><li class="external refPlaceHolder" id="bibsbref78" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref78'] = new Array();
        </script></li><li id="bib79" data-t="r"><ul class="reference" id="bibsbref79"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib79" class="label intra_ref">[79]</a></li><li class="author">S.Q. Zhao, J. Wang, L. Zhang, J.T. Li, X. Gu, G. Gao, L. Wei</li><li class="title"><p>BOAT: Basic Oligonucleotide Alignment Tool</p></li><li class="source"><p>BMC Genomics, 10 (Suppl 3) (2009), p. S2</p></li><li class="external refPlaceHolder" id="bibsbref79" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref79'] = new Array();
        </script></li><li id="bib80" data-t="r"><ul class="reference" id="bibsbref80"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib80" class="label intra_ref">[80]</a></li><li class="author">K.J. McKernan, H.E. Peckham, G.L. Costa, S.F. McLaughlin, Y. Fu, E.F. Tsung, C.R. Clouser, C. Duncan, J.K. Ichikawa, C.C. Lee, Z. Zhang, S.S. Ranade, E.T. Dimalanta, F.C. Hyland, T.D. Sokolsky, L. Zhang, A. Sheridan, H. Fu, C.L. Hendrickson, B. Li, L. Kotler, J.R. Stuart, J.A. Malek, J.M. Manning, A.A. Antipova, D.S. Perez, M.P. Moore, K.C. Hayashibara, M.R. Lyons, R.E. Beaudoin, B.E. Coleman, M.W. Laptewicz, A.E. Sannicandro, M.D. Rhodes, R.K. Gottimukkala, S. Yang, V. Bafna, A. Bashir, A. MacBride, C. Alkan, J.M. Kidd, E.E. Eichler, M.G. Reese, F.M. De La Vega, A.P. Blanchard</li><li class="title"><p>Sequence and structural variation in a human genome uncovered by short-read, massively parallel ligation sequencing using two-base encoding</p></li><li class="source"><p>Genome Res., 19 (2009), pp. 1527–1541</p></li><li class="external refPlaceHolder" id="bibsbref80" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref80'] = new Array();
        </script></li></ol></div><!--VALIDHTML--><div class="refText svRefs"><ol class="references"><li id="bib81" data-t="r"><ul class="reference" id="bibsbref81"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib81" class="label intra_ref">[81]</a></li><li class="author">H. Lin, Z. Zhang, M.Q. Zhang, B. Ma, M. Li</li><li class="title"><p>ZOOM! Zillions of oligos mapped</p></li><li class="source"><p>Bioinformatics, 24 (2008), pp. 2431–2437</p></li><li class="external refPlaceHolder" id="bibsbref81" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref81'] = new Array();
        </script></li><li id="bib82" data-t="r"><ul class="reference" id="bibsbref82"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib82" class="label intra_ref">[82]</a></li><li class="author">S. Lee, F. Hormozdiari, C. Alkan, M. Brudno</li><li class="title"><p>MoDIL: detecting small indels from clone-end sequencing with mixtures of distributions</p></li><li class="source"><p>Nat. Methods, 6 (2009), pp. 473–474</p></li><li class="external refPlaceHolder" id="bibsbref82" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref82'] = new Array();
        </script></li><li id="bib83" data-t="r"><ul class="reference" id="bibsbref83"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib83" class="label intra_ref">[83]</a></li><li class="author">F. Hormozdiari, C. Alkan, E.E. Eichler, S.C. Sahinalp</li><li class="title"><p>Combinatorial algorithms for structural variation detection in high-throughput sequenced genomes</p></li><li class="source"><p>Genome Res., 19 (2009), pp. 1270–1278</p></li><li class="external refPlaceHolder" id="bibsbref83" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref83'] = new Array();
        </script></li><li id="bib84" data-t="r"><ul class="reference" id="bibsbref84"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib84" class="label intra_ref">[84]</a></li><li class="author">K. Chen, J.W. Wallis, M.D. McLellan, D.E. Larson, J.M. Kalicki, C.S. Pohl, S.D. McGrath, M.C. Wendl, Q. Zhang, D.P. Locke, X. Shi, R.S. Fulton, T.J. Ley, R.K. Wilson, L. Ding, E.R. Mardis</li><li class="title"><p>BreakDancer: an algorithm for high-resolution mapping of genomic structural variation</p></li><li class="source"><p>Nat. Methods, 6 (2009), pp. 677–681</p></li><li class="external refPlaceHolder" id="bibsbref84" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref84'] = new Array();
        </script></li><li id="bib85" data-t="r"><ul class="reference" id="bibsbref85"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib85" class="label intra_ref">[85]</a></li><li class="author">M. Pop, A. Phillippy, A.L. Delcher, S.L. Salzberg</li><li class="title"><p>Comparative genome assembly</p></li><li class="source"><p>Brief. Bioinform., 5 (2004), pp. 237–248</p></li><li class="external refPlaceHolder" id="bibsbref85" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref85'] = new Array();
        </script></li><li id="bib86" data-t="r"><ul class="reference" id="bibsbref86"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib86" class="label intra_ref">[86]</a></li><li class="author">L.W. Hillier, G.T. Marth, A.R. Quinlan, D. Dooling, G. Fewell, D. Barnett, P. Fox, J.I. Glasscock, M. Hickenbotham, W. Huang, V.J. Magrini, R.J. Richt, S.N. Sander, D.A. Stewart, M. Stromberg, E.F. Tsung, T. Wylie, T. Schedl, R.K. Wilson, E.R. Mardis</li><li class="title"><p>Whole-genome sequencing and variant discovery in C. elegans</p></li><li class="source"><p>Nat. Methods, 5 (2008), pp. 183–188</p></li><li class="external refPlaceHolder" id="bibsbref86" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref86'] = new Array();
        </script></li><li id="bib87" data-t="r"><ul class="reference" id="bibsbref87"><li class="referenceLabel"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bib87" class="label intra_ref">[87]</a></li><li class="author">S.L. Salzberg, D.D. Sommer, M.C. Schatz, A.M. Phillippy, P.D. Rabinowicz, S. Tsuge, A. Furutani, H. Ochiai, A.L. Delcher, D. Kelley, R. Madupu, D. Puiu, D. Radune, M. Shumway, C. Trapnell, G. Aparna, G. Jha, A. Pandey, P.B. Patil, H. Ishihara, D.F. Meyer, B. Szurek, V. Verdier, R. Koebnik, J.M. Dow, R.P. Ryan, H. Hirata, S. Tsuyumu, S. Won Lee, P.C. Ronald, R.V. Sonti, M.A. Van Sluys, J.E. Leach, F.F. White, A.J. Bogdanove</li><li class="title"><p>Genome sequence and rapid evolution of the rice pathogen Xanthomonas oryzae pv. oryzae PXO99A</p></li><li class="source"><p>BMC Genomics, 9 (2008), p. 204</p></li><li class="external refPlaceHolder" id="bibsbref87" data-refres="N"><img class="refimgLoader" alt="" src="./Assembly algorithms for next-generation sequencing data_files/loading_txt_icon.gif"></li></ul><script language="Javascript" type="text/javascript">
             myMap['bibsbref87'] = new Array();
        </script></li></ol></div><!--VALIDHTML--><div class="artFooterContent"><dl class="correspondence" id="cor1"><dt class="label topPadd"><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#bcor1" class="intra_ref auth_corr" title="Corresponding author contact information"><sup></sup></a></dt><dd>Corresponding author.</dd></dl><!--footerNotes--></div><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_13"></div></div><div id="frag_13" class="page_fragment" data-fid="13" style="display: block;"><div class="articleText svArticle" id="vitae"><h2 id="m4vt1">Vitae</h2><div id="vt2" class="vitae"><span class="imgFull"><img class="imgLazyJSB inlineImage" height="155" align="middle" width="111" alt="Image" title="Image" src="./Assembly algorithms for next-generation sequencing data_files/grey_pxl.gif" data-inlimgeid="1-s2.0-S0888754310000492-fx2.jpg" style="display: inline;"></span><div class="imgFull"><p id=""><span id=""><img class="imgLazyJSB inlineImage" height="155" width="111" alt="Full-size image (19 K)" title="Full-size image (19 K)" src="./Assembly algorithms for next-generation sequencing data_files/grey_pxl.gif" data-inlimgeid="1-s2.0-S0888754310000492-fx2.jpg" style="display: inline;"><noscript>&lt;img height="155" border="0" style="vertical-align:bottom" width="111" alt="Full-size image (19 K)" title="Full-size image (19 K)" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0888754310000492-fx2.jpg"&gt;</noscript></span><strong class="boldFont">Jason Rafe Miller</strong> manages software research and development on whole-genomeshotgun assembly at the J. Craig Venter Institute (JCVI). He previously contributed to assembly infrastructure development at TIGR, genome comparison and annotation software at Celera Genomics, and genomics visualization software at GlaxoSmithKline.Mr. Miller received a Master’s degree from University of Pennsylvania and a Bachelor’s degree from New York University.</p></div></div><div id="vt1" class="vitae"><span class="imgFull"><img class="imgLazyJSB inlineImage" height="155" align="middle" width="111" alt="Image" title="Image" src="./Assembly algorithms for next-generation sequencing data_files/grey_pxl.gif" data-inlimgeid="1-s2.0-S0888754310000492-fx1.jpg" style="display: inline;"></span><div class="imgFull"><p id=""><span id=""><img class="imgLazyJSB inlineImage" height="155" width="111" alt="Full-size image (17 K)" title="Full-size image (17 K)" src="./Assembly algorithms for next-generation sequencing data_files/grey_pxl.gif" data-inlimgeid="1-s2.0-S0888754310000492-fx1.jpg" style="display: inline;"><noscript>&lt;img height="155" border="0" style="vertical-align:bottom" width="111" alt="Full-size image (17 K)" title="Full-size image (17 K)" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0888754310000492-fx1.jpg"&gt;</noscript></span><strong class="boldFont">Sergey Koren</strong> a software engineer at the J. Craig Venter Institute, is a Ph.D. student at the University of Maryland, College Park, where he received his Master of Science andBachelor of Science (cum laude, with honors) degrees in Computer Science. His research interests include genome assembly, application of graph analysis to metagenomics, and applications of high-performance computing. He is a contributor to the Celera Assembler, AMOS, and k-mer Tools projects hosted on Source Forge.</p></div></div><div id="vt3" class="vitae"><span class="imgFull"><img class="imgLazyJSB inlineImage" height="155" align="middle" width="111" alt="Image" title="Image" src="./Assembly algorithms for next-generation sequencing data_files/grey_pxl.gif" data-inlimgeid="1-s2.0-S0888754310000492-fx3.jpg" style="display: inline;"></span><div class="imgFull"><p id=""><span id=""><img class="imgLazyJSB inlineImage" height="155" width="111" alt="Full-size image (21 K)" title="Full-size image (21 K)" src="./Assembly algorithms for next-generation sequencing data_files/grey_pxl.gif" data-inlimgeid="1-s2.0-S0888754310000492-fx3.jpg" style="display: inline;"><noscript>&lt;img height="155" border="0" style="vertical-align:bottom" width="111" alt="Full-size image (21 K)" title="Full-size image (21 K)" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0888754310000492-fx3.jpg"&gt;</noscript></span><strong class="boldFont">Granger Sutton</strong> is Senior Director of Informatics at the J. Craig Venter Institute (JCVI). Prior to joining JCVI, Dr. Sutton was a director in the Informatics Research department at Celera Genomics where he developed and managed research programs ingene finding, comparative genomics, and shotgun fragment assembly including the development of the Celera Assembler for assembling the human genome. As Computer Scientist at The Institute for Genomic Research (TIGR), he developed protein homology search, multiple sequence alignment, and shotgun fragment assembly algorithms. Dr.Sutton also worked at AT&amp;T Bell Labs to design and implement office automation software. Dr. Sutton earned his Bachelor's degree in electrical engineering and Doctoratein Computer Science from University of Maryland, College Park, and a Master’s degreein Computer Engineering from Stanford University.</p></div></div></div><!--VALIDHTML--><p class="copyright">Copyright © 2010 Elsevier Inc. All rights reserved.</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_14"></div></div><div data-last="true" class="page_fragment" id="frag_14" data-fid="14" style="display: block;"></div>

               

       
   
        
    </div>
    <div id="footer-area-dummy"></div>
    <div id="footer-area" aria-label="product support" role="contentinfo" style="display: block;">
      <div id="sdFoot">
      <div id="footer" aria-label="Footer" role="navigation">	
  <div class="bottomArticle"></div>
  <div class="padding">
    <div id="linkArea">
      <div class="padding links">
        <div style="float:left;width:0;height:1px;"></div>
        <div id="banner-bottom">
          <a class="S_C_aboutElsevier" href="http://www.elsevier.com/" target="_blank" title="Elsevier homepage (opens in a new window)"></a>
        </div>
        <ul class="col5">
<li><a target="_blank" href="http://www.elsevier.com/solutions/sciencedirect">About ScienceDirect</a></li>
</ul>
<ul class="col5">
<li> <a target="_blank" href="http://www.sciencedirect.com/science/contactus">Contact and support</a></li>
</ul>
<ul class="col5">
<li><a target="_blank" href="http://www.elsevier.com/locate/termsandconditions?utm_source=sciencedirect&amp;utm_medium=link&amp;utm_campaign=terms">Terms and conditions</a></li>
</ul>
<ul class="col5">
<li><a target="_blank" href="http://www.elsevier.com/privacypolicy?utm_source=sciencedirect&amp;utm_medium=link&amp;utm_campaign=privacy">Privacy policy</a></li>
</ul>
        <div class="clear"></div>
        <!-- CSAS will be providing the links-->
      </div>
      <div class="padding copyright">
	<hr>
        Copyright © 2016 Elsevier B.V. or its licensors or contributors. ScienceDirect® is a registered trademark of Elsevier B.V.
      </div>
      <div class="padding" style="padding-bottom:4px;font-size:11px;">
              Cookies are used by this site. To decline or learn more, visit our <a href="https://www.elsevier.com/solutions/sciencedirect/support/cookies" target="_blank">Cookies</a> page.
        </div> 
      
        
          <div class="padding links">
            <div id="goToggleSite"><a class="mLink" site="Mobile" cknm="Site" lnkorg="article" lnkzn="footer" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#">Switch to Mobile Site</a></div>
          </div>
	  <form id="pgReload" method="get" action=""></form>
        
      
    </div>
  </div>
</div> <!--footer-->

      </div><!--sdFooter-->
    </div><!--footer-area-->
    </div>
 </div>
 <div id="rightPane" class="column" aria-label="Related content" role="complementary" style="width: 297px;">
  <div id="rightOuter" class="rightOuter" style="height: 36089.4px;">
    <div id="rightInner" class="rightInner">
       <div class="innerPadding">

    <div id="SD_BIB" class="hidden entry separator appBox sgfNoTitleBar sgfNoGadgetBorder ui-sortable"></div>
 

    
    
        <div id="recommend_related_articles" class="entry separator related rendered firstBlk">
            
                <h2><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#articles" class="toggle" aria-controls="relArtList" aria-expanded="true"><span class="iconArrow">&nbsp;</span>Recommended articles</a></h2>
             
             <ol id="relArtList"><li><ol class="articles"><li class="artTitle"><a title="Applications of next-generation sequencing technologies in functional genomics" href="http://www.sciencedirect.com/science/article/pii/S0888754308001651" class="cLink S_C_artTitle" querystr="?_origin=article&amp;_zone=citeRelatedArt&amp;panel=citeRelatedArt&amp;_piiSuggestedFrom=S0888754310000492&amp;_srcFr=EPFL" style="text-overflow: ellipsis; overflow: hidden; white-space: nowrap;">Applications of next-generation sequencing technologies in functional genomics</a></li><li class="srcTitle" style="text-overflow: ellipsis; overflow: hidden; white-space: nowrap;">2008, Genomics<div class="moreInfo"><a data-url="http://www.sciencedirect.com/science/more/S0888754308001651/2/issn/08887543" class="toggleMoreInfo">more</a></div></li><li class="showMore hidden"><div class="moreInfoContent"><ol class="details"><li>Olena Morozova, Marco A. Marra</li><li class="title">Applications of next-generation sequencing technologies in functional genomics</li><li class="fullSrcTitle">Genomics, Volume 92, Issue 5, November 2008, Pages 255–264</li><li class="docSubType">Review Article</li><div class="abs"></div><ul class="links"><li class="clearfix pdf"><div class="icon_pdf"><a rel="nofollow" title="Download PDF" target="_blank" class="cLink pdfRightPaneLink" href="http://www.sciencedirect.com/science/article/pii/S0888754308001651/pdfft?md5=3071f70545028b0eb6fa6132d71fd80e&amp;pid=1-s2.0-S0888754308001651-main.pdf" querystr="?_origin=article&amp;_zone=citeRelatedArt&amp;panel=citeRelatedArt&amp;_piiSuggestedFrom=S0888754310000492&amp;_srcFr=EPFL"> PDF (317 K)</a><div id="entitlement" class="checkEntitlement" style="display:none"></div></div></li></ul></ol></div></li></ol></li><li><ol class="articles"><li class="artTitle"><a title="Bioinformatics challenges of new sequencing technology" href="http://www.sciencedirect.com/science/article/pii/S016895250800022X" class="cLink S_C_artTitle" querystr="?_origin=article&amp;_zone=citeRelatedArt&amp;panel=citeRelatedArt&amp;_piiSuggestedFrom=S0888754310000492&amp;_srcFr=EPFL" style="text-overflow: ellipsis; overflow: hidden; white-space: nowrap;">Bioinformatics challenges of new sequencing technology</a></li><li class="srcTitle" style="text-overflow: ellipsis; overflow: hidden; white-space: nowrap;">2008, Trends in Genetics<div class="moreInfo"><a data-url="http://www.sciencedirect.com/science/more/S016895250800022X/2/issn/01689525" class="toggleMoreInfo">more</a></div></li><li class="showMore hidden"><div class="moreInfoContent"><ol class="details"><li>Mihai Pop, Steven L. Salzberg</li><li class="title">Bioinformatics challenges of new sequencing technology</li><li class="fullSrcTitle">Trends in Genetics, Volume 24, Issue 3, March 2008, Pages 142–149</li><li class="docSubType">Review Article</li><div class="abs"></div><ul class="links"><li class="clearfix pdf"><div class="icon_pdf"><a rel="nofollow" title="Download PDF" target="_blank" class="cLink pdfRightPaneLink" href="http://www.sciencedirect.com/science/article/pii/S016895250800022X/pdfft?md5=5d070f1c2c58ff4f69960cd3ce8fd034&amp;pid=1-s2.0-S016895250800022X-main.pdf" querystr="?_origin=article&amp;_zone=citeRelatedArt&amp;panel=citeRelatedArt&amp;_piiSuggestedFrom=S0888754310000492&amp;_srcFr=EPFL"> PDF (348 K)</a><div id="entitlement" class="checkEntitlement" style="display:none"></div></div></li></ul></ol></div></li></ol></li><li><ol class="articles"><li class="artTitle"><a title="Quality-score guided error correction for short-read sequencing data using CUDA" href="http://www.sciencedirect.com/science/article/pii/S1877050910001262" class="cLink S_C_artTitle" querystr="?_origin=article&amp;_zone=citeRelatedArt&amp;panel=citeRelatedArt&amp;_piiSuggestedFrom=S0888754310000492&amp;_srcFr=EPFL" style="text-overflow: ellipsis; overflow: hidden; white-space: nowrap;">Quality-score guided error correction for short-read sequencing data using CUDA</a></li><li class="srcTitle" style="text-overflow: ellipsis; overflow: hidden; white-space: nowrap;">2010, Procedia Computer Science<div class="moreInfo"><a data-url="http://www.sciencedirect.com/science/more/S1877050910001262/2/issn/18770509" class="toggleMoreInfo">more</a></div></li><li class="showMore hidden"><div class="moreInfoContent"><ol class="details"><li>Haixiang Shi, Bertil Schmidt, Weiguo Liu, Wolfgang Müller-Wittig</li><li class="title">Quality-score guided error correction for short-read sequencing data using CUDA</li><li class="fullSrcTitle">Procedia Computer Science, Volume 1, Issue 1, May 2010, Pages 1129–1138</li><li class="docSubType">Original Research Article</li><div class="abs"></div><ul class="links"><li class="clearfix pdf"><div class="icon_pdf"><a rel="nofollow" title="Download PDF" target="_blank" class="cLink pdfRightPaneLink" href="http://www.sciencedirect.com/science/article/pii/S1877050910001262/pdfft?md5=0619d3f598fbba3442c900a54872a1f2&amp;pid=1-s2.0-S1877050910001262-main.pdf" querystr="?_origin=article&amp;_zone=citeRelatedArt&amp;panel=citeRelatedArt&amp;_piiSuggestedFrom=S0888754310000492&amp;_srcFr=EPFL"> PDF (233 K)</a><div id="entitlement" class="checkEntitlement" style="display:none"></div></div></li></ul></ol></div></li></ol></li><li><a href="http://www.sciencedirect.com/science/recommended-articles/S0888754310000492" class="viewMoreArticles cLink">View more articles »</a></li></ol>
        </div><script>prs.rt('scr_rec_art')</script>
    
    
    
        <div id="cited_by_articles" class="entry separator related closed">
            <h2><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#articles" class="toggle" aria-controls="relArtList" aria-expanded="false"><span class="iconArrow">&nbsp;</span>Citing articles (<span id="hitcount">329</span>)</a></h2>
            <ol id="citedByList"><li class="note"><p>This article has not been cited.</p></li></ol> 
        </div>
    
    
    
        <div id="SD_CRA" class="hidden entry separator appBox sgfNoTitleBar sgfNoGadgetBorder ui-sortable"></div>  
    
    
        
            <div id="related_book_content" class="entry separator related closed">
                <h2><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#articles" class="toggle" aria-controls="relArtList" aria-expanded="false"><span class="iconArrow">&nbsp;</span>Related book content</a></h2>
                <ol id="relBkList"><li class="note" style="margin-left:15px;"><p>No articles found.</p></li></ol>
            </div>
        
     
    
        <div id="metrics" class="entry separator rendered related" style="display: block;">
            <h2><a id="metrics_info" aria-expanded="true" aria-controls="metrics" class="toggle" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#articles"><span class="iconArrow">&nbsp;</span>Metrics</a></h2>
            <div id="metric_details"><a class="badge" target="_blank" href="http://www.altmetric.com/details.php?template=sciencedirect&amp;doi=10.1016/j.ygeno.2010.03.001" title="Altmetric score of 23. Open Altmetric details page"><img width="120px" height="18px" style="border:0; margin:0;" src="./Assembly algorithms for next-generation sequencing data_files/saved_resource" alt="Altmetric score of 23. Open Altmetric details page"><span class="score">23</span> </a><div class="metrichelp"><a aria-describedby="metrics_info" href="http://www.sciencedirect.com/science?_ob=HelpURL&amp;_file=metrics.htm" target="sdhelp" title="Help (Opens new window)" onclick="var helpWin; helpWin=window.open(&#39;/science?_ob=HelpURL&amp;_file=metrics.htm&#39;,&#39;sdhelp&#39;,&#39;scrollbars=yes,resizable=yes,directories=no,toolbar=no,menubar=no,status=no,width=984,height=738&#39;); helpWin.focus()" class="noul icon_qmarkHelpsci_dir">Help</a></div><ul class="metric_list"><li class="details" style="border-left: 9px solid #FFD140;">2 blogs</li><li class="details" style="border-left: 9px solid #74CFED;">5 Tweeters</li><li class="details" style="border-left: 9px solid #958899;">2 Wikipedia pages</li></ul><div class="metric_references"> <div><h3>Saved to reference managers</h3></div><ul class="metric_refence_list"><li class="details" style="border-left: 9px solid #A60000;">1411 Mendeley readers</li><li class="details" style="border-left: 9px solid #BCD2EF;">77 CiteULike</li><li class="details" style="border-left: 9px solid #FF4040;">2 Connotea</li></ul></div><div class="moreMetricDetails"> <a aria-describedby="metrics_info" aria-haspopup="true" href="http://www.altmetric.com/details.php?template=sciencedirect&amp;doi=10.1016/j.ygeno.2010.03.001">View more details »</a></div></div>
        </div>
     



    
        <div class="appContainerDummy" style="height: 149px; display: block;"></div>
        <div class="appContainer js_application_content fixedPane" data-scroll="19322.85714285714" data-top="687" style="top: 101px; max-height: 149px; width: 297px; right: -538.143px;">
            <ul class="hidden" id="dataRepHolder"></ul>
            
                <div id="SD_AT1P" class="boxLink sgfNoTitleBar sgfNoGadgetBorder ui-sortable"><div id="SD_AT1P_299100" class="containerApplOver invisible"><div id="remote_iframe_0_userPrefsDialog" style="display: none;z-index: 10; left: 0px;top:0px;border: 1px solid #CDF0A4;border-top:0px ;font: smaller;background-color:#EEF7E3; width:99%;font-weight:bold;" class=""></div><div id="maincontremote_iframe_0" style="display:;border-style:none ;border-top:" class="containerAppDetails "><iframe id="remote_iframe_0" onload="sgf.killTimer(&#39;299100&#39;, &#39;SD_AT1P&#39;);" name="remote_iframe_0" src="./Assembly algorithms for next-generation sequencing data_files/ifr.html" style=" align: left;text-align: left;" "="" frameborder="no" scrolling="auto" height="" width="100%"></iframe></div></div><div id="SD_AT1P_301623" class="containerApplOver invisible"><div id="remote_iframe_1_userPrefsDialog" style="display: none;z-index: 10; left: 0px;top:0px;border: 1px solid #CDF0A4;border-top:0px ;font: smaller;background-color:#EEF7E3; width:99%;font-weight:bold;" class=""></div><div id="maincontremote_iframe_1" style="display:;border-style:none ;border-top:" class="containerAppDetails "><iframe id="remote_iframe_1" onload="sgf.killTimer(&#39;301623&#39;, &#39;SD_AT1P&#39;);" name="remote_iframe_1" src="./Assembly algorithms for next-generation sequencing data_files/ifr(1).html" style="text-align: left; height: 194px;" "="" frameborder="no" scrolling="auto" height="" width="100%"></iframe></div></div><div id="SD_AT1P_322844" class="containerApplOver invisible"><div id="remote_iframe_2_userPrefsDialog" style="display: none;z-index: 10; left: 0px;top:0px;border: 1px solid #CDF0A4;border-top:0px ;font: smaller;background-color:#EEF7E3; width:99%;font-weight:bold;" class=""></div><div id="maincontremote_iframe_2" style="display:;border-style:none ;border-top:" class="containerAppDetails "><iframe id="remote_iframe_2" onload="sgf.killTimer(&#39;322844&#39;, &#39;SD_AT1P&#39;);" name="remote_iframe_2" src="./Assembly algorithms for next-generation sequencing data_files/ifr(2).html" style=" align: left;text-align: left;" "="" frameborder="no" scrolling="auto" height="" width="100%"></iframe></div></div><div id="SD_AT1P_324033" class="containerApplOver invisible"><div id="remote_iframe_3_userPrefsDialog" style="display: none;z-index: 10; left: 0px;top:0px;border: 1px solid #CDF0A4;border-top:0px ;font: smaller;background-color:#EEF7E3; width:99%;font-weight:bold;" class=""></div><div id="maincontremote_iframe_3" style="display:;border-style:none ;border-top:" class="containerAppDetails "><iframe id="remote_iframe_3" onload="sgf.killTimer(&#39;324033&#39;, &#39;SD_AT1P&#39;);" name="remote_iframe_3" src="./Assembly algorithms for next-generation sequencing data_files/ifr(3).html" style=" align: left;text-align: left;" "="" frameborder="no" scrolling="auto" height="" width="100%"></iframe></div></div><div id="SD_AT1P_323928" class="containerApplOver"><div id="remote_iframe_4_userPrefsDialog" style="display: none;z-index: 10; left: 0px;top:0px;border: 1px solid #CDF0A4;border-top:0px ;font: smaller;background-color:#EEF7E3; width:99%;font-weight:bold;" class=""></div><div id="maincontremote_iframe_4" style="display:;border-style:none ;border-top:" class="containerAppDetails "><iframe id="remote_iframe_4" onload="sgf.killTimer(&#39;323928&#39;, &#39;SD_AT1P&#39;);" name="remote_iframe_4" src="./Assembly algorithms for next-generation sequencing data_files/ifr(4).html" style=" align: left;text-align: left;" "="" frameborder="no" scrolling="auto" height="" width="100%"></iframe></div></div><div id="SD_AT1P_300959" class="containerApplOver invisible"><div id="remote_iframe_5_userPrefsDialog" style="display: none;z-index: 10; left: 0px;top:0px;border: 1px solid #CDF0A4;border-top:0px ;font: smaller;background-color:#EEF7E3; width:99%;font-weight:bold;" class=""></div><div id="maincontremote_iframe_5" style="display:;border-style:none ;border-top:" class="containerAppDetails "><iframe id="remote_iframe_5" onload="sgf.killTimer(&#39;300959&#39;, &#39;SD_AT1P&#39;);" name="remote_iframe_5" src="./Assembly algorithms for next-generation sequencing data_files/ifr(5).html" style="text-align: left; height: 50px;" "="" frameborder="no" scrolling="auto" height="" width="100%"></iframe></div></div></div>
            

            
                <div id="SD_FTA1" class="boxLink sgfHorToolbar sgfShowMyApps sgfShowMoreLessApps"><div id="sciverse_toolbar" class="sgfSciverseHoriToolBar">	<div class="toolbarfoot">		<div class="toolbarBodyHorizontal" id="toolbarBody">			<div id="toolbar_SD_FTA1" class="sgfInnerToolbar ui-sortable" style="width: 97%; min-height: 155px; text-align: center; margin: 5px auto;"></div>			<div id="showMoreApps">				<a onclick="showMore()" class="plusSign">+</a>				<a id="showMore" onclick="showMore()" class="showMoreLessApps" title="Show more applications">Show more applications</a>			</div>			<div id="showLessApps">				<a onclick="showLess()" class="plusSign">-</a>				<a id="showFew" onclick="showLess()" class="showMoreLessApps" title="Show fewer applications">Show fewer applications</a>			</div>		</div>	</div></div><div class="sgfToolBarToggleImage" style="display: none;">	<div id="sgfSlideBar" class="sgfSideCollapse" onmouseout="sgfToggleToolBarOut();" onmouseover="sgfToggleToolBarOver();" title="undefined"></div>	<div class="clear"></div></div></div>
            
        </div>
    


</div>

<div class="js_workspace_content popup" aria-live="polite" role="region" aria-label="Overlay" aria-describedby="skuip-3" style="top: 101px; display: block; right: -539.143px;">
    <a aria-label="close overlay" title="close overlay" alt="close overlay" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#" id="buttonClose" role="button" class="closeBt">&nbsp;</a>
    <div class="scrollArea"><div aria-label="Go to previous and next in sidebar" role="navigation" class="buttonNavBox">  <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig2" class="previous S_C_prev">« previous figure</a>  <a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig4" class="next S_C_next">next figure »</a></div><dl class="figure"><dt><img src="./Assembly algorithms for next-generation sequencing data_files/1-s2.0-S0888754310000492-gr3.jpg" title="Complexity in K-mer graphs can be diagnosed with read multiplicity information. ..." alt="Complexity in K-mer graphs can be diagnosed with read multiplicity information. ..." class="figure"></dt> <dd><p id="">Complexity in K-mer graphs can be diagnosed with read multiplicity information. In these graphs, edges represented in more reads are drawn with thicker arrows. (a) An errant base call toward the end of a read causes a “spur” or short dead-end branch. The same pattern could be induced by coincidence of zero coverage after polymorphism near a repeat. (b) An errant base call near a read middle causes a “bubble” or alternate path. Polymorphisms between donor chromosomes would be expected to induce a bubble with parity of read multiplicity on the divergent paths. (c) Repeat sequences lead to the “frayed rope” pattern of convergent and divergent paths.</p></dd></dl><ul class="links"><li><a href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#fig3" class="js_article viewInArt"><span class="icon viewInArticle_sci_dir"></span>View in article</a></li><li><span class="icon viewFullsize"></span><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0888754310000492&amp;_image=1-s2.0-S0888754310000492-gr3.jpg&amp;_cid=272501&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=08887543&amp;_pii=S0888754310000492&amp;md5=334599c580859baf4ccd3e08af21e2f7" data-label="labelCaptionfig3" data-ref="" data-format="S0888754310000492" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#gr3" title="View full-size image - Opens in new window">View full-size image</a></li></ul><div id="workSpaceView"></div></div>
</div>

    </div><!--rightInner-->
  </div><!--rightOuter-->
 </div><!--rightPane-->
</div><!--page-area-->



<div style="clear:both;"></div>

<form name="citationInfo" id="citationInfo" method="POST" action="">
    <input type="hidden" name="art_citation" value="">
</form>


<script src="./Assembly algorithms for next-generation sequencing data_files/sdX_1601R2.js"></script>




<script src="./Assembly algorithms for next-generation sequencing data_files/sd_1601R2.js"></script>



<script src="./Assembly algorithms for next-generation sequencing data_files/pa_1601R2.js"></script>


<script>
var sciverse="empty";
var SDM=new Object;
SDM.tm=new Object;
SDM.tm.ahm='h'



SDM.ep=new Object;
SDM.ep.downloadCsvURL = "http://services.elsevier.com/SDWebExport/export/table/csv";
SDM.ep.downloadActionURL = "http://www.sciencedirect.com/science/downloadActions/keyEvent/S0888754310000492/08887543";
SDM.ep.gigyaLib="//cdn.gigya.com/JS/socialize.js?apiKey=3__Bt7bHrGM8mjKPwBuOVjWyMi678aF6-PewcnoHJJ6UR1fS5t9NG9RBbKHjE0oA6J";
SDM.ep.imgUrlPrefix = "http://ars.els-cdn.com/content/image/";
SDM.ep.thumbimgUrlPrefix = "http://ars.els-cdn.com/content/thumbimage/";
SDM.ep.mendeleyJS = "https://www.mendeley.com/minified/bookmarklet.js";
SDM.ep.prDevice = "http://pa.elsevier.com";
SDM.ep.pru = "http://pa.elsevier.com";
SDM.ep.ref2ref2by = "http://www.sciencedirect.com/science/relLink/ajaxReq?_eid=1-s2.0-S0888754310000492";
SDM.ep.staticImgUrlPrefix = "http://cdn.els-cdn.com/sd";
SDM.ep.urlPrefix = "http://www.sciencedirect.com/science";
SDM.ep.urlQualifier = "science";
SDM.ep.urlScheme = "http://www.sciencedirect.com";
SDM.ep.userActionURL = "http://www.sciencedirect.com/science/UserActions";
 SDM.helpUrlDomain = "#";




SDM.pm=new Object;
SDM.pm.cid = '272501';
SDM.pm.contentFamily = "serial";
SDM.pm.contentType = "JL";
SDM.pm.coverDate = '30-Jun-2010';

SDM.pm.doi = '10.1016/j.ygeno.2010.03.001';

SDM.doi = '10.1016/j.ygeno.2010.03.001';
SDM.pm.eid = '1-s2.0-S0888754310000492';
SDM.pm.entitled = true;
 SDM.pm.hubEid = '1-s2.0-S0888754310X00064'; 
SDM.pm.indexType = '';
SDM.pm.issnisbn = "08887543";

SDM.pm.itemWeight = 'FULL-TEXT';
SDM.pm.mod_cid="";
SDM.pm.mod_srcTitle="";
SDM.pm.mod_ISBN = "";
SDM.pm.pdfEid = '1-s2.0-S0888754310000492-main.pdf';
SDM.pm.pdfSize = '448192';
SDM.pm.pii = 'S0888754310000492';
SDM.pm.prodColor="sci_dir";if(SDM.pm.prodColor=="sci_dir")SDM.pm.prodColor="";
 SDM.pm.rev = '20150515T022422506833'; 
 SDM.pm.hrev = 'vH.11'; 
 SDM.pm.pptURL = 'http://services.elsevier.com/SDWebExport/export/figure/'; 

SDM.pm.altmetric_avail = 'TRUE';
SDM.pm.altmetric_helpurl = '/science?_ob=HelpURL&_file=metrics.htm';




SDM.pf = new Object;
SDM.pf.pdfAvail = true;
SDM.pf.reqAllFrags = true;
 SDM.pf.maxElemsAllFrags = 12000
SDM.pf.relatedBkAvail=true;
SDM.pf.mlktAvail = true;
SDM.pf.hpccAvail = true;
SDM.pf.blkSocialLogins=true;


SDM.browserUserMsg=true;


SDM.pf.citedByScAvail = true;
SDM.pf.aetEnabled = false;

 SDM.pf.pdfMenuAvail=false; 


SDM.pf.ads = new Object;
SDM.pf.ads.leftUrl="/science/advertisement/left/num/620905/mainCat/general/cat/general/acct/ddd49027ac058c4db14dac8cc0563abc/isn/08887543/doi/10.1016_j.ygeno.2010.03.001/page/article/artOAstatus/Y";




SDM.pf.dc = new Object;
SDM.pf.dc.retry = 10;
SDM.pf.dc.retryDelay = 300;
SDM.pf.dc.retryFactor = 1.0;
SDM.pf.dc.respObj = '';


SDM.pf.er = new Object();






SDM.pf.frag = new Object();
SDM.pf.frag.fat = '7ad37990ed7b30a52cfd87bd497417e02eb3f36a8a054e1f3645442abb04794ec2861cb83b37a31bbfdee78fb9fc7c2cb91dd138ff6d161d8549183b5114974ab9537a5939139568bed2983f2f9c37a30c0bf80ed48c3c2468f3a7496471fbd1dd734529569177739fcd225291654865ea27102ee05d537118d1f3aee19b9272491f146f96dae8603c7a1b37796584dabc13a92342e01947';
SDM.pf.frag.mapRetry = 20;
SDM.pf.frag.mapRetryDelay = 2000;
SDM.pf.frag.mapRetryFactor = 1.1;
SDM.pf.frag.retry = 5;
SDM.pf.frag.retryDelay = 2000;
SDM.pf.frag.foldFactor = 1.5;
SDM.pf.frag.hashWait = 5000;






SDM.pf.gadget = new Object();
SDM.pf.gadget.displayGadgets=true;
SDM.pf.gadget.sciverseUrlPfx="http://www.sciencedirect.com";
SDM.pf.gadget.sciverseJsonObject='';
SDM.pf.gadget.sciverseGadgetDetailsID = "";
SDM.pf.gadget.svShindigLib="http://sciverse-shindig.elsevier.com/gadgets/js/shindig-container:pubsub-2.js?c=1";
SDM.pf.gadget.svSciverseLib="http://www.sciencedirect.com/gadgetservices/gadgets/v9/js/SciverseShindigService,GadgetToolBar,jquery-outside,jquery.ba-bbq.min,SciverseShindigContainer,ScienceDirectShindigServices.js";



SDM.pf.mrwmod = new Object();
SDM.pf.mrwmod.breadCrumbPathURL = "http://www.sciencedirect.com/science/module/breadCrumbPaths/isbn//pii/S0888754310000492";

SDM.pf.mrwmod.ldrAvail = true;


 


SDM.pf.oa = new Object();
SDM.pf.oa.item=true;
SDM.pf.oa.archive=true;



SDM.pf.oa.userLicense='http%3A%2F%2Fwww.elsevier.com%2Fopen-access%2Fuserlicense%2F1.0%2F';


SDM.pf.ref = new Object();
SDM.pf.ref.bs = 20;
SDM.pf.ref.citedByURL = "http://www.sciencedirect.com/science/referenceResolution/ajaxRefResol?_updateCitedBy=";
SDM.pf.ref.fd = 300;

SDM.pf.ref.resolveURL = "http://www.sciencedirect.com/science/referenceResolution/ajaxRefResol?_pii=S0888754310000492&_refCnt=87&_docType=article";





SDM.pf.toc=new Object;
SDM.pf.toc.retry = 20;
SDM.pf.toc.retryDelay = 1000;
SDM.pf.toc.retryFactor = 1.09;
SDM.pf.toc.outlineGraphics = false;
SDM.pf.toc.outlineTab = "";

SDM.pf.toc.urlTOCLink="http://www.sciencedirect.com/science/article/pii/S0888754310000492";
SDM.pf.toc.urlTOCLinkQueryStr="?_rdoc=1&_fmt=full&_origin=&md5=22543fb95e34d36391c44b9d9cd72533";

SDM.pf.ddm = new Object();
SDM.pf.ddm.endpoint = "http://services.elsevier.com/SDWebExport/export/pdfdownload/pdf";






SDM.si = new Object();
SDM.si.webUserId = "12975512";
SDM.si.sds = "a8324bf8e4a6db6a3a86609c9b4054e460a2b8ea28c0389818bdace4153228520736e3d077b1ee62";
SDM.si.euidCookieObject='b3647403c71e5000';

SDM.pageTransKey = "01/21/2016 17:38:43 EST#2845_002#9637#195.113.224.147";
SDM.pageType = "article_full";
SDM.tid = "b9b54578-c08f-11e5-a7ed-00000aacb35d";
SDM.ud = "12975512||C000228598|291352|GUEST";
SDM.ieVersion="8";

SDM.dac = new Object;
SDM.svapi = SVApi;

</script>




<script src="./Assembly algorithms for next-generation sequencing data_files/importer-api_1601R2.js.min"></script>


<script>

initializeArticlePage();
</script>




<script type="text/javascript">/*{literal}<![CDATA[*/window.lightningjs||function(c){function g(b,d){d&&(d+=(/\?/.test(d)?"&":"?")+"lv=1");c[b]||function(){var i=window,h=document,j=b,g=h.location.protocol,l="load",k=0;(function(){function b(){a.P(l);a.w=1;c[j]("_load")}c[j]=function(){function m(){m.id=e;return c[j].apply(m,arguments)}var b,e=++k;b=this&&this!=i?this.id||0:0;(a.s=a.s||[]).push([e,b,arguments]);m.then=function(b,c,h){var d=a.fh[e]=a.fh[e]||[],j=a.eh[e]=a.eh[e]||[],f=a.ph[e]=a.ph[e]||[];b&&d.push(b);c&&j.push(c);h&&f.push(h);return m};return m};var a=c[j]._={};a.fh={};a.eh={};a.ph={};a.l=d?d.replace(/^\/\//,(g=="https:"?g:"http:")+"//"):d;a.p={0:+new Date};a.P=function(b){a.p[b]=new Date-a.p[0]};a.w&&b();i.addEventListener?i.addEventListener(l,b,!1):i.attachEvent("on"+l,b);var q=function(){function b(){return["<head></head><",c,' onload="var d=',n,";d.getElementsByTagName('head')[0].",d,"(d.",g,"('script')).",i,"='",a.l,"'\"></",c,">"].join("")}var c="body",e=h[c];if(!e)return setTimeout(q,100);a.P(1);var d="appendChild",g="createElement",i="src",k=h[g]("div"),l=k[d](h[g]("div")),f=h[g]("iframe"),n="document",p;k.style.display="none";e.insertBefore(k,e.firstChild).id=o+"-"+j;f.frameBorder="0";f.id=o+"-frame-"+j;/MSIE[ ]+6/.test(navigator.userAgent)&&(f[i]="javascript:false");f.allowTransparency="true";l[d](f);try{f.contentWindow[n].open()}catch(s){a.domain=h.domain,p="javascript:var d="+n+".open();d.domain='"+h.domain+"';",f[i]=p+"void(0);"}try{var r=f.contentWindow[n];r.write(b());r.close()}catch(t){f[i]=p+'d.write("'+b().replace(/"/g,String.fromCharCode(92)+'"')+'");d.close();'}a.P(2)};a.l&&setTimeout(q,0)})()}();c[b].lv="1";return c[b]}var o="lightningjs",k=window[o]=g(o);k.require=g;k.modules=c}({});
window.usabilla_live = lightningjs.require("usabilla_live", "//w.usabilla.com/11e800d34be0.js");
/*]]>{/literal}*/
if(typeof SDM != "undefined") {
  if(SDM.ud) { usabilla_live("data", {"custom": {"ud":SDM.ud} }); }
  if(SDM.pf&&SDM.pf.oa) {
    if(SDM.pf.oa.item) { usabilla_live("data", {"custom": {"open_access":SDM.pf.oa.item} });}
    if(SDM.pf.oa.archive) { usabilla_live("data", {"custom": {"open_archive":SDM.pf.oa.archive} });}
  }
}
</script>


<div id="sdpa"></div>
<script>SDPA.init();</script>
<div id="appBackground"></div>
<div id="appDialog" class="files" style="position: fixed;">
    <div id="header">
        <h1 class="headline">Download PDFs</h1>
        <img id="logo" class="sd-bubbles" src="./Assembly algorithms for next-generation sequencing data_files/sciencedirect.svg" onerror="this.src=&#39;http://cdn.els-cdn.com/sd/sciencedirect.png&#39;">
        <a class="close" id="closeWindow" href="http://www.sciencedirect.com/science/article/pii/S0888754310000492#"><img src="./Assembly algorithms for next-generation sequencing data_files/close-icon.png"></a>
    </div>
    <div id="contentBlk" class="view">
        <div class="help_button" id="ddmMainHelp"><a class="ddmHelp" target="sdhelp" href="http://www.sciencedirect.com/science?_ob=HelpURL&amp;_file=DM_main.htm&amp;_btn=Y&amp;_zone=TopNavBar&amp;_origin=article" title="Help (Opens new window)" onclick="var helpWin; helpWin=window.open(&#39;/science?_ob=HelpURL&amp;_file=DM_main.htm&amp;_btn=Y&amp;_zone=TopNavBar&amp;_origin=article&#39;,&#39;sdhelp&#39;,&#39;scrollbars=yes,resizable=yes,directories=no,toolbar=no,menubar=no,status=no,width=984,height=738&#39;); helpWin.focus(); return false">Help</a></div>
        <div class="help_button" id="ddmNewformatHelp"><a class="ddmHelp" target="sdhelp" href="http://www.sciencedirect.com/science?_ob=HelpURL&amp;_file=DM_newformat.htm&amp;_btn=Y&amp;_zone=TopNavBar&amp;_origin=article" title="Help (Opens new window)" onclick="var helpWin; helpWin=window.open(&#39;/science?_ob=HelpURL&amp;_file=DM_newformat.htm&amp;_btn=Y&amp;_zone=TopNavBar&amp;_origin=article&#39;,&#39;sdhelp&#39;,&#39;scrollbars=yes,resizable=yes,directories=no,toolbar=no,menubar=no,status=no,width=984,height=738&#39;); helpWin.focus(); return false">Help</a></div>
        <div id="filesTemplate"></div>
        <div id="namingTemplate"></div> 
    </div>
    <div class="footer"></div>
</div>








<img width="0" height="0" src="./Assembly algorithms for next-generation sequencing data_files/resource" style="display: none;"><img width="0" height="0" src="./Assembly algorithms for next-generation sequencing data_files/resource(1)" style="display: none;"><div class="usabilla_live_button_container" tabindex="0" aria-hidden="true" style="width:117px;height:34px;z-index:600;bottom:0px;right:2%;position:fixed"><iframe src="" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" data-tags="bottom" title="Visual Feedback Button" style="width: 117px; height: 34px; margin: 0px; padding: 0px; border: 0px; overflow: hidden; z-index: 9998; position: absolute; left: 0px; top: 0px; box-shadow: 0px 0px 0px; background-color: transparent;"></iframe></div><img width="0" height="0" src="./Assembly algorithms for next-generation sequencing data_files/pageReport" style="display: none;"><img width="0" height="0" src="./Assembly algorithms for next-generation sequencing data_files/idx" style="display: none;"><div id="sgfGhost" style="display: none;" class="ui-sortable"><div id="sgfGhost_299152" class="containerApplOver invisible"><div id="footergadgets-gadget-title-bar-6" class="curveBoxleftNormal ActiveGadget  sgfOpen"><img src="./Assembly algorithms for next-generation sequencing data_files/icon.jpg" border="none" class="headingGadgetIcon " title="Entitle Gadget"><h3 title="Move Application" id="GadgetName6" class="elsevierTimeLOver truncateTitle">Entitle Gadget</h3><div id="Controls_6" class="Controls menuControlParent" align="right"><ul class="menu"><li class="menuControls"> <a href="javascript:void(0);" onclick="shindig.container.getGadget(6).handleToggle(this);"><div id="toggleBtn_6" class="sprite-close openCloseBtn" title="Close This Application" style="background-image: url(&#39;http://www.sciencedirect.com/gadgetservices/gadgets/v8/images/gsframeworksprite.png&#39;);"></div></a></li><li class="menuControls"><a href="javascript:void(0);" id="button-remote_iframe_6_userPrefsDialog" onclick="shindig.container.getGadget(6).openMenuOptions();return false;"><div class="sprite-options" title="Open Options" style="background-image: url(&#39;http://www.sciencedirect.com/gadgetservices/gadgets/v8/images/gsframeworksprite.png&#39;);"></div></a></li><ul class="submenu" id="sb-remote_iframe_6_userPrefsDialog"></ul></ul></div><div id="Close_6" class="Controls_close" style="display:none;"><a href="javascript:shindig.container.getGadget(6).closeGadget();"><div class="sprite-close_hoover" title="Close" style="background-image: url(&#39;http://www.sciencedirect.com/gadgetservices/gadgets/v8/images/gsframeworksprite.png&#39;);"></div></a></div></div><div id="remote_iframe_6_userPrefsDialog" style="display: none;z-index: 10; left: 0px;top:0px;border: 1px solid #CDF0A4;border-top:0px ;font: smaller;background-color:#EEF7E3; width:99%;font-weight:bold;" class=""></div><div id="maincontremote_iframe_6" style="display:;border-style:solid ;border-top:" class="containerAppDetails "><iframe id="remote_iframe_6" onload="sgf.killTimer(&#39;299152&#39;, &#39;sgfGhost&#39;);" name="remote_iframe_6" src="./Assembly algorithms for next-generation sequencing data_files/ifr(6).html" style=" align: left;text-align: left;" "="" frameborder="no" scrolling="auto" height="" width="100%"></iframe></div></div></div></body></html>