%%% Bibliography (literature used as a source)
%%%
%%% We employ bibTeX to construct the bibliography. It processes
%%% citations in the text (e.g., the \cite{...} macro) and looks up
%%% relevant entries in the bibliography.bib file.
%%%
%%% The \bibliographystyle command selects, which style will be used
%%% for references from the text. The argument in curly brackets is
%%% the name of the corresponding style file (*.bst). Both styles
%%% mentioned in this template are included in LaTeX distributions.

% \bibliographystyle{plainnat}    %% Author (year)
% \bibliographystyle{unsrt}     %% [number]
\bibliographystyle{ieeetr}

\renewcommand{\bibname}{References}

%%% Generate the bibliography. Beware that if you cited no works,
%%% the empty list will be omitted completely.
% \bibliography{References}

%%% If case you prefer to write the bibliography manually (without bibTeX),
%%% you can use the following. Please follow the ISO 690 standard and
%%% citation conventions of your field of research.

\begin{thebibliography}{99}

\bibitem{alg-compare}
{\sc Li, } Zhenyu, {\sc Chen,} Yanxiang, {\sc Mu,} Desheng, {\sc Yuan,} Jianying, {\sc Shi,} Yujian, {\sc Zhang,} Hao, {\sc Gan,} Jun, {\sc Li,} Nan, {\sc Hu,} Xuesong, {\sc Binghang Liu,} Yang, {\sc Bicheng,} Fan Wu:
\emph{Comparison of two major classes of assembly algorithms: overlap-major-consensus and de-bruijn graph}.
Advance Access, 19 December 2011

\bibitem{dbg-apply}
{\sc Compeau,} Phillip C., {\sc Pevzner,} Pavel A., {\sc Tesler,} Glenn:
\emph{How to apply de Bruijn graphs to genome assembly}
Nature Biotechnology, Volume 29, Number 11, November 2011

\bibitem{haplocall}
{\sc d'Auvera,} Geraldine: 
\emph{HC step 2: Local re-assembly and haplotype determination}
https://gatkforums.broadinstitute.org/gatk/discussion/4146/hc-step-2-local-re-assembly-and-haplotype-determination
20 Jul 2017

\bibitem{haplocall-1}
{\sc GATK Dev Team}: 
\emph{HC step 1: Defining ActiveRegions by measuring data entropy}
https://software.broadinstitute.org/gatk/documentation/article.php?id=4147
20 Jul 2017

\bibitem{abyss}
{\sc Simpson,} Jared T., {\sc Wong,} Kim, {\sc Jackman,} Shuan D., {\sc Schein,} Jacqueline E., {\sc Jones,} Steven J.M., Birol Innac: 
\emph{AbySS: A parallel assembler for short read seuqnce data}
Cold Spring Harbor Laboratory, 2009
ISBN 1088-9051/09

\bibitem{vcf-format}
{\sc Genome Research Limited}: 
\emph{The Variant Call Format Specification (VCFv4.3 and BCFv2.2)}
https://github.com/samtools/hts-specs.
18 Jul 2016

\bibitem{fermikit}
{\sc Li,} Heng: 
\emph{FermiKit: assembly-based variant calling for Illumina resequencing data}
Cornell University Library, 24. 4. 2015
https://arxiv.org/abs/1504.06574
20 Jul 2017

\bibitem{fermi-lite}
{\sc Li,} Heng: 
\emph{The Fermi-lite project}
https://github.com/lh3/fermi-lite
20 Jul 2017

\bibitem{sambam}
{\sc Genome Research Limited}: 
\emph{Sequence Alignment / Map Format Specification}
The SAM/BAM Format Specification Working Group
https://github.com/samtools/hts-specs
28 Aug 2015

\bibitem{testreads}
{\sc 100 Genomes Project}:
\emph{chr1 Illumina high coverage Read set}
ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/pilot2\_high\_cov\_GRCh37\_bams/data/NA12878/alignment/
20 Jul 2017

\bibitem{testvcf}
{\sc 100 Genomes Project}: 
\emph{VCF files for the Illumina test read set}
ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/
20 Jul 2017

\bibitem{testref}
{\sc 100 Genomes Project}: 
\emph{Reference sequence GRCh37}
http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human\_g1k\_v37.fasta.gz
20 Jul 2017

\bibitem{samtools}
{\sc Genome Research Limited}:
\emph{The SAMTools project}
http://www.htslib.org/
20 Jul 2017

\bibitem{rtgeval}
{\sc Li,} Heng: 
\emph{The rtgeval project}
https://github.com/lh3/rtgeval
20 Jul 2017


\end{thebibliography}
