\chapter{Results}
\label{chap:results}

When an algorithm is being designed, its evaluation against existing solutions belongs to important stages of its development. This chapter describes this stage, informing about a data set used to debug and improve our solution, and the method of comparison with other solutions, such as fermi-kit and GATK. The last part of the chapter covers certain variants that proved to be interesting when examined by our algorithm.

\section{Test Data Set}
\label{sec:test-data-set}

The algorithm was tested on the first 40 megabases of chromosome 1 of the human genome. The test set is a high-coverage one and was obtained from the 1000 Genome Project. Except the input reads, variants called by fermi.kit and GATK are also available in form of VCF files. The VCF files were used as a measure of algorithm quality. Since our algorithm also requires a reference sequence to work, we took the GRCh37 version.

The input read set consists of 12475011 reads with lengh of 151 bases. Figure \ref{fig:test-kmer-frequency-distribution} shows k-mer frequency distribution of the set with k-mer size of 21 bases. The shape of the graph, when compared to Figure \ref{fig:kmer-frequency-distribution} suggests that the set contains read errors. Hence, an error correction step was applied.

As Table \ref{tab:test-correction} indicates, the error correction process removed certain amount or shortened of reads. About 21 \% of the input reads was subject to repairs. Figure \ref{fig:test-repair-frequency} shows a distribution of the number of repaired bases per read, not including effects of read shortage. It is clear, that in most cases, only several bases were fixed. Figure \ref{fig:test-pos-repair-frequency} shows the repair frequency at individual read positions. It suggests that most bases were repaired at read's beginning or end, alghtough the difference is not too signifficant.

Figure \ref{fig:test-kmer-frequence-distribution2} shows the k-mer frequency distribution of the corrected read set. Although quite far from perfect, the graph shape definitely resembles the ideal one better then in case of the raw  read set. 

As described in Section \ref{sec:data-preprocessing}, not all input reads, even from the corrected set, can be processed by our algorithm. Table \ref{tab:corrected-set-categories} summarizes numbers of reads unacceptable for various reasons. The preprocessing phase removed nearly one fifth of the corrected data set (19.23 \%). Most of the reads were removed due to being possible duplicates (86 \%). Quite a large portion of  reads were not accepted because of their low mapping quality (14 \%). Also, about 3 \% of all the reads were shortened in order to remove soft-clipped regions.

\begin{itemize}
\item describe the data set used for tests and its sources,
\item describe algorithms and software to evaluate the results (fermikit, fermikit run at individual regions, GATK, mpileup of samtools, rtgeval for evaluation),
\item Show the position-based and genotype results of rtgeval.
\item describe interesting variants (variants that are not found by other software, explain some false negatives, demonstrate that graph optimizations actually revealed some variants...).
\end{itemize}

BAM:
ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/pilot2_high_cov_GRCh
37_bams/data/NA12878/alignment/

VCF:
ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/

Reference:
http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human_g1k_v37.fasta.gz



Total reads:         12475011
Bad reads:           2398872 (19.23 %)
Zero POS:            0 (0.00 %)
Bad MAPQ:            339617 (14.16 %)
Unmapped:            36517 (1.52 %)
Supplementary:       33639 (1.40 %)
Secondary:           0 (0.00 %)
Duplicate:           2074173 (86.46 %)
Soft clipped:        325801 (3.23 %)
Hard clipped:        0 (0.00 %)
Both clipped:        0 (0.00 %)
Correcting reads...
Total reads:         12475011
Bad reads:           2398872 (19.23 %)
Zero POS:            0 (0.00 %)
Bad MAPQ:            339617 (14.16 %)
Unmapped:            36517 (1.52 %)
Supplementary:       33639 (1.40 %)
Secondary:           0 (0.00 %)
Duplicate:           2074173 (86.46 %)
Soft clipped:        305205 (3.03 %)
Hard clipped:        0 (0.00 %)
Both clipped:        0 (0.00 %)
K:                  21
Total reads:        12475011
Removed reads:      72065
Shortened reads:    944
Total bases:        1880123991
Total repairs:      5096756
Repair count distribution:
0,      9855395 79.00 %
1,      1673028 13.41 %
2,      388808  3.12 %
3,      169914  1.36 %
4,      96470   0.77 %
5,      61740   0.49 %
6,      41508   0.33 %
7,      29129   0.23 %
8,      20605   0.17 %
9,      15239   0.12 %
10,     11585   0.09 %
11,     8775    0.07 %
12,     6710    0.05 %
13,     5213    0.04 %
14,     4035    0.03 %
15,     3192    0.03 %
16,     2514    0.02 %
17,     2071    0.02 %
18,     1668    0.01 %
19,     1276    0.01 %
20,     1025    0.01 %
21,     765     0.01 %
22,     633     0.01 %
23,     460     0.00 %
24,     363     0.00 %
25,     259     0.00 %
26,     171     0.00 %
27,     127     0.00 %
28,     92      0.00 %
29,     63      0.00 %
30,     41      0.00 %
31,     28      0.00 %
32,     22      0.00 %
33,     7       0.00 %
34,     3       0.00 %
35,     4       0.00 %
36,     3       0.00 %
38,     1       0.00 %
39,     3       0.00 %
41,     1       0.00 %

Repair base position distribution:
0,      55240   1.08 %
1,      47995   0.94 %
2,      47409   0.93 %
3,      47206   0.93 %
4,      46168   0.91 %
5,      47002   0.92 %
6,      46480   0.91 %
7,      45120   0.89 %
8,      44134   0.87 %
9,      43195   0.85 %
10,     42162   0.83 %
11,     45054   0.88 %
12,     41804   0.82 %
13,     40194   0.79 %
14,     40568   0.80 %
15,     41678   0.82 %
16,     38689   0.76 %
17,     37328   0.73 %
18,     36880   0.72 %
19,     35356   0.69 %
20,     38042   0.75 %
21,     35397   0.69 %
22,     38590   0.76 %
23,     34153   0.67 %
24,     35025   0.69 %
25,     33639   0.66 %
26,     35271   0.69 %
27,     33486   0.66 %
28,     33542   0.66 %
29,     32552   0.64 %
30,     32949   0.65 %
31,     32626   0.64 %
32,     32180   0.63 %
33,     34169   0.67 %
34,     33159   0.65 %
35,     31646   0.62 %
36,     30569   0.60 %
37,     34107   0.67 %
38,     30156   0.59 %
39,     30814   0.60 %
40,     29847   0.59 %
41,     38036   0.75 %
42,     39758   0.78 %
43,     35484   0.70 %
44,     28678   0.56 %
45,     29447   0.58 %
46,     28846   0.57 %
47,     28710   0.56 %
48,     28503   0.56 %
49,     28171   0.55 %
50,     32662   0.64 %
51,     27354   0.54 %
52,     26591   0.52 %
53,     26732   0.52 %
54,     27319   0.54 %
55,     27240   0.53 %
56,     26642   0.52 %
57,     26992   0.53 %
58,     40677   0.80 %
59,     26883   0.53 %
60,     26531   0.52 %
61,     25771   0.51 %
62,     26634   0.52 %
63,     26422   0.52 %
64,     25195   0.49 %
65,     26969   0.53 %
66,     26999   0.53 %
67,     26377   0.52 %
68,     26778   0.53 %
69,     25507   0.50 %
70,     28997   0.57 %
71,     25893   0.51 %
72,     25870   0.51 %
73,     25341   0.50 %
74,     29023   0.57 %
75,     25360   0.50 %
76,     28735   0.56 %
77,     25324   0.50 %
78,     25937   0.51 %
79,     25938   0.51 %
80,     29922   0.59 %
81,     25969   0.51 %
82,     27181   0.53 %
83,     26668   0.52 %
84,     26945   0.53 %
85,     26585   0.52 %
86,     25476   0.50 %
87,     26030   0.51 %
88,     26823   0.53 %
89,     25706   0.50 %
90,     26647   0.52 %
91,     26559   0.52 %
92,     40129   0.79 %
93,     26894   0.53 %
94,     26499   0.52 %
95,     27060   0.53 %
96,     27008   0.53 %
97,     26193   0.51 %
98,     26316   0.52 %
99,     27088   0.53 %
100,    32332   0.63 %
101,    27965   0.55 %
102,    28472   0.56 %
103,    28255   0.55 %
104,    29021   0.57 %
105,    29303   0.57 %
106,    28341   0.56 %
107,    35126   0.69 %
108,    39478   0.77 %
109,    37792   0.74 %
110,    29896   0.59 %
111,    30447   0.60 %
112,    30025   0.59 %
113,    33605   0.66 %
114,    30448   0.60 %
115,    30955   0.61 %
116,    32561   0.64 %
117,    33280   0.65 %
118,    32000   0.63 %
119,    31928   0.63 %
120,    32800   0.64 %
121,    32064   0.63 %
122,    33056   0.65 %
123,    33850   0.66 %
124,    35010   0.69 %
125,    33737   0.66 %
126,    35199   0.69 %
127,    34252   0.67 %
128,    37958   0.74 %
129,    34727   0.68 %
130,    37360   0.73 %
131,    35288   0.69 %
132,    36878   0.72 %
133,    37110   0.73 %
134,    38409   0.75 %
135,    40703   0.80 %
136,    40572   0.80 %
137,    39782   0.78 %
138,    41047   0.81 %
139,    44559   0.87 %
140,    41550   0.82 %
141,    42470   0.83 %
142,    43407   0.85 %
143,    44826   0.88 %
144,    45778   0.90 %
145,    46764   0.92 %
146,    46373   0.91 %
147,    46835   0.92 %
148,    46653   0.92 %
149,    47741   0.94 %
150,    55163   1.08 %